---
title: "AspPna_gk"
author: "GK"
date: "12/20/2022"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r}
library(mia)
library(miaViz)
library(scater)
library(dplyr)
library(ggplot2)
library(patchwork)
# library("tidyverse")
# library(ggsignif)
library(ggpubr)
library(vegan)
library(picante)
require(cowplot)
require(ggrepel)
library(ecodist)
library(pheatmap)
library(dendextend)
# library(tidySummarizedExperiment)
# library(ANCOMBC)
# library(ALDEx2)
library(knitr)
library(reshape2)
# library(palmerpenguins)
library(ggforce)
# library(DESeq2)
# library(scran)
library(ggordiplots)
library(bluster)
library(DirichletMultinomial)                                                                                                               
library(parallel)  
library(tableone)
library(edgeR)
library(survival)
library(survminer)
library(data.table)
library(stringr)
library(vegan)
library(reshape2)
library(ggplot2)
#library(plotly)
library(ggpubr)
library(tibble)
library(plyr)
library(tools)
library(remotes)
library("viridis")
#library(FSA)
library(readxl)
library(psych)
library(epiDisplay)
# library(indicspecies) # https://onlinelibrary.wiley.com/doi/abs/10.1111/j.1600-0706.2010.18334.x
library(rstatix)
#library(RVAideMemoire)
#library(compositions)
library("tidyr") 
library(ggbeeswarm)
library(dplyr)
library(jpeg)
library(survminer)
library(phyloseq)
library(ANCOMBC)

library(biclust)
library("ComplexHeatmap")

library(ggcorrplot)
library(lubridate)
```

```{r set working directory}
# set a working directory
getwd()
setwd(getwd())

```

```{r import datasets}
######################

# It creates one data and meta-data files for Aspiration Pneumonia
##############
# Load the data
counts_file = './data/0102_ALIR_2022.taxa.genus.cmF.cln.Clean.summary_table.tsv'
counts <- read.csv(counts_file,sep = '\t')
row.names(counts) <- counts$sample_id
counts <- counts[,-1]
# 
#### import clinical metadata and merge with the SampleID files that contain the sequencing sample identifiers. 
alirSampleID_5.4.22 <- read.csv(file = "./data/alirSampleID_5.4.22.csv")

### import the aspiration metadata file by Vi. 

aspirationPneumonia.meta.data_Vi <- read.csv(file = "./data/aspirationPneumonia.meta.data.csv")

names(aspirationPneumonia.meta.data_Vi)
### keep only the pertinent variables of the sequencing sampleID files and the specific ones for aspiration. 
aspirationPneumonia.meta.data_Vi <- subset(aspirationPneumonia.meta.data_Vi, select = c(sample_id,total, sampletype, depleted_tracheal, study_id_num, study_id, study_day,  SubjectID_day.x,study_day,  rs_soiled, qPCR_16S_mean, RiskFactorAspPna, AspPnaGroup,  SumRF, Aspiration.y, Stroke_history,  Dementia,  Bulbar_dysfunction_hx,  AMS_onIntub, Seizure_beforeIntub, DrugOD, DrugOD_Stimulant, ETOH_Intox_or_Withdrawal, Esophag_Path_hx, Antiacid_chronic, Home02, SNFresident.y, Witnessed_Aspiration, Vomiting_Hx, SuspectedAspirationbyClinician, BasilarDensityinCXR, RadiologistReportAspiration))

aspirationPneumonia.meta.data_Vi$SubjectIDDay <- aspirationPneumonia.meta.data_Vi$SubjectID_day.x

#### SELECT ELIGIBLE SAMPLES FOR ASP PNEUMONIA PROJECT INCLUDING CONTROLS
table(aspirationPneumonia.meta.data_Vi$sampletype)
##### add value to AspPnaGroup variable
table(aspirationPneumonia.meta.data_Vi$study_id)

aspirationPneumonia.meta.data_Vi$AspPnaGroup[aspirationPneumonia.meta.data_Vi$study_id=="LHMP"] <- "HealthyControls"
## add value to AspPnaGroup variable for experimental controls. 
aspirationPneumonia.meta.data_Vi$AspPnaGroup[aspirationPneumonia.meta.data_Vi$study_id=="expercontrol"] <- "expercontrol"

##### filter for aspiration pneumonia project cases
aspirationPneumonia.meta.data_Vi <- filter(aspirationPneumonia.meta.data_Vi, !is.na(AspPnaGroup)) # 864 entries

### merge with updated clinical dataset

aspirationPneumonia.meta.data_Vi <- merge(aspirationPneumonia.meta.data_Vi, alirSampleID_5.4.22, by = "SubjectIDDay", all.x = TRUE)

### create flowchart 
 aspirationPneumonia.meta.data_ViforMergewithSampleIDforFlowchart <- subset(aspirationPneumonia.meta.data_Vi, select = c(SubjectID, AspPnaGroup))
 aspirationPneumonia.meta.data_ViforMergewithSampleIDforFlowchart <- filter(aspirationPneumonia.meta.data_ViforMergewithSampleIDforFlowchart, !is.na(SubjectID))
alirSubjectID_5.4.22 <- filter(alirSampleID_5.4.22, studydayfollowup=="baseline")
alirSubjectID_5.4.22_Aspiration <- merge(aspirationPneumonia.meta.data_ViforMergewithSampleIDforFlowchart, alirSubjectID_5.4.22, by = "SubjectID", all.y = TRUE)

n_distinct(alirSubjectID_5.4.22_Aspiration$SubjectID) # 832
alirSubjectID_5.4.22_Aspiration <- filter(alirSubjectID_5.4.22_Aspiration, studydayfollowup=="baseline")
alirSubjectID_5.4.22_Aspiration <- filter(alirSubjectID_5.4.22_Aspiration, !duplicated(SubjectID))


#### filter for period between April 2015 and February 2020

alirSubjectID_5.4.22_Aspiration <- filter(alirSubjectID_5.4.22_Aspiration, StudyDate > "2015-04-01") # 703
alirSubjectID_5.4.22_Aspiration <- filter(alirSubjectID_5.4.22_Aspiration, StudyDate < "2020-03-01") # 471

table(alirSubjectID_5.4.22_Aspiration$GroupClassification.x, alirSubjectID_5.4.22_Aspiration$Pneumonia)

table(alirSubjectID_5.4.22_Aspiration$GroupClassification.x, is.na(alirSubjectID_5.4.22_Aspiration$qpcr16S_eta), alirSubjectID_5.4.22_Aspiration$Pneumonia)

###### include convex antibiotic scores. 
total_ABx_score_Convex <- read.csv(file = "./data/total_ABx_score_Convex.csv")
aspirationPneumonia.meta.data_Vi$SubjectID
total_ABx_score_Convex$SubjectID <- total_ABx_score_Convex$X
aspirationPneumonia.meta.data_Vi <- merge(aspirationPneumonia.meta.data_Vi, total_ABx_score_Convex, by = "SubjectID", all.x = TRUE)

ggplot(aspirationPneumonia.meta.data_Vi, aes(x=AspPnaGroup, y = total_ABx_score_day1_Convex)) + geom_boxplot() + geom_jitter() + stat_compare_means()

ggsave("./results/total_ABx_score_day1_ConvexVsAspPnaGroup.jpg")
#### include the CxClassification data

CxClassificationAspPNA <- read.csv(file = "./data/CxClassificationAspPNA.csv")

aspirationPneumonia.meta.data_Vi <- merge(aspirationPneumonia.meta.data_Vi, CxClassificationAspPNA, by = "SubjectID", all.x = TRUE)


aspirationPneumonia.meta.data_Vi %>% filter(!is.na(CxClassification)) %>% ggplot(aes(x=CxClassification, y = log(BacterialReads_nanopore))) + geom_boxplot() + geom_jitter() + stat_compare_means()



# rename as AspMetaData
AspMetadata <- aspirationPneumonia.meta.data_Vi
AspMetadata <- filter(AspMetadata, !duplicated(sample_id))
ids = row.names(counts)
# 
matched_ids1 = which(ids%in%AspMetadata$sample_id)
matched_ids2 = which(AspMetadata$sample_id%in%ids)
# # Available Samples
taxatableAsp <- counts[matched_ids1,]
AspMetadata <- AspMetadata[matched_ids2,]
table(AspMetadata$AspPnaGroup)
#### replace new names for the groups. 
AspMetadata$AspPnaGroup[AspMetadata$AspPnaGroup=="MacroAsP"] <- "MAsP"
AspMetadata$AspPnaGroup[AspMetadata$AspPnaGroup=="NonAsP"] <- "NonMAsP"
AspMetadata$AspPnaGroup[AspMetadata$AspPnaGroup=="Ctrl"] <- "IntubControl"


AspMetadata$AspPnaGroup <- ordered(AspMetadata$AspPnaGroup, levels = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"))

############### descriptive analysis on risk factors. 
# chronic factors
table(AspMetadata$Stroke_history) 
table(AspMetadata$Dementia) 
table(AspMetadata$Bulbar_dysfunction_hx) 
table(AspMetadata$Esophag_Path_hx) 
table(AspMetadata$Antiacid_chronic) 
table(AspMetadata$Home02) 

# acute factors
table(AspMetadata$AMS_onIntub) 
table(AspMetadata$Seizure_beforeIntub) 
table(AspMetadata$DrugOD) 
table(AspMetadata$ETOH_Intox_or_Withdrawal) 


AspMetadata$ChronicRF <- AspMetadata$Stroke_history + AspMetadata$Dementia + AspMetadata$Bulbar_dysfunction_hx + AspMetadata$Esophag_Path_hx + AspMetadata$Antiacid_chronic + AspMetadata$Home02

AspMetadata$AcuteRF <- AspMetadata$AMS_onIntub + AspMetadata$Seizure_beforeIntub + AspMetadata$DrugOD + AspMetadata$ETOH_Intox_or_Withdrawal

table(AspMetadata$ChronicRF, AspMetadata$AcuteRF)


AspMetadata <- AspMetadata %>% mutate(ChronicRF_categ  = case_when(ChronicRF>1 ~ "yes",
                                                                                     ChronicRF<2 ~ "no"
                                                                                  ))

AspMetadata <- AspMetadata %>% mutate(AcuteRF_categ  = case_when(AcuteRF>1 ~ "yes",
                                                                                     AcuteRF<2 ~ "no"
                                                                                  ))

AspMetadata %>% ggplot(aes(ChronicRF)) + geom_histogram()
AspMetadata %>% ggplot(aes(AcuteRF)) + geom_histogram()

AspMetadata %>% ggplot(aes(AcuteRF)) + geom_bar()
AspMetadata %>% ggplot(aes(ChronicRF)) + geom_bar()

table(AspMetadata$ChronicRF_categ, AspMetadata$AcuteRF_categ)

##### distributed in different categories. either high acute or high chronic aspiration risk factor. 
#### define new variable. 
##### break down the pneumonia cohort into those with chronic vs. acute vs. no aspiration risk factor. 

AspMetadata <- AspMetadata %>% mutate(RiskFactor_categ  = case_when(AcuteRF_categ == "yes" ~ "AcuteRF",
                                                                                      ChronicRF_categ == "yes" ~ "ChronicRF",
                                                                                      ChronicRF_categ == "no" &  AcuteRF_categ == "no"~ "None"
                                                                                     
                                                                                  ))

table(AspMetadata$RiskFactor_categ)


######## incorporate anaerobe antibiotics. 

tblAntibioticHistory = read.csv("./data/tblAntibioticHistory.csv") 

names(tblAntibioticHistory)

tblAntibioticHistory <- filter(tblAntibioticHistory, day1Date !="")


tblAntibioticHistory$ZosynStartDate

      tblAntibioticHistory <- tblAntibioticHistory %>% rowwise() %>%
        mutate(onZosyn = ifelse(between(day1Date, ZosynStartDate, ZosynEndDate), 1, 0),
               onMetronidazole = ifelse(between(day1Date, MetronidazoleStartDate, MetronidazoleEndDate), 1, 0),
               onMeropenem = ifelse(between(day1Date, MeropenemStartDate, MeropenemEndDate), 1, 0),
               onSulbactam = ifelse(between(day1Date, SulbactamStartDate, SulbactamEndDate), 1, 0))

      table(tblAntibioticHistory$onZosyn)
      table(tblAntibioticHistory$onMetronidazole)
      
      
      tblAntibioticHistory <- tblAntibioticHistory %>% rowwise() %>%
        mutate(day1AnaerobicAbx = ifelse(max(onZosyn, onMetronidazole, onMeropenem,onSulbactam) == 1, "yes", "no"))
      

      tblAntibioticHistory$day1OnABX

##### take only the pertinent variables by subjectid
      
tblAntibioticHistory <- subset(tblAntibioticHistory, select = c(SubjectID, onZosyn, onMetronidazole, onMeropenem,onSulbactam, day1OnABX, day1AnaerobicAbx ))

AspMetadata <- merge(AspMetadata, tblAntibioticHistory, by = "SubjectID", all.x = TRUE)


write.csv(AspMetadata,file = "./results/AspMetadata_12.20.22.csv")
write.table(taxatableAsp,file = "./results/taxatableAsp_12.20.22.csv",row.names=TRUE, sep = ",")
############# Descriptive Table Analyses
AspMetadata = read.csv("./results/AspMetadata_12.20.22.csv")
taxatableAsp = read.table(file = "./results/taxatableAsp_12.20.22.csv",row.names=1, sep = ",")
AspMetadata$AspPnaGroup <- ordered(AspMetadata$AspPnaGroup, levels = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"))
n_distinct(AspMetadata$SubjectID) # 159 cases
### create SubjectID file for descriptive table analyses
AspMetadataSubjectID <- filter(AspMetadata, !is.na(SubjectID))
#### obtain only day 1 variables
AspMetadataSubjectID <- filter(AspMetadataSubjectID, StudyDay==1)
#### remove "duplicates" to obtain single row for each subject
AspMetadataSubjectID <-  filter(AspMetadataSubjectID, !duplicated(SubjectID)) # 158
table(AspMetadataSubjectID$AspPnaGroup) # 91, 56, 11
#####


median(AspMetadataSubjectID$timesamplefromintubation) # 1
quantile(AspMetadataSubjectID$timesamplefromintubation, na.rm = TRUE, probs  = c(0.33, 0.66)) # 1-1

table(AspMetadataSubjectID$timesamplefromintubation)
table(AspMetadataSubjectID$AspPnaGroup)

##### create tableone for aspPnagroups. 

vars_continuous <- c("Age","BMI", "WBC",  "HGB", 
                     "Platelets", "Creatinine", "BUN", "CO2", "Temperature", "il6", "il8", "il10", "tnfr1_final", "st2", "fractalkine",  "rage",  "ang2", "procalcitonin", "pentraxin3",  "il6_eta_pradj", "il8_eta_pradj",  "il10_eta_pradj", "tnfr1_eta_pradj", "st2_eta_pradj",  "fractalkine_eta_pradj", "rage_eta_pradj",  "ang2_eta_pradj", "procalcitonin_eta_pradj","pentraxin3_eta_pradj" , "totProt_ETA", "IgM_ETA_pradj", 
  #### additional biomarkers
  "Urea_eta" ,    "gdf15_eta_final_pradj"  ,     "qpcr16S_eta_pradj"    ,               
"TSP1_eta_pradj"  ,                    "MPO_eta_pradj"        ,              
"NeutElas_eta_pradj"      ,            "SPD_ETA_pradj"     ,                 
 "M65_ETA_pradj"               ,                    
"mcp1_eta_pradj"        ,              "cathepsinS_eta_pradj"        ,       
"ccl5_eta_pradj"                ,      "cd14_eta_pradj"            ,         
"pf4_eta_pradj"          ,             "il1beta_eta_pradj",                  
"mmp2_eta_pradj"           ,           "mmp8_eta_pradj"         ,            
"mmp9_eta_pradj"       ,               "s100a8_eta_pradj"      ,             
"slpi_eta_pradj"             ,         "factorb_ngml_eta_pradj"     ,        
 "hemoglobin_nM_eta_pradj"      , "chemenn_eta_pradj",    "il1beta_eta_pradj", 
"gdf15"     , "fabp2", "spd", "cd14", "LBP", "fabp4", "spd_luminex" , "CH50", "AH50", "FactorB", "Properdin", "FactorH", "nDNA", "mtDNA", "human_MPM", "TotalMicrobial_mcfDNA", "Bacteria_mcfDNA" , "bdg", "NET_OD_mean"	, "NET_sigl_ratio",                                          "HR" , "RR", "SBP", "SaturationSpO2", "PHa", "PaCO2", "TidalVolume", "TotalRR", "MinuteVolumeTotal", "PEEP" , "PeakInspiratory", "PlateauPressure", "PlateauPr_imput", "VentRatio",  "WorstPaO2FiO2Ratio",   "SOFAScRSD" , "lips_score", "VFD", "total_ABx_score_day1_Convex")


###### define mortality 14 day
AspMetadataSubjectID <- AspMetadataSubjectID %>% mutate(Mortality14Day = case_when(DODfromIntubation<15 ~ 1, DODfromIntubation>14 ~ 0, is.na(DODfromIntubation) ~ 0))

AspMetadataSubjectID$Mortality14Day

vars_categorical <- c( "Gender", "Race",  "HistDiabetes", 
          "HistCOPD", "HistCCF",   "HistCRF",   "IsActiveNeoplasm",   "HistImmunosuppression" , "HistCLD", "HistPulmonaryFibrosis", "group",    "HistAlcohol",   "EndOfStayType",    "DischargeTo",  "Mortality14Day",  "Mortality30Day",   "Mortality90Day" ,   "VasopressorFirstWeek",   "phenotype", "CalfeePhenotype", "SinhaPhenotype", 
          ### aspiration risk factors
          "Stroke_history"       ,               "Dementia"      ,                      "Bulbar_dysfunction_hx"     ,          "AMS_onIntub"       ,                 
"Seizure_beforeIntub"    ,             "DrugOD"            ,                  "DrugOD_Stimulant"       ,             "ETOH_Intox_or_Withdrawal"     ,      
"Esophag_Path_hx"     ,                "Antiacid_chronic"         ,           "Home02"            ,                  "SNFresident.y"      ,                
"Witnessed_Aspiration"     ,           "Vomiting_Hx"             ,            "SuspectedAspirationbyClinician"    ,  "BasilarDensityinCXR"      ,          
"RadiologistReportAspiration", "onZosyn", "onMetronidazole", "onMeropenem" , "onSulbactam", "day1OnABX.y", "day1AnaerobicAbx", "CxClassification") 

#### by AspPnaGroups
AspMetadataSubjectID$AspPnaGroup <- as.character(AspMetadataSubjectID$AspPnaGroup)
cat_table_AspPna<- print(CreateCatTable(vars = vars_categorical, strata = "AspPnaGroup" , data=AspMetadataSubjectID, test = TRUE, testExact = fisher.test, includeNA = FALSE), pDigits = 2)
con_table_AspPna <- print(CreateContTable(vars = vars_continuous, strata = "AspPnaGroup" , data=AspMetadataSubjectID), nonnormal = vars_continuous, digits  = 1, pDigits = 2)

write.csv(cat_table_AspPna, "./results/cat_table_AspPna.csv")
write.csv(con_table_AspPna, "./results/con_table_AspPna.csv")


####### table among pneumonia subjects for survivors vs. nonsurvivors

AspMetadataSubjectID_pna <- filter(AspMetadataSubjectID, AspPnaGroup !="IntubControl")


cat_table_Mortality<- print(CreateCatTable(vars = vars_categorical, strata = "Mortality90Day" , data=AspMetadataSubjectID_pna, test = TRUE, testExact = fisher.test, includeNA = FALSE), pDigits = 2)
con_table_Mortality <- print(CreateContTable(vars = vars_continuous, strata = "Mortality90Day" , data=AspMetadataSubjectID_pna), nonnormal = vars_continuous, digits  = 1, pDigits = 2)

write.csv(cat_table_Mortality, "./results/cat_table_Mortality.csv")
write.csv(con_table_Mortality, "./results/con_table_Mortality.csv")


AspMetadataSubjectID_pna <- filter(AspMetadataSubjectID, AspPnaGroup !="IntubControl")

#### insightful difference. 
survival_fit_bygroup90 <- survfit(Surv(survival_time90_fromICUAdmission, survival_status90)~RiskFactor_categ , data=AspMetadataSubjectID_pna)
summary(survival_fit_bygroup90)
ggsurvplot(survival_fit_bygroup90, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days", risk.table = TRUE, tables.theme = clean_theme()) 


AspMetadataSubjectID_MASP <- filter(AspMetadataSubjectID_pna, AspPnaGroup=="MAsP")

survival_fit_bygroup90 <- survfit(Surv(survival_time90_fromICUAdmission, survival_status90)~RiskFactor_categ , data=AspMetadataSubjectID_MASP)
summary(survival_fit_bygroup90)
ggsurvplot(survival_fit_bygroup90, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days", risk.table = TRUE, tables.theme = clean_theme()) 


AspMetadataSubjectID_NonMASP <- filter(AspMetadataSubjectID_pna, AspPnaGroup=="NonMAsP")

survival_fit_bygroup90 <- survfit(Surv(survival_time90_fromICUAdmission, survival_status90)~RiskFactor_categ , data=AspMetadataSubjectID_NonMASP)
summary(survival_fit_bygroup90)
ggsurvplot(survival_fit_bygroup90, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days", risk.table = TRUE, tables.theme = clean_theme()) 




cat_table_RiskFactors<- print(CreateCatTable(vars = vars_categorical, strata = "RiskFactor_categ" , data=AspMetadataSubjectID_pna, test = TRUE, testExact = fisher.test, includeNA = FALSE), pDigits = 2)
con_table_RiskFactors <- print(CreateContTable(vars = vars_continuous, strata = "RiskFactor_categ" , data=AspMetadataSubjectID_pna), nonnormal = vars_continuous, digits  = 1, pDigits = 2)

write.csv(cat_table_RiskFactors, "./results/cat_table_RiskFactors.csv")
write.csv(con_table_RiskFactors, "./results/con_table_RiskFactors.csv")




#### from these insightful graphs and analyses, include new survival curves in the manuscript to address the Reviewer 2 comments on risk factors. 

###### survival analysis by AspGroup
###### create a dataframe with alir cases only and baseline only samples. 
# create 60 day survival
AspMetadataSubjectID$survival_time60_fromICUAdmission <- AspMetadataSubjectID$survival_time90_fromICUAdmission
AspMetadataSubjectID$survival_time60_fromICUAdmission[AspMetadataSubjectID$survival_time60_fromICUAdmission>60] <-61
AspMetadataSubjectID$survival_status60 <- AspMetadataSubjectID$survival_status90
table(AspMetadataSubjectID$survival_status60, as.factor(AspMetadataSubjectID$survival_time60_fromICUAdmission>60))
# no events beyond day 60 recorded



survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~AspPnaGroup , data=AspMetadataSubjectID)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days", risk.table = TRUE, tables.theme = clean_theme()) 

# publishable plot
### #
surv_AspPnagroup <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE, title = "60 day survival by Aspiration Diagnosis", legend.labs=c("MAsP",  "NonMAsP", "IntubControl"), palette = c("#000000", "#E69F00", "#56B4E9"), risk.table = TRUE, tables.theme = clean_theme(), legend.title = "Group:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation")  
jpeg("./results/surv_AspPnagroup.jpeg", width = 600, height = 500)
print(surv_AspPnagroup, newpage = FALSE)
while (!is.null(dev.list()))  dev.off()


table(AspMetadataSubjectID$RiskFactor_categ)

AspMetadataSubjectID %>% ggplot(aes(AcuteRF)) + geom_bar()
ggsave("./results/AcuteRFburden.jpeg")
AspMetadataSubjectID %>% ggplot(aes(ChronicRF)) + geom_bar()
ggsave("./results/ChronicRFburden.jpeg")

######## look at MASP by risk factors. 

AspMetadataSubjectID_MASP <- filter(AspMetadataSubjectID, AspPnaGroup=="MAsP")



survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~RiskFactor_categ , data=AspMetadataSubjectID_MASP)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days", risk.table = TRUE, tables.theme = clean_theme()) 

# publishable plot
### #
surv_MASP_RF <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE, title = "60 day survival by Risk Factor Category in MAsP", legend.labs=c("AcuteRF",  "ChronicRF", "None"), palette = c("#CC0033", "#006633", "#333333"), risk.table = TRUE, tables.theme = clean_theme(), legend.title = "Group:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation")  
jpeg("surv_MASP_RF.jpeg", width = 600, height = 500)
print(surv_MASP_RF, newpage = FALSE)
dev.off()


AspMetadataSubjectID_NonMASP <- filter(AspMetadataSubjectID, AspPnaGroup=="NonMAsP")


survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~RiskFactor_categ , data=AspMetadataSubjectID_NonMASP)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days", risk.table = TRUE, tables.theme = clean_theme()) 

# publishable plot
### #
surv_NonMASP_RF <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE, title = "60 day survival by Risk Factor Category in NonMAsP", legend.labs=c("AcuteRF",  "ChronicRF", "None"), palette = c("#CC0033", "#006633", "#333333"), risk.table = TRUE, tables.theme = clean_theme(), legend.title = "Group:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation")  
jpeg("surv_NonMASP_RF.jpeg", width = 600, height = 500)
print(surv_NonMASP_RF, newpage = FALSE)
dev.off()





```


```{r quality control microbiome analyses}

######################## CREATE A single DATASET for all samples to look for differences between healthy controls, ICU patients and experimental controls. 
#### create Baseline dataset for all 
AspMetadata$studydayfollowup[AspMetadata$AspPnaGroup=="HealthyControls"] <- "baseline"
AspMetadata$studydayfollowup[is.na(AspMetadata$AspPnaGroup)] <- "baseline"
# change the oral wash to oral swab
table(AspMetadata$sampletype)
# AspMetadata$sampletype[AspMetadata$sampletype=="lhmp_oralwash"] <- "oral_swab"
# AspMetadata$sampletype[AspMetadata$sampletype=="lhmp_bal"] <- "tracheal"
AspMetadata$sampletype[AspMetadata$sampletype=="mobioextrneg"] <- "NegativeControl"
AspMetadata$sampletype[AspMetadata$sampletype=="pcrneg"] <- "NegativeControl"
AspMetadata$sampletype[AspMetadata$sampletype=="pcrposzymo"] <- "PositiveControl"
AspMetadata$sampletype[AspMetadata$sampletype=="pcrposzymo"] <- "PositiveControl"

# ensure that metadata and taxa table match for the oral dataset
allSamplesMetadata <- filter(AspMetadata, sampletype!="rectal_swab")
 allSamplesMetadata <- filter(allSamplesMetadata, sampletype!="stool") #  # 648 observations.
 allSamplesMetadata <- filter(allSamplesMetadata, studydayfollowup == "baseline") #  # 496 observations.

# set rownames by sample_id
rownames(allSamplesMetadata) <- allSamplesMetadata$sample_id
alirmatchingids_v <- as.character(allSamplesMetadata$sample_id)
AllTaxatableBas <- subset(taxatableAsp, rownames(taxatableAsp) %in% alirmatchingids_v)
#### arrange by rownames. 
allSamplesMetadata <-allSamplesMetadata[sort(rownames(allSamplesMetadata)), ]
AllTaxatableBas <-AllTaxatableBas[sort(rownames(AllTaxatableBas)), ]
# ensure all rows equal
all.equal(rownames(allSamplesMetadata), rownames(AllTaxatableBas))

table(allSamplesMetadata$sampletype)
  ##########################################################
  # remove total
  AllTaxatableBas = AllTaxatableBas[,-1]
  
  ####################################################################################
  # Creating a SingleCellExperiment object for the downstream analysis
  ################################################################################
  # separate column names by taxa
  allSamples_df <- data.frame(x = names(AllTaxatableBas))
  taxa <- allSamples_df %>% separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                               "Family", "Genus"), sep = "\\.")
  # use Genus as column names
  names(AllTaxatableBas) <- taxa$Genus
   # create a single cell experiment object
  allSamples_se <- SingleCellExperiment(assays = list(counts = t(AllTaxatableBas)), ### transposed the taxatable (data) #The assays slot contains the experimental data as count matrices. Multiple matrices can be stored the result of assays is actually a list of matrices.
                             colData = allSamplesMetadata, ### contains the data on samples
                             rowData = taxa) # rowData contains data on the features of the analyzed samples. Of particular interest for the microbiome field this is used to store taxonomic information. of note, the taxa dataframe here includes all 6 levels of the taxonomy
  
  
  ### Transformations
  allSamples_se <- transformSamples(allSamples_se, method = "relabundance")
  allSamples_se <- transformSamples(x = allSamples_se, abund_values = "counts", method = "clr", 
                         pseudocount = 1, name = "clr")
 #     head(assay(se, "clr"))
  
  table(allSamplesMetadata$sampletype)
    ##########################################################################################
    # Alpha Diversity - all
    #########################################################################################
    allSamples_se <- mia::estimateDiversity(allSamples_se, 
                                 abund_values = "relabundance", # can either be counts or relabundance
                                 index = "shannon", 
                                 name = "shannon")
    allSamples_df <- as.data.frame(colData(allSamples_se)[colData(allSamples_se)$sampletype %in% 
                                      c("lhmp_bal", "lhmp_oralwash", "NegativeControl", "PositiveControl", "oral_swab", "tracheal"), ])
    allSamples_df$sampletype <- factor(allSamples_df$sampletype)
    allSamples_df$sampletype <- ordered(allSamples_df$sampletype, levels = c("lhmp_bal", "lhmp_oralwash", "NegativeControl", "PositiveControl", "oral_swab", "tracheal"))
    # this is a complicated way of finding the comparison groups. 
  #   comp <- split(t(combn(levels(df$AspPnaGroup), 2)), 
     #               seq(nrow(t(combn(levels(df$AspPnaGroup), 2)))))
    
    
    #### obtain colorblind friendly palette
# The palette with grey:
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# The palette with black:
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
    
# Pairwise comparisons.
mycomparisons <- list(c("lhmp_bal", "tracheal"), c("lhmp_oralwash", "oral_swab"), c("NegativeControl", "oral_swab"), c("NegativeControl", "tracheal"))
# mycomparisons <- list(c("Healthy_BAL", "ETA_ICU"), c("Healthy_OralWash", "OPS_ICU"), c("NegativeControl", "OPS_ICU"), c("NegativeControl", "ETA_ICU"))

shannon_allsamples <- ggboxplot(allSamples_df, x = "sampletype", y = "shannon", 
                   color = "sampletype",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Shannon Index", legend = "none" ) + 
   scale_x_discrete(labels=c("lhmp_bal" = "Healthy_BAL", "lhmp_oralwash" = "Healthy_OralWash", "NegativeControl"  = "NegativeControl", "PositiveControl" = "PositiveControl", "oral_swab" = "OPS_ICU", "tracheal" = "ETA_ICU")) +
   stat_compare_means(comparisons = mycomparisons) 
ggsave("./results/shannon_allsamples.jpeg",  width = 10, height = 5)


  #######################################################################################################
    # Beta Diversity all
    ######################################################################################################
   allsamples_rel_abund_assay <- assays(allSamples_se)$relabundance
    
    # Calculates distances between samples. Because taxa is in
    # columns, it is used to compare different samples. We transpose the
    # assay to get taxa to columns
    allsamples_manhattan_dist <- vegan::vegdist(t(allsamples_rel_abund_assay), method = "manhattan")
#   allsamples_manhattan_mds <- vegan::metaMDS(allsamples_manhattan_dist)
    allsamples_manhattan_pcoa <- ecodist::pco(allsamples_manhattan_dist)
    # Creates a data frame from principal coordinates
    allsamples_manhattan_pcoa_df <- data.frame(pcoa1 = allsamples_manhattan_pcoa$vectors[,1], 
                                     pcoa2 = allsamples_manhattan_pcoa$vectors[,2])
    
    ## merge with metadata to make this analyzable by ggpot as per Schloss youtube video approach
    rownames(allsamples_manhattan_pcoa_df) <-  rownames(allSamplesMetadata)
    allsamples_manhattan_pcoa_df$sample_id <-  rownames(allsamples_manhattan_pcoa_df) 
    allsamples_MetadataBas_nmds <- merge(allSamplesMetadata, allsamples_manhattan_pcoa_df, by = "sample_id", all = TRUE) 
    
    
    table(allsamples_MetadataBas_nmds$sampletype)
#### obtain centroids
centroid_allsamples <- allsamples_MetadataBas_nmds %>% dplyr::group_by(sampletype) %>% summarize(pcoa1mean = mean(pcoa1), pcoa2mean = mean(pcoa2))
    
allsamples_MetadataBas_nmds %>% filter(sampletype=="lhmp_bal") %>% dplyr::group_by(sampletype) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="lhmp_oralwash") %>% dplyr::group_by(sampletype) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="NegativeControl") %>% dplyr::group_by(sampletype) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="PositiveControl") %>% dplyr::group_by(sampletype) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="oral_swab") %>% dplyr::group_by(sampletype) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="tracheal") %>% dplyr::group_by(sampletype) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))


allsamples_MetadataBas_nmds <- allsamples_MetadataBas_nmds %>% mutate(pcoa1mean = case_when(sampletype=="lhmp_bal" ~ -0.002216707, 
                             sampletype=="lhmp_oralwash" ~  -0.004005273,
                             sampletype=="NegativeControl" ~ 0.002624881,
                             sampletype=="PositiveControl" ~ 0.006869789, 
                             sampletype=="oral_swab" ~ -0.001467188, 
                             sampletype=="tracheal" ~ -0.0008092699
                                ))

allsamples_MetadataBas_nmds %>% filter(sampletype=="lhmp_bal") %>% dplyr::group_by(sampletype) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="lhmp_oralwash") %>% dplyr::group_by(sampletype) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="NegativeControl") %>% dplyr::group_by(sampletype) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="PositiveControl") %>% dplyr::group_by(sampletype) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="oral_swab") %>% dplyr::group_by(sampletype) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
allsamples_MetadataBas_nmds %>% filter(sampletype=="tracheal") %>% dplyr::group_by(sampletype) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))


allsamples_MetadataBas_nmds <- allsamples_MetadataBas_nmds %>% mutate(pcoa2mean = case_when(sampletype=="lhmp_bal" ~ 0.00357661, 
                             sampletype=="lhmp_oralwash" ~ 0.004245876 ,
                             sampletype=="NegativeControl" ~ -0.0004988231,
                             sampletype=="PositiveControl" ~ 0.0002673492, 
                             sampletype=="oral_swab" ~  0.0005685255, 
                             sampletype=="tracheal" ~ -0.001317013
                                ))


allsamples_pcoa <- 
ggplot(allsamples_MetadataBas_nmds, aes(x=pcoa1, xend = pcoa1mean,y = pcoa2, yend = pcoa2mean, color = sampletype, fill = sampletype)) + 
 stat_ellipse(level = 0.8,  show.legend = FALSE, size = 1.2) +
  geom_point(size=1.5, show.legend = FALSE) +
geom_point(data = allsamples_MetadataBas_nmds, mapping = aes(x=pcoa1mean, y = pcoa2mean), size = 5, show.legend = FALSE) +
geom_segment(size = 0.1, show.legend = FALSE) +
labs(x= "PCoA axis1", y = "PCoA axis 2") + scale_color_manual(name = NULL, breaks = c("lhmp_bal", "lhmp_oralwash", "NegativeControl", "PositiveControl", "oral_swab", "tracheal"), values = c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2")) + theme_classic() +ggtitle("PCoA for All Samples") + annotate(geom="text", x=0.004, y=-0.012, label="R2=0.28, p<0.0001",color="red")
#+ theme(legend.key.height = unit(0.5, "cm"), legend.position = c(0.8,0.2), legend.background = element_rect(fill = "NA", color = "gray"))
ggsave("./results/allsamples_pcoa.jpeg", width = 10, height = 5)
    

    
 #######################################################
    ##### Permanova for oral
    #######################################################
    
    permanova_allsamplesSampletype <- vegan::adonis2(t(assay(allSamples_se,"relabundance")) ~ sampletype,
                                           data = colData(allSamples_se),
                                           permutations = 9999)
# permanova_allsamplesSampletype # R2=28.131 # 1e-04

###### include a figure of total number of reads by samples. 



#### export supplemental figure for all samples
##### ecological figure all samples

mycomparisons <- list(c("lhmp_bal", "tracheal"), c("lhmp_oralwash", "oral_swab"), c("NegativeControl", "oral_swab"), c("NegativeControl", "tracheal"))
# mycomparisons <- list(c("Healthy_BAL", "ETA_ICU"), c("Healthy_OralWash", "OPS_ICU"), c("NegativeControl", "OPS_ICU"), c("NegativeControl", "ETA_ICU"))

nofreads_allsamples <- ggboxplot(allSamples_df, x = "sampletype", y = "total", 
                   color = "sampletype",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Total N of 16S reads", legend = "none" ) + 
   scale_x_discrete(labels=c("lhmp_bal" = "Healthy_BAL", "lhmp_oralwash" = "Healthy_OralWash", "NegativeControl"  = "NegativeControl", "PositiveControl" = "PositiveControl", "oral_swab" = "OPS_ICU", "tracheal" = "ETA_ICU")) + geom_hline(yintercept = 300, linetype="dashed") +
   stat_compare_means(comparisons = mycomparisons) 
ggsave("./results/nofreads_allsamples.jpeg",  width = 10, height = 5)



ecological_AllSamples <- ggarrange(nofreads_allsamples, shannon_allsamples, allsamples_pcoa, ncol = 1, nrow = 3)
ggsave("./results/ecological_AllSamples.jpeg", width = 10, height = 15)




```




```{r microbiome analyses}



####################
# Read meta data
# data <-  read.table(file = "./data/aspirationPneumonia.csv", sep = ",")
# meta.data <- read.csv(file = "./data/aspirationPneumonia.meta.data.csv")

######################## CREATE SEPARATE DATASETS FOR ORAL AND TRACHEAL INCLUDING THE NORMAL CONTROLS. 
# change the oral wash to oral swab
table(AspMetadata$sampletype)
AspMetadata$sampletype[AspMetadata$sampletype=="lhmp_oralwash"] <- "oral_swab"
AspMetadata$sampletype[AspMetadata$sampletype=="lhmp_bal"] <- "tracheal"

AspMetadata$AspPnaGroup

# ensure that metadata and taxa table match for the oral dataset
oralMetadataBas <- AspMetadata[which(AspMetadata$studydayfollowup=="baseline" & AspMetadata$sampletype=="oral_swab" ),] # 169 observations.

# set rownames by sample_id
rownames(oralMetadataBas) <- oralMetadataBas$sample_id
alirmatchingids_v <- as.character(oralMetadataBas$sample_id)
oralTaxatableBas <- subset(taxatableAsp, rownames(taxatableAsp) %in% alirmatchingids_v)
#### arrange by rownames. 
oralMetadataBas <-oralMetadataBas[sort(rownames(oralMetadataBas)), ]
oralTaxatableBas <-oralTaxatableBas[sort(rownames(oralTaxatableBas)), ]
# ensure all rows equal
all.equal(rownames(oralMetadataBas), rownames(oralTaxatableBas))


  ##########################################################
  # quality control: checking counts per sample
  #########################################################
reads.per.sample <- oralMetadataBas$total
hist(reads.per.sample, breaks = 100)
samples_with_few_reads = oralMetadataBas$sample_id[reads.per.sample<300]
### remove samples with <300 reads from analyses for 16s
oralMetadataBas <- subset(oralMetadataBas, !rownames(oralMetadataBas) %in% samples_with_few_reads) # 156 samples
oralTaxatableBas <- subset(oralTaxatableBas, !rownames(oralTaxatableBas) %in% samples_with_few_reads) # 156 samples
# remain matched
all.equal(rownames(oralMetadataBas), rownames(oralTaxatableBas))

  # remove total
  oralTaxatableBas = oralTaxatableBas[,-1]
  
  ####################################################################################
  # Creating a SingleCellExperiment object for the downstream analysis
  ################################################################################
  # separate column names by taxa
  oral_df <- data.frame(x = names(oralTaxatableBas))
  taxa <- oral_df %>% separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                               "Family", "Genus"), sep = "\\.")
  # use Genus as column names
  names(oralTaxatableBas) <- taxa$Genus
   # create a single cell experiment object
  oral_se <- SingleCellExperiment(assays = list(counts = t(oralTaxatableBas)), ### transposed the taxatable (data) #The assays slot contains the experimental data as count matrices. Multiple matrices can be stored the result of assays is actually a list of matrices.
                             colData = oralMetadataBas, ### contains the data on samples
                             rowData = taxa) # rowData contains data on the features of the analyzed samples. Of particular interest for the microbiome field this is used to store taxonomic information. of note, the taxa dataframe here includes all 6 levels of the taxonomy
  
  
  ### Transformations
  oral_se <- transformSamples(oral_se, method = "relabundance")
  oral_se <- transformSamples(x = oral_se, abund_values = "counts", method = "clr", 
                         pseudocount = 1, name = "clr")
 #     head(assay(se, "clr"))
  
 

  # get taxa that exceed the given prevalence and detection thresholds
  # detection: relative abundance threshold
  # prevalence: relative population frequencies; at 1% compositional abundance threshold
  oral_se <- subsetByPrevalentTaxa(oral_se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                              prevalence = 5/100)
 
    ##########################################################################################
    # Alpha Diversity - oral
    #########################################################################################
    oral_se <- mia::estimateDiversity(oral_se, 
                                 abund_values = "relabundance", # can either be counts or relabundance
                                 index = "shannon", 
                                 name = "shannon")
    oral_df <- as.data.frame(colData(oral_se)[colData(oral_se)$AspPnaGroup %in% 
                                      c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"), ])
    oral_df$AspPnaGroup <- factor(oral_df$AspPnaGroup)
    oral_df$AspPnaGroup <- ordered(oral_df$AspPnaGroup, levels = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"))
    # this is a complicated way of finding the comparison groups. 
  #   comp <- split(t(combn(levels(df$AspPnaGroup), 2)), 
     #               seq(nrow(t(combn(levels(df$AspPnaGroup), 2)))))
    
    
    #### obtain colorblind friendly palette
# The palette with grey:
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# The palette with black:
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
    
# Pairwise comparisons.
mycomparisons <- list(c("MAsP", "NonMAsP"), c("MAsP", "IntubControl"), c("MAsP", "HealthyControls"))
shannon_oral <- ggboxplot(oral_df, x = "AspPnaGroup", y = "shannon", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 
ggsave("./results/shannon_oral.jpeg",  width = 6, height = 5)

    

    ##########################################################################################
    # Alpha Diversity - tracheal
    #########################################################################################

# ensure that metadata and taxa table match for the tracheal dataset
trachealMetadataBas <- AspMetadata[which(AspMetadata$studydayfollowup=="baseline" & AspMetadata$sampletype=="tracheal" ),] # 209 observations.

# set rownames by sample_id
rownames(trachealMetadataBas) <- trachealMetadataBas$sample_id
alirmatchingids_v <- as.character(trachealMetadataBas$sample_id)
trachealTaxatableBas <- subset(taxatableAsp, rownames(taxatableAsp) %in% alirmatchingids_v)
#### arrange by rownames. 
trachealMetadataBas <-trachealMetadataBas[sort(rownames(trachealMetadataBas)), ]
trachealTaxatableBas <-trachealTaxatableBas[sort(rownames(trachealTaxatableBas)), ]
# ensure all rows equal
all.equal(rownames(trachealMetadataBas), rownames(trachealTaxatableBas))


  ##########################################################
  # quality control: checking counts per sample
  #########################################################
reads.per.sample <- trachealMetadataBas$total
hist(reads.per.sample, breaks = 100)
samples_with_few_reads = trachealMetadataBas$sample_id[reads.per.sample<300]
### remove samples with <300 reads from analyses for 16s
trachealMetadataBas <- subset(trachealMetadataBas, !rownames(trachealMetadataBas) %in% samples_with_few_reads) # 187 samples
trachealTaxatableBas <- subset(trachealTaxatableBas, !rownames(trachealTaxatableBas) %in% samples_with_few_reads) # 187 samples
# remain matched
all.equal(rownames(trachealMetadataBas), rownames(trachealTaxatableBas))

  # remove total
  trachealTaxatableBas = trachealTaxatableBas[,-1]
  
  ####################################################################################
  # Creating a SingleCellExperiment object for the downstream analysis
  ################################################################################
  # separate column names by taxa
  tracheal_df <- data.frame(x = names(trachealTaxatableBas))
  taxa <- tracheal_df %>% separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                               "Family", "Genus"), sep = "\\.")
  # use Genus as column names
  names(trachealTaxatableBas) <- taxa$Genus
   # create a single cell experiment object
  tracheal_se <- SingleCellExperiment(assays = list(counts = t(trachealTaxatableBas)), ### transposed the taxatable (data) #The assays slot contains the experimental data as count matrices. Multiple matrices can be stored the result of assays is actually a list of matrices.
                             colData = trachealMetadataBas, ### contains the data on samples
                             rowData = taxa) # rowData contains data on the features of the analyzed samples. Of particular interest for the microbiome field this is used to store taxonomic information. of note, the taxa dataframe here includes all 6 levels of the taxonomy
  
  
  ### Transformations
  tracheal_se <- transformSamples(tracheal_se, method = "relabundance")
  tracheal_se <- transformSamples(x = tracheal_se, abund_values = "counts", method = "clr", 
                         pseudocount = 1, name = "clr")
 #     head(assay(se, "clr"))
  
 

  # get taxa that exceed the given prevalence and detection thresholds
  # detection: relative abundance threshold
  # prevalence: relative population frequencies; at 1% compositional abundance threshold
  tracheal_se <- subsetByPrevalentTaxa(tracheal_se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                              prevalence = 5/100)
 
    tracheal_se <- mia::estimateDiversity(tracheal_se, 
                                 abund_values = "relabundance", # can either be counts or relabundance
                                 index = "shannon", 
                                 name = "shannon")
    tracheal_df <- as.data.frame(colData(tracheal_se)[colData(tracheal_se)$AspPnaGroup %in% 
                                      c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"), ])
    tracheal_df$AspPnaGroup <- factor(tracheal_df$AspPnaGroup)
    tracheal_df$AspPnaGroup <- ordered(tracheal_df$AspPnaGroup, levels = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"))

    # Pairwise comparisons.
 mycomparisons <- list(c("MAsP", "NonMAsP"), c("MAsP", "IntubControl"), c("MAsP", "HealthyControls"))
    shannon_tracheal <- ggboxplot(tracheal_df, x = "AspPnaGroup", y = "shannon", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) 
ggsave("./results/shannon_tracheal.jpeg",  width = 6, height = 5)
    

    #######################################################################################################
    # qpcr graphs
    ######################################################################################################
### oral
    mycomparisons <- list(c("MAsP", "NonMAsP"), c("MAsP", "IntubControl"))
    qpcr_oral <- oral_df %>% filter(AspPnaGroup !="HealthyControls") %>% ggboxplot(x = "AspPnaGroup", y = "qPCR_16S_mean", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_y_log10() +
     stat_compare_means(comparisons = mycomparisons) 
ggsave("./results/qpcr_oral.jpeg",  width = 6, height = 5)

# tracheal
    qpcr_tracheal <- tracheal_df %>% filter(AspPnaGroup !="HealthyControls") %>% ggboxplot(x = "AspPnaGroup", y = "qPCR_16S_mean", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_y_log10() +
     stat_compare_means(comparisons = mycomparisons) 
ggsave("./results/qpcr_tracheal.jpeg",  width = 6, height = 5)


    #######################################################################################################
    # Beta Diversity oral
    ######################################################################################################
    oral_rel_abund_assay <- assays(oral_se)$relabundance
    
    # Calculates distances between samples. Because taxa is in
    # columns, it is used to compare different samples. We transpose the
    # assay to get taxa to columns
    oral_manhattan_dist <- vegan::vegdist(t(oral_rel_abund_assay), method = "manhattan")
    # oral_manhattan_mds <- vegan::metaMDS(oral_manhattan_dist)
    oral_manhattan_pcoa <- ecodist::pco(oral_manhattan_dist)
    # Creates a data frame from principal coordinates
    oral_manhattan_pcoa_df <- data.frame(pcoa1 = oral_manhattan_pcoa$vectors[,1], 
                                     pcoa2 = oral_manhattan_pcoa$vectors[,2])
    
    ## merge with metadata to make this analyzable by ggpot as per Schloss youtube video approach
    rownames(oral_manhattan_pcoa_df) <-  rownames(oralMetadataBas)
    oral_manhattan_pcoa_df$sample_id <-  rownames(oral_manhattan_pcoa_df) 
    oralMetadataBas_nmds <- merge(oralMetadataBas, oral_manhattan_pcoa_df, by = "sample_id", all = TRUE) 
    
#### obtain centroids
centroid_oral <- oralMetadataBas_nmds %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa1mean = mean(pcoa1), pcoa2mean = mean(pcoa2))
    
oralMetadataBas_nmds %>% filter(AspPnaGroup=="MAsP") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
oralMetadataBas_nmds %>% filter(AspPnaGroup=="NonMAsP") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
oralMetadataBas_nmds %>% filter(AspPnaGroup=="IntubControl") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
oralMetadataBas_nmds %>% filter(AspPnaGroup=="HealthyControls") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
oralMetadataBas_nmds <- oralMetadataBas_nmds %>% mutate(pcoa1mean = case_when(AspPnaGroup=="MAsP" ~ 0.0006656204, 
                             AspPnaGroup=="NonMAsP" ~  0.004272598,
                             AspPnaGroup=="IntubControl" ~ -0.009212845,
                             AspPnaGroup=="HealthyControls" ~ -0.01174542
                                ))
oralMetadataBas_nmds %>% filter(AspPnaGroup=="MAsP") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
oralMetadataBas_nmds %>% filter(AspPnaGroup=="NonMAsP") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
oralMetadataBas_nmds %>% filter(AspPnaGroup=="IntubControl") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
oralMetadataBas_nmds %>% filter(AspPnaGroup=="HealthyControls") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
oralMetadataBas_nmds <- oralMetadataBas_nmds %>% mutate(pcoa2mean = case_when(AspPnaGroup=="MAsP" ~ 0.001954934, 
                             AspPnaGroup=="NonMAsP" ~  -0.003024352,
                             AspPnaGroup=="IntubControl" ~ -0.003619398,
                             AspPnaGroup=="HealthyControls" ~ 0.007113444
                                ))

oral_pcoa <- 
ggplot(oralMetadataBas_nmds, aes(x=pcoa1, xend = pcoa1mean,y = pcoa2, yend = pcoa2mean, color = AspPnaGroup, fill = AspPnaGroup)) + 
 stat_ellipse(level = 0.8,  show.legend = FALSE, size = 1.2) +
  geom_point(size=1.5, show.legend = FALSE) +
geom_point(data = oralMetadataBas_nmds, mapping = aes(x=pcoa1mean, y = pcoa2mean), size = 5, show.legend = FALSE) +
geom_segment(size = 0.1, show.legend = FALSE) +
labs(x= "PCoA axis1", y = "PCoA axis 2") + scale_color_manual(name = NULL, breaks = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"), values = c("#000000", "#E69F00", "#56B4E9", "#009E73")) + theme_classic() +ggtitle("PCoA for OPS samples") + annotate(geom="text", x=0.03, y=-0.03, label="R2=0.064, p<0.0001",color="red")
#+ theme(legend.key.height = unit(0.5, "cm"), legend.position = c(0.8,0.2), legend.background = element_rect(fill = "NA", color = "gray"))
ggsave("./results/oral.pcoa.jpeg", width = 6, height = 5)
    
    
 #######################################################
    ##### Permanova for oral
    #######################################################
    
    permanova_AspPnaGroup <- vegan::adonis2(t(assay(oral_se,"relabundance")) ~ AspPnaGroup,
                                           data = colData(oral_se),
                                           permutations = 9999)
# AspPnaGroup # R2=0.06393125 # 1e-04

#############

    #######################################################################################################
    # Beta Diversity TRACHEAL
    ######################################################################################################
    tracheal_rel_abund_assay <- assays(tracheal_se)$relabundance
    
    # Calculates distances between samples. Because taxa is in
    # columns, it is used to compare different samples. We transpose the
    # assay to get taxa to columns
    tracheal_manhattan_dist <- vegan::vegdist(t(tracheal_rel_abund_assay), method = "manhattan")
    #oral_manhattan_mds <- vegan::metaMDS(oral_manhattan_dist)
   tracheal_manhattan_pcoa <- ecodist::pco(tracheal_manhattan_dist)
    # Creates a data frame from principal coordinates
    tracheal_manhattan_pcoa_df <- data.frame(pcoa1 = tracheal_manhattan_pcoa$vectors[,1], 
                                     pcoa2 = tracheal_manhattan_pcoa$vectors[,2])
    
    ## merge with metadata to make this analyzable by ggpot as per Schloss youtube video approach
    rownames(tracheal_manhattan_pcoa_df) <-  rownames(trachealMetadataBas)
    tracheal_manhattan_pcoa_df$sample_id <-  rownames(tracheal_manhattan_pcoa_df) 
    trachealMetadataBas_nmds <- merge(trachealMetadataBas, tracheal_manhattan_pcoa_df, by = "sample_id", all = TRUE) 
    
#### obtain centroids
trachealMetadataBas_nmds %>% filter(AspPnaGroup=="MAsP") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
trachealMetadataBas_nmds %>% filter(AspPnaGroup=="NonMAsP") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
trachealMetadataBas_nmds %>% filter(AspPnaGroup=="IntubControl") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
trachealMetadataBas_nmds %>% filter(AspPnaGroup=="HealthyControls") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
trachealMetadataBas_nmds <- trachealMetadataBas_nmds %>% mutate(pcoa1mean = case_when(AspPnaGroup=="MAsP" ~ -0.0001225873, 
                             AspPnaGroup=="NonMAsP" ~  -0.0002745729,
                             AspPnaGroup=="IntubControl" ~ -0.008907177,
                             AspPnaGroup=="HealthyControls" ~ 0.006514197
                                ))
trachealMetadataBas_nmds %>% filter(AspPnaGroup=="MAsP") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
trachealMetadataBas_nmds %>% filter(AspPnaGroup=="NonMAsP") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
trachealMetadataBas_nmds %>% filter(AspPnaGroup=="IntubControl") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
trachealMetadataBas_nmds %>% filter(AspPnaGroup=="HealthyControls") %>% dplyr::group_by(AspPnaGroup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
trachealMetadataBas_nmds <- trachealMetadataBas_nmds %>% mutate(pcoa2mean = case_when(AspPnaGroup=="MAsP" ~ -0.001181902, 
                             AspPnaGroup=="NonMAsP" ~  -0.001458901,
                             AspPnaGroup=="IntubControl" ~ 0.001942333,
                             AspPnaGroup=="HealthyControls" ~ 0.007220277
                                ))

tracheal_pcoa <- 
ggplot(trachealMetadataBas_nmds, aes(x=pcoa1, xend = pcoa1mean,y = pcoa2, yend = pcoa2mean, color = AspPnaGroup, fill = AspPnaGroup)) + 
 stat_ellipse(level = 0.8,  show.legend = FALSE, size = 1.2) +
  geom_point(size=1.5, show.legend = FALSE) +
geom_point(data = trachealMetadataBas_nmds, mapping = aes(x=pcoa1mean, y = pcoa2mean), size = 5, show.legend = FALSE) +
geom_segment(size = 0.1, show.legend = FALSE) +
labs(x= "PCoA axis1", y = "PCoA axis 2") + scale_color_manual(name = NULL, breaks = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"), values = c("#000000", "#E69F00", "#56B4E9", "#009E73")) + theme_classic() +ggtitle("PCoA for ETA samples") + annotate(geom="text", x=-0.02, y=-0.03, label="R2=0.052, p<0.0001",color="red") #+ theme(legend.key.height = unit(0.5, "cm"), legend.position = c(0.8,0.2), legend.background = element_rect(fill = "NA", color = "gray")) 
ggsave("./results/tracheal.pcoa.jpeg", width = 6, height = 5)
    
 #######################################################
    ##### Permanova for tracheal
    #######################################################
    tracheal_permanova_AspPnaGroup <- vegan::adonis2(t(assay(tracheal_se,"relabundance")) ~ AspPnaGroup, data = colData(tracheal_se),permutations = 9999)
                                                     
                                      
 #  Df SumOfSqs      R2      F Pr(>F)    
# AspPnaGroup   3    3.176 0.05171 3.3264  1e-04 ***


####### arrange graph with basic ecological features across all samples. 

aspirationEcologicalFigure <- ggarrange(qpcr_oral, shannon_oral, oral_pcoa, qpcr_tracheal, shannon_tracheal, tracheal_pcoa, ncol = 3, nrow = 2, widths = c(1, 1, 2))
ggsave("./results/aspirationEcologicalFigure.jpeg", width = 20, height = 10)
 
###################################################################################3
################## perform alpha and beta comparisons for risk factors     for both oral and tracheal. 
#######################################
########

oral_df$Antiacid_chronic
oral_df$Home02
oral_df$AMS_onIntub
oral_df$Seizure_beforeIntub
oral_df$DrugOD
oral_df$SuspectedAspirationbyClinician
oral_df$ETOH_Intox_or_Withdrawal
oral_df$RadiologistReportAspiration
    
####### ORAL SHANNON
  oralshannon_antacid <- oral_df %>% filter(!is.na(Antiacid_chronic)) %>% ggboxplot(x = "Antiacid_chronic", y = "shannon", 
                   color = "Antiacid_chronic",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="Chronic Anti-Acid Use",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
  
    oralshannon_home02 <- oral_df %>% filter(!is.na(Home02)) %>% ggboxplot(x = "Home02", y = "shannon", 
                   color = "Home02",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="Chronic O2  Use",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
    
      oralshannon_AMS_onIntub <- oral_df %>% filter(!is.na(AMS_onIntub)) %>% ggboxplot(x = "AMS_onIntub", y = "shannon", 
                   color = "AMS_onIntub",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="Altered Mental Status on Intubation",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
      
        oralshannon_Seizure_beforeIntub <- oral_df %>% filter(!is.na(Seizure_beforeIntub)) %>% ggboxplot(x = "Seizure_beforeIntub", y = "shannon", 
                   color = "Seizure_beforeIntub",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="Seizure Prior to Intubation",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
        
          oralshannon_DrugOD <- oral_df %>% filter(!is.na(DrugOD)) %>% ggboxplot(x = "DrugOD", y = "shannon", 
                   color = "DrugOD",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="Drug Overdose",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
          
            oralshannon_ETOH_Intox_or_Withdrawal <- oral_df %>% filter(!is.na(ETOH_Intox_or_Withdrawal)) %>% ggboxplot(x = "ETOH_Intox_or_Withdrawal", y = "shannon", 
                   color = "ETOH_Intox_or_Withdrawal",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="ETOH intoxication or withdrawal",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
            
              oralshannon_SuspectedAspirationbyClinician<- oral_df %>% filter(!is.na(SuspectedAspirationbyClinician)) %>% ggboxplot(x = "SuspectedAspirationbyClinician", y = "shannon", 
                   color = "SuspectedAspirationbyClinician",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="Suspected Aspiration by Clinician",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
              
                oralshannon_RadiologistReportAspiration<- oral_df %>% filter(!is.na(RadiologistReportAspiration)) %>% ggboxplot(x = "RadiologistReportAspiration", y = "shannon", 
                   color = "RadiologistReportAspiration",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="Aspiration Reported by Radiologist",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means()       

### summary shannon figure
    OralShannonRiskFactors <- ggarrange(oralshannon_antacid, oralshannon_home02, oralshannon_AMS_onIntub, oralshannon_Seizure_beforeIntub, oralshannon_DrugOD, oralshannon_ETOH_Intox_or_Withdrawal, oralshannon_SuspectedAspirationbyClinician, oralshannon_RadiologistReportAspiration, ncol = 4, nrow = 2)
    ggsave("./results/OralShannonRiskFactors.jpeg", width = 20, height = 12)

####### TRACHEAL SHANNON   
trachealshannon_antacid <- tracheal_df %>% filter(!is.na(Antiacid_chronic)) %>% ggboxplot(x = "Antiacid_chronic", y = "shannon", 
                   color = "Antiacid_chronic",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="Chronic Anti-Acid Use",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
  
    trachealshannon_home02 <- tracheal_df %>% filter(!is.na(Home02)) %>% ggboxplot(x = "Home02", y = "shannon", 
                   color = "Home02",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="Chronic O2  Use",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
    
      trachealshannon_AMS_onIntub <- tracheal_df %>% filter(!is.na(AMS_onIntub)) %>% ggboxplot(x = "AMS_onIntub", y = "shannon", 
                   color = "AMS_onIntub",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="Altered Mental Status on Intubation",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
      
        trachealshannon_Seizure_beforeIntub <- tracheal_df %>% filter(!is.na(Seizure_beforeIntub)) %>% ggboxplot(x = "Seizure_beforeIntub", y = "shannon", 
                   color = "Seizure_beforeIntub",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="Seizure Prior to Intubation",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
        
          trachealshannon_DrugOD <- tracheal_df %>% filter(!is.na(DrugOD)) %>% ggboxplot(x = "DrugOD", y = "shannon", 
                   color = "DrugOD",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="Drug Overdose",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
          
            trachealshannon_ETOH_Intox_or_Withdrawal <- tracheal_df %>% filter(!is.na(ETOH_Intox_or_Withdrawal)) %>% ggboxplot(x = "ETOH_Intox_or_Withdrawal", y = "shannon", 
                   color = "ETOH_Intox_or_Withdrawal",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="ETOH intoxication or withdrawal",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
            
              trachealshannon_SuspectedAspirationbyClinician<- tracheal_df %>% filter(!is.na(SuspectedAspirationbyClinician)) %>% ggboxplot(x = "SuspectedAspirationbyClinician", y = "shannon", 
                   color = "SuspectedAspirationbyClinician",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="Suspected Aspiration by Clinician",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() 
              
                trachealshannon_RadiologistReportAspiration<- tracheal_df %>% filter(!is.na(RadiologistReportAspiration)) %>% ggboxplot(x = "RadiologistReportAspiration", y = "shannon", 
                   color = "RadiologistReportAspiration",add = "jitter" , palette = "Dark2", xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="Aspiration Reported by Radiologist",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means()       

### summary shannon figure
    trachealShannonRiskFactors <- ggarrange(trachealshannon_antacid, trachealshannon_home02, trachealshannon_AMS_onIntub, trachealshannon_Seizure_beforeIntub, trachealshannon_DrugOD, trachealshannon_ETOH_Intox_or_Withdrawal, trachealshannon_SuspectedAspirationbyClinician, trachealshannon_RadiologistReportAspiration, ncol = 4, nrow = 2)
    ggsave("./results/trachealShannonRiskFactors.jpeg", width = 20, height = 12)    
    
    
######### apcr for oral by risk factors    
  oralqpcr_antacid <- oral_df %>% filter(!is.na(Antiacid_chronic)) %>% ggboxplot(x = "Antiacid_chronic", y = "qPCR_16S_mean", 
                   color = "Antiacid_chronic",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="Chronic Anti-Acid Use",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
  
    oralqpcr_home02 <- oral_df %>% filter(!is.na(Home02)) %>% ggboxplot(x = "Home02", y = "qPCR_16S_mean", 
                   color = "Home02",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="Chronic O2  Use",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
    
      oralqpcr_AMS_onIntub <- oral_df %>% filter(!is.na(AMS_onIntub)) %>% ggboxplot(x = "AMS_onIntub", y = "qPCR_16S_mean", 
                   color = "AMS_onIntub",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="Altered Mental Status on Intubation",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
      
        oralqpcr_Seizure_beforeIntub <- oral_df %>% filter(!is.na(Seizure_beforeIntub)) %>% ggboxplot(x = "Seizure_beforeIntub", y = "qPCR_16S_mean", 
                   color = "Seizure_beforeIntub",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="Seizure Prior to Intubation",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
        
          oralqpcr_DrugOD <- oral_df %>% filter(!is.na(DrugOD)) %>% ggboxplot(x = "DrugOD", y = "qPCR_16S_mean", 
                   color = "DrugOD",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="Drug Overdose",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
          
            oralqpcr_ETOH_Intox_or_Withdrawal <- oral_df %>% filter(!is.na(ETOH_Intox_or_Withdrawal)) %>% ggboxplot(x = "ETOH_Intox_or_Withdrawal", y = "qPCR_16S_mean", 
                   color = "ETOH_Intox_or_Withdrawal",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="ETOH intoxication or withdrawal",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
            
              oralqpcr_SuspectedAspirationbyClinician<- oral_df %>% filter(!is.na(SuspectedAspirationbyClinician)) %>% ggboxplot(x = "SuspectedAspirationbyClinician", y = "qPCR_16S_mean", 
                   color = "SuspectedAspirationbyClinician",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="Suspected Aspiration by Clinician",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
              
                oralqpcr_RadiologistReportAspiration<- oral_df %>% filter(!is.na(RadiologistReportAspiration)) %>% ggboxplot(x = "RadiologistReportAspiration", y = "qPCR_16S_mean", 
                   color = "RadiologistReportAspiration",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="Aspiration Reported by Radiologist",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means()       + scale_y_log10()

### summary qpcr figure
    OralqpcrRiskFactors <- ggarrange(oralqpcr_antacid, oralqpcr_home02, oralqpcr_AMS_onIntub, oralqpcr_Seizure_beforeIntub, oralqpcr_DrugOD, oralqpcr_ETOH_Intox_or_Withdrawal, oralqpcr_SuspectedAspirationbyClinician, oralqpcr_RadiologistReportAspiration, ncol = 4, nrow = 2)
    ggsave("./results/OralqpcrRiskFactors.jpeg", width = 20, height = 12)    
    
######### apcr for tracheal by risk factors   
    
     trachealqpcr_antacid <- tracheal_df %>% filter(!is.na(Antiacid_chronic)) %>% ggboxplot(x = "Antiacid_chronic", y = "qPCR_16S_mean", 
                   color = "Antiacid_chronic",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="Chronic Anti-Acid Use",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
  
    trachealqpcr_home02 <- tracheal_df %>% filter(!is.na(Home02)) %>% ggboxplot(x = "Home02", y = "qPCR_16S_mean", 
                   color = "Home02",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="Chronic O2  Use",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
    
      trachealqpcr_AMS_onIntub <- tracheal_df %>% filter(!is.na(AMS_onIntub)) %>% ggboxplot(x = "AMS_onIntub", y = "qPCR_16S_mean", 
                   color = "AMS_onIntub",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="Altered Mental Status on Intubation",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
      
        trachealqpcr_Seizure_beforeIntub <- tracheal_df %>% filter(!is.na(Seizure_beforeIntub)) %>% ggboxplot(x = "Seizure_beforeIntub", y = "qPCR_16S_mean", 
                   color = "Seizure_beforeIntub",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="Seizure Prior to Intubation",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
        
          trachealqpcr_DrugOD <- tracheal_df %>% filter(!is.na(DrugOD)) %>% ggboxplot(x = "DrugOD", y = "qPCR_16S_mean", 
                   color = "DrugOD",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="Drug Overdose",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
          
            trachealqpcr_ETOH_Intox_or_Withdrawal <- tracheal_df %>% filter(!is.na(ETOH_Intox_or_Withdrawal)) %>% ggboxplot(x = "ETOH_Intox_or_Withdrawal", y = "qPCR_16S_mean", 
                   color = "ETOH_Intox_or_Withdrawal",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="ETOH intoxication or withdrawal",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
            
              trachealqpcr_SuspectedAspirationbyClinician<- tracheal_df %>% filter(!is.na(SuspectedAspirationbyClinician)) %>% ggboxplot(x = "SuspectedAspirationbyClinician", y = "qPCR_16S_mean", 
                   color = "SuspectedAspirationbyClinician",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="Suspected Aspiration by Clinician",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means() + scale_y_log10()
              
                trachealqpcr_RadiologistReportAspiration<- tracheal_df %>% filter(!is.na(RadiologistReportAspiration)) %>% ggboxplot(x = "RadiologistReportAspiration", y = "qPCR_16S_mean", 
                   color = "RadiologistReportAspiration",add = "jitter" , palette = "Dark2", xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="Aspiration Reported by Radiologist",   labels=c("0" = "No", "1" = "Yes")) + stat_compare_means()       + scale_y_log10()

### summary qpcr figure
    trachealqpcrRiskFactors <- ggarrange(trachealqpcr_antacid, trachealqpcr_home02, trachealqpcr_AMS_onIntub, trachealqpcr_Seizure_beforeIntub, trachealqpcr_DrugOD, trachealqpcr_ETOH_Intox_or_Withdrawal, trachealqpcr_SuspectedAspirationbyClinician, trachealqpcr_RadiologistReportAspiration, ncol = 4, nrow = 2)
    ggsave("./results/trachealqpcrRiskFactors.jpeg", width = 20, height = 12)   
    
    

##### Look at summary risk factors. 

    
    my_comparisons <- list (c("None", "ChronicRF"), c("ChronicRF", "AcuteRF"))

    
trachealByRiskFactorShannon <-        tracheal_df %>% filter(!is.na(RiskFactor_categ)) %>% ggboxplot(x = "RiskFactor_categ", y = "shannon", 
                   color = "RiskFactor_categ",add = "jitter" , palette = c("#333333", "#006633","#CC0033" ), xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="Aspiration Risk Factor Category")  + stat_compare_means(comparisons = my_comparisons)   
    # labels=c("0" = "No", "1" = "Yes")) 
  
 oralByRiskFactorShannon <-        oral_df %>% filter(!is.na(RiskFactor_categ)) %>% ggboxplot(x = "RiskFactor_categ", y = "shannon", 
                   color = "RiskFactor_categ",add = "jitter" , palette = c("#333333", "#006633","#CC0033" ), xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="Aspiration Risk Factor Category")  + stat_compare_means(comparisons = my_comparisons)   
 
 trachealByRiskFactorqpcr <-        tracheal_df %>% filter(!is.na(RiskFactor_categ)) %>% ggboxplot(x = "RiskFactor_categ", y = "qPCR_16S_mean", 
                   color = "RiskFactor_categ",add = "jitter" , palette = c("#333333", "#006633","#CC0033" ), xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="Aspiration Risk Factor Category")  + stat_compare_means(comparisons = my_comparisons)   + scale_y_log10()
    # labels=c("0" = "No", "1" = "Yes")) 
  
 oralByRiskFactorqpcr <-        oral_df %>% filter(!is.na(RiskFactor_categ)) %>% ggboxplot(x = "RiskFactor_categ", y = "qPCR_16S_mean", 
                   color = "RiskFactor_categ",add = "jitter" , palette = c("#333333", "#006633","#CC0033" ), xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="Aspiration Risk Factor Category")  + stat_compare_means(comparisons = my_comparisons)   + scale_y_log10()

   RiskFactorsShannonQPCR <- ggarrange(oralByRiskFactorShannon, trachealByRiskFactorShannon, oralByRiskFactorqpcr, trachealByRiskFactorqpcr, ncol = 2, nrow = 2)
    ggsave("./results/RiskFactorsShannonQPCR.jpeg", width = 12, height = 12)   
    
    
  ############## look at culture differences
    oral_df$CxClassification
    
    
  oralbyCulturesqpcr <-   oral_df %>% filter(!is.na(CxClassification)) %>% filter(AspPnaGroup!="IntubControl") %>% ggboxplot(x = "CxClassification", y = "qPCR_16S_mean", 
                   color = "CxClassification",add = "jitter" , palette = c( "#6633FF", "#FF9933" ), xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_x_discrete(name ="Culture Results")  + stat_compare_means()   + scale_y_log10()

    TrachealbyCulturesqpcr <-   tracheal_df %>% filter(!is.na(CxClassification)) %>% filter(AspPnaGroup!="IntubControl") %>% ggboxplot(x = "CxClassification", y = "qPCR_16S_mean", 
                   color = "CxClassification",add = "jitter" , palette = c( "#6633FF", "#FF9933" ), xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_x_discrete(name ="Culture Results")  + stat_compare_means()   + scale_y_log10()
  
   oralbyCulturesshannon <-   oral_df %>% filter(!is.na(CxClassification)) %>% filter(AspPnaGroup!="IntubControl") %>% ggboxplot(x = "CxClassification", y = "shannon", 
                   color = "CxClassification",add = "jitter" , palette = c( "#6633FF", "#FF9933" ), xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + scale_x_discrete(name ="Culture Results")  + stat_compare_means()   

    TrachealbyCulturesshannon <-   tracheal_df %>% filter(!is.na(CxClassification)) %>% filter(AspPnaGroup!="IntubControl") %>% ggboxplot(x = "CxClassification", y = "shannon", 
                   color = "CxClassification",add = "jitter" , palette = c( "#6633FF", "#FF9933" ), xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + scale_x_discrete(name ="Culture Results")  + stat_compare_means()   
    
    CulturesShannonQPCR <- ggarrange(oralbyCulturesshannon, TrachealbyCulturesshannon, oralbyCulturesqpcr, TrachealbyCulturesqpcr, ncol = 2, nrow = 2)
    ggsave("./results/CulturesShannonQPCR.jpeg", width = 12, height = 12)   
  
#################### Relative abundance per group for most common taxa ###################

trachealGroupDensityAbundance <- plotAbundanceDensity(tracheal_se, layout = "density", abund_values = "relabundance", n = 10, colour_by="AspPnaGroup" ) + scale_x_log10() + scale_fill_manual(name = NULL, breaks = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"), values = c("#000000", "#E69F00", "#56B4E9", "#009E73")) + 
 scale_color_manual(name = NULL, breaks = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"), values = c("#000000", "#E69F00", "#56B4E9", "#009E73")) + theme(legend.position = "bottom")

oralGroupDensityAbundance  <- plotAbundanceDensity(oral_se, layout = "density", abund_values = "relabundance",n = 10, colour_by="AspPnaGroup" ) + scale_x_log10()+  scale_fill_manual(name = NULL, breaks = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"), values = c("#000000", "#E69F00", "#56B4E9", "#009E73")) + 
 scale_color_manual(name = NULL, breaks = c("MAsP", "NonMAsP", "IntubControl", "HealthyControls"), values = c("#000000", "#E69F00", "#56B4E9", "#009E73")) + 
  theme(legend.position = "bottom")

GroupDensityAbundance <- ggarrange(oralGroupDensityAbundance, trachealGroupDensityAbundance, ncol = 2, nrow = 1)
ggsave("./results/GroupDensityAbundance.jpeg", width = 16, height = 8)


#####################################################################
# Differential Abundance 
####################################################################

######## ANCOMBC APPROACH FOR ORAL

# currently, ancombc requires the phyloseq format, but we can easily convert:
tse <- TreeSummarizedExperiment(oral_se)

pseq <- makePhyloseqFromTreeSummarizedExperiment(oral_se)


# perform the analysis 
out = ancombc(
  phyloseq = pseq, 
  formula = "AspPnaGroup", 
  p_adj_method = "fdr", 
  zero_cut = 1, # no prev filtering necessary anymore 
  lib_cut = 0, 
  group = "AspPnaGroup", 
  struc_zero = TRUE, 
  neg_lb = TRUE, 
  tol = 1e-5, 
  max_iter = 100, 
  conserve = TRUE, 
  alpha = 0.05, 
  global = TRUE
)
# store the results in res 
res <- out$res

###### take the results from ancombc output in form of co-efficients (beta) and adjusted p-values (q)
beta <-as.data.frame(res$beta)
beta$taxa <-rownames(beta)

# oldnames = c( "AspPnaGroupHealthyControls", "AspPnaGroupMAsP", "AspPnaGroupNonMAsP", "taxa")
oldnames = names(beta)
newnames = c( "beta_AspPnaGroupHealthyControls", "beta_AspPnaGroupMAsP", "beta_AspPnaGroupNonMAsP", "taxa")
beta <- beta %>% rename_at(vars(oldnames), ~ newnames)

q_val <-as.data.frame(res$q_val)
q_val$taxa <-rownames(q_val)

# oldnames = c( "AspPnaGroupHealthyControls", "AspPnaGroupMAsP", "AspPnaGroupNonMAsP", "taxa")
oldnames = names(q_val)
newnames = c( "q_AspPnaGroupHealthyControls", "q_AspPnaGroupMAsP", "q_AspPnaGroupNonMAsP", "taxa")
q_val <- q_val %>% rename_at(vars(oldnames), ~ newnames)

#merge these dataframes
ancombcresults_oral <- merge(beta, q_val, by = "taxa", all = TRUE)

   ancombcresults_oral_MacrovsIntubControl <- subset(ancombcresults_oral, select = c(taxa, beta_AspPnaGroupMAsP,q_AspPnaGroupMAsP ))
 ########  
   
   ancombcresults_oral_MacrovsIntubControl["Significance"] <- "NotSig"
   ancombcresults_oral_MacrovsIntubControl[which(ancombcresults_oral_MacrovsIntubControl['q_AspPnaGroupMAsP'] < 0.05),"Significance"] <- "P.Value < 0.05"   
   
    top_peaks <- ancombcresults_oral_MacrovsIntubControl[with(ancombcresults_oral_MacrovsIntubControl, order(desc(abs(beta_AspPnaGroupMAsP)), q_AspPnaGroupMAsP)),][1:20,]
        top_peaks <- top_peaks %>% filter(Significance!="NotSig")
   
##### MAsP vs. Controls.    
ggplot(ancombcresults_oral_MacrovsIntubControl, aes(x = beta_AspPnaGroupMAsP, y = -log10(q_AspPnaGroupMAsP))) +
        geom_point(aes(color = Significance)) +
        scale_color_manual(values = c("black","blue")) +
        theme_bw(base_size = 16) +
        geom_text_repel(
          data = top_peaks,
          aes(label = taxa),
          size = 5,
          box.padding = unit(0.35, "lines"),
          point.padding = unit(0.3, "lines")
        )
   
     
   ###### CLR FOR ORAL WITH LIMMA
clr = assays(oral_se)$clr
    group_ = colData(oral_se)$AspPnaGroup
    mm <- model.matrix(~0 + group_)
    fit <- lmFit(clr, mm)
    head(fit)
    
    for(comp in 1:4){
      if(comp == 1){
        # Comparison between group_MAsP minus group_IntubControl
        contr <- makeContrasts(group_MAsP - group_IntubControl, levels = colnames(coef(fit)))
        file_name = "oral_diff_abundance_MAsP_IntubControl"
      }else if(comp == 2){
        # Comparison between group_NonMAsP minus group_IntubControl
        contr <- makeContrasts(group_NonMAsP - group_IntubControl, levels = colnames(coef(fit)))
        file_name = "oral_diff_abundance_NonMAsP_IntubControl"
      }else if(comp == 3){
        # Comparison between group_MAsP minus group_NonMAsP
        contr <- makeContrasts(group_MAsP - group_NonMAsP, levels = colnames(coef(fit)))
        file_name = "oral_diff_abundance_MAsP_NonMAsP"
      }  else if(comp == 4){
        # Comparison between group_MAsP minus group_HealthyControls
        contr <- makeContrasts(group_MAsP - group_HealthyControls, levels = colnames(coef(fit)))
        file_name = "oral_diff_abundance_MAsP_HealthyControls"
      }
      
      # Estimate contrast for each gene
      tmp <- contrasts.fit(fit, contr)
      # Empirical Bayes smoothing of standard errors
      tmp <- eBayes(tmp)
            top.table <- topTable(tmp, sort.by = "P", n = Inf)
      write.csv(top.table,file = paste0('./results/',file_name,'.csv'))
      
      # x = tapply(top.table$logFC, row.names(top.table), function(x) max(x))
      # x = sort(x, TRUE)
      # top.table$Genus = factor(as.character(row.names(top.table)), levels=names(x))
      # top.table <- top.table[which(top.table$P.Value<0.05),]
      # P <- ggplot(top.table, aes(x=Genus, y=logFC, color=P.Value)) + geom_point(size=3) + 
      #   theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust=0.5))
      # 
      # save_plot(paste0('./results/',stype,file_name,'.jpg'), P,
      #           base_aspect_ratio = 1.5, base_height =7)
      
   
      diff_taxa = top.table
      # row.names(diff_genes) = diff_genes$X
      diff_df <- diff_taxa[c( "logFC","adj.P.Val")]
      names(diff_df) <- c("logFC","adj.P.Val")
      
      diff_df["Genus"]<-row.names(diff_taxa)
      head(diff_df)
      diff_df["Significance"] <- "NotSig"
      diff_df[which(diff_df['adj.P.Val'] < 0.05),"Significance"] <- "adj.P.Val < 0.05"
      
      sig_diff_df = diff_df[which(diff_df['adj.P.Val'] < 0.05),]
      pos_fc = sig_diff_df[which(sig_diff_df$logFC>0),]
      top_peaks <- pos_fc[with(pos_fc, order(-pos_fc$logFC,pos_fc$adj.P.Val)),][1:20,]
      top_peaks <- top_peaks %>% drop_na()
      
      neg_fc = sig_diff_df[which(sig_diff_df$logFC<0),]
      top_peaks <- rbind(top_peaks, neg_fc[with(neg_fc, order(neg_fc$logFC,neg_fc$adj.P.Val)),][1:10,])
      top_peaks <- top_peaks %>% drop_na()
      

      p <- ggplot(diff_df, aes(x = logFC, y = -log10(adj.P.Val))) +
        geom_point(aes(color = Significance)) +
        scale_color_manual(values = c("blue","black")) +
        theme_bw(base_size = 16) +
        geom_text_repel(
          data = top_peaks,
          aes(label = Genus),
          size = 5,
          box.padding = unit(0.35, "lines"),
          point.padding = unit(0.3, "lines")
        )
      
      save_plot(paste0('./results/','_',file_name,'.jpg'), p,
                base_aspect_ratio = 1.5, base_height =7)
    }

######### differential abundance with limma for tracheal
    
    
     ###### CLR FOR tracheal WITH LIMMA
clr = assays(tracheal_se)$clr
    group_ = colData(tracheal_se)$AspPnaGroup
    mm <- model.matrix(~0 + group_)
    fit <- lmFit(clr, mm)
    head(fit)
    
    for(comp in 1:4){
      if(comp == 1){
        # Comparison between group_MAsP minus group_IntubControl
        contr <- makeContrasts(group_MAsP - group_IntubControl, levels = colnames(coef(fit)))
        file_name = "tracheal_diff_abundance_MAsP_IntubControl"
      }else if(comp == 2){
        # Comparison between group_NonMAsP minus group_IntubControl
        contr <- makeContrasts(group_NonMAsP - group_IntubControl, levels = colnames(coef(fit)))
        file_name = "tracheal_diff_abundance_NonMAsP_IntubControl"
      }else if(comp == 3){
        # Comparison between group_MAsP minus group_NonMAsP
        contr <- makeContrasts(group_MAsP - group_NonMAsP, levels = colnames(coef(fit)))
        file_name = "tracheal_diff_abundance_MAsP_NonMAsP"
      }  else if(comp == 4){
        # Comparison between group_MAsP minus group_HealthyControls
        contr <- makeContrasts(group_MAsP - group_HealthyControls, levels = colnames(coef(fit)))
        file_name = "tracheal_diff_abundance_MAsP_HealthyControls"
      }
      
      # Estimate contrast for each gene
      tmp <- contrasts.fit(fit, contr)
      # Empirical Bayes smoothing of standard errors
      tmp <- eBayes(tmp)
            top.table <- topTable(tmp, sort.by = "P", n = Inf)
      write.csv(top.table,file = paste0('./results/',file_name,'.csv'))
      
      # x = tapply(top.table$logFC, row.names(top.table), function(x) max(x))
      # x = sort(x, TRUE)
      # top.table$Genus = factor(as.character(row.names(top.table)), levels=names(x))
      # top.table <- top.table[which(top.table$P.Value<0.05),]
      # P <- ggplot(top.table, aes(x=Genus, y=logFC, color=P.Value)) + geom_point(size=3) + 
      #   theme(axis.text.x = element_text(angle = -45, hjust = 0, vjust=0.5))
      # 
      # save_plot(paste0('./results/',stype,file_name,'.jpg'), P,
      #           base_aspect_ratio = 1.5, base_height =7)
      
   
      diff_taxa = top.table
      # row.names(diff_genes) = diff_genes$X
      diff_df <- diff_taxa[c( "logFC","adj.P.Val")]
      names(diff_df) <- c("logFC","adj.P.Val")
      
      diff_df["Genus"]<-row.names(diff_taxa)
      head(diff_df)
      diff_df["Significance"] <- "NotSig"
      diff_df[which(diff_df['adj.P.Val'] < 0.05),"Significance"] <- "adj.P.Val < 0.05"
      
      sig_diff_df = diff_df[which(diff_df['adj.P.Val'] < 0.05),]
      pos_fc = sig_diff_df[which(sig_diff_df$logFC>0),]
      top_peaks <- pos_fc[with(pos_fc, order(-pos_fc$logFC,pos_fc$adj.P.Val)),][1:20,]
      top_peaks <- top_peaks %>% drop_na()
      
      neg_fc = sig_diff_df[which(sig_diff_df$logFC<0),]
      top_peaks <- rbind(top_peaks, neg_fc[with(neg_fc, order(neg_fc$logFC,neg_fc$adj.P.Val)),][1:10,])
      top_peaks <- top_peaks %>% drop_na()
      

      p <- ggplot(diff_df, aes(x = logFC, y = -log10(adj.P.Val))) +
        geom_point(aes(color = Significance)) +
        scale_color_manual(values = c("blue","black")) +
        theme_bw(base_size = 16) +
        geom_text_repel(
          data = top_peaks,
          aes(label = Genus),
          size = 5,
          box.padding = unit(0.35, "lines"),
          point.padding = unit(0.3, "lines")
        )
      
      save_plot(paste0('./results/','_',file_name,'.jpg'), p,
                base_aspect_ratio = 1.5, base_height =7)
    }
    
    

    #####################################################################
    # Removing IntubControl Samples for the rest of the analysis
    ####################################################################
    indx = which(oral_se$AspPnaGroup=="MAsP" | oral_se$AspPnaGroup=="NonMAsP")
    oral_pneumonia_se = oral_se[,indx]
    
     table(oral_pneumonia_se$AspPnaGroup)
    
    set.seed(1)
    #######################################################
    ##### Permanova
    #######################################################
    
    permanova_AspPnaGroup <- vegan::adonis2(t(assay(oral_pneumonia_se,"relabundance")) ~ AspPnaGroup,
                                           data = colData(oral_pneumonia_se),
                                           permutations = 9999)
#             Df SumOfSqs      R2     F Pr(>F)
# AspPnaGroup   1    0.379 0.01074 1.324 0.1904
    

    
 #######################################
    # Dirichlet-Multinomial Mixture Model
    ######################################
    
    ###### ORAL
    
    se_dmn <- runDMN(oral_pneumonia_se, name = "DMN", k = 1:7)
    oralPna_Laplace <- plotDMNFit(se_dmn, type = "laplace")
    ggsave("./results/oralPna_Laplace.jpeg")
    
    
    RS.best_fit <- getBestDMNFit(se_dmn, type = "laplace")
    #### 2 class is best fit for oral samples. 
    ### get the assignment probabilities
    prob <- DirichletMultinomial::mixture(getBestDMNFit(se_dmn))
    # Add column names
    colnames(prob) <- c("comp1", "comp2")
    # For each row, finds column that has the highest value. Then extract the column 
# names of highest values and merge with the corresponding metadata dataframe extracted from the summarized experiment. 
vec <- colnames(prob)[max.col(prob,ties.method = "first")]
    vec <-as.data.frame(vec)
  # extract corresponding dataframe
    oral_pneumonia_df <- as.data.frame(colData(oral_pneumonia_se))
    ##### obtain the DMM cluster assignment from vec and merge with dataframe. 
    rownames(vec) <- oral_pneumonia_df$sample_id
    vec$sample_id <- rownames(vec) 
    oral_pneumonia_df <- merge(oral_pneumonia_df, vec, by = "sample_id", all = TRUE)
    
    oral_pneumonia_df$cluster <- oral_pneumonia_df$vec
    ggplot(oral_pneumonia_df, aes(x=cluster, y = shannon)) + geom_boxplot() + stat_compare_means()
      
    oral_pneumonia_df$cluster[oral_pneumonia_df$vec=="comp1"] <- "LowDiversity"
    oral_pneumonia_df$cluster[oral_pneumonia_df$vec=="comp2"] <- "HighDiversity"
   
     # SHANNON BY CLUSTER - ORAL. 
    mycomparisons <- list(c("LowDiversity", "HighDiversity"))
    shannon_oral_cluster <- ggboxplot(oral_pneumonia_df, x = "cluster", y = "shannon", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#0072B2"), xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) 
ggsave("./results/shannon_oral_cluster.jpeg",  width = 6, height = 5)
    
      # QPCR BY CLUSTER - ORAL. 
    mycomparisons <- list(c("LowDiversity", "HighDiversity"))
    qpcr_oral_cluster <- ggboxplot(oral_pneumonia_df, x = "cluster", y = "qPCR_16S_mean", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#0072B2"), xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_y_log10() +
     stat_compare_means(comparisons = mycomparisons) 
ggsave("./results/qpcr_oral_cluster.jpeg",  width = 6, height = 5)
    
    
    #######################################################
    ##### Permanova
    #######################################################
 ###### TRACHEAL DMM

 indx = which(tracheal_se$AspPnaGroup=="MAsP" | tracheal_se$AspPnaGroup=="NonMAsP")
    tracheal_pneumonia_se = tracheal_se[,indx]

    set.seed(1)
    #######################################################
    ##### Permanova
    #######################################################
    
    permanova_AspPnaGroup <- vegan::adonis2(t(assay(tracheal_pneumonia_se,"relabundance")) ~ AspPnaGroup,
                                           data = colData(tracheal_pneumonia_se),
                                           permutations = 9999)
 #               Df SumOfSqs      R2      F Pr(>F)
# AspPnaGroup   1    0.418 0.00832 1.2333 0.2284
  ##### perform tracheal DMM clusters
    
      tracheal_se_dmn <- runDMN(tracheal_pneumonia_se, name = "DMN", k = 1:7)
    trachealPna_Laplace <- plotDMNFit(tracheal_se_dmn, type = "laplace")
    ggsave("./results/trachealPna_Laplace.jpeg")
    
    
    RS.best_fit_tracheal <- getBestDMNFit(tracheal_se_dmn, type = "laplace")
    #### 3 class is best fit for tracheal samples. 
    ### get the assignment probabilities
    probtracheal <- DirichletMultinomial::mixture(getBestDMNFit(tracheal_se_dmn))
    # Add column names
    colnames(probtracheal) <- c("comp1", "comp2", "comp3")
    # For each row, finds column that has the highest value. Then extract the column 
# names of highest values and merge with the corresponding metadata dataframe extracted from the summarized experiment. 
vectracheal <- colnames(probtracheal)[max.col(probtracheal, ties.method = "first")]
    vectracheal <-as.data.frame(vectracheal)
    table(vectracheal$vectracheal)
  # extract corresponding dataframe for tracheal samples. 
    tracheal_pneumonia_df <- as.data.frame(colData(tracheal_pneumonia_se))
    ##### obtain the DMM cluster assignment from vec and merge with dataframe. 
    rownames(vectracheal) <- tracheal_pneumonia_df$sample_id
    vectracheal$sample_id <- rownames(vectracheal) 
    tracheal_pneumonia_df <- merge(tracheal_pneumonia_df, vectracheal, by = "sample_id", all = TRUE)
    
    tracheal_pneumonia_df$cluster <- tracheal_pneumonia_df$vectracheal
    ggplot(tracheal_pneumonia_df, aes(x=cluster, y = shannon)) + geom_boxplot() + stat_compare_means()
      
    tracheal_pneumonia_df$cluster[tracheal_pneumonia_df$vectracheal=="comp1"] <- "LowDiversity"
    tracheal_pneumonia_df$cluster[tracheal_pneumonia_df$vectracheal=="comp2"] <- "HighDiversity"
   tracheal_pneumonia_df$cluster[tracheal_pneumonia_df$vectracheal=="comp3"] <- "IntermediateDiversity"
   
   tracheal_pneumonia_df$cluster <- ordered(tracheal_pneumonia_df$cluster, levels = c("LowDiversity", "IntermediateDiversity", "HighDiversity"))

     # SHANNON BY CLUSTER - Tracheal 
   
    mycomparisons <- list(c("LowDiversity", "HighDiversity"), c("IntermediateDiversity", "HighDiversity"))
    shannon_tracheal_cluster <- ggboxplot(tracheal_pneumonia_df, x = "cluster", y = "shannon", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + theme(axis.text.x = element_text(size=6) )
ggsave("./results/shannon_tracheal_cluster.jpeg",  width = 6, height = 5)
    

      # QPCR BY CLUSTER - tracheal 
    qpcr_tracheal_cluster <- ggboxplot(tracheal_pneumonia_df, x = "cluster", y = "qPCR_16S_mean", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_y_log10() +
     stat_compare_means(comparisons = mycomparisons) + theme(axis.text.x = element_text(size=6) )
ggsave("./results/qpcr_tracheal_cluster.jpeg",  width = 6, height = 5)
    


  #######################################################################################################
    # Beta Diversity oral BY DMM CLUSTERS
    ######################################################################################################
    oral_pneumonia_rel_abund_assay <- assays(oral_pneumonia_se)$relabundance
    
    # Calculates distances between samples. Because taxa is in
    # columns, it is used to compare different samples. We transpose the
    # assay to get taxa to columns
    oral_pneumonia_manhattan_dist <- vegan::vegdist(t(oral_pneumonia_rel_abund_assay), method = "manhattan")
    oral_pneumonia_manhattan_pcoa <- ecodist::pco(oral_pneumonia_manhattan_dist)
    # Creates a data frame from principal coordinates
    oral_pneu_manhattan_pcoa_df <- data.frame(pcoa1 = oral_pneumonia_manhattan_pcoa$vectors[,1], 
                                     pcoa2 = oral_pneumonia_manhattan_pcoa$vectors[,2])
    
    ## merge with metadata to make this analyzable by ggpot as per Schloss youtube video approach
    rownames(oral_pneu_manhattan_pcoa_df) <-  oral_pneumonia_df$sample_id
    oral_pneu_manhattan_pcoa_df$sample_id <-  rownames(oral_pneu_manhattan_pcoa_df) 
    oral_pneumonia_df <- merge(oral_pneumonia_df, oral_pneu_manhattan_pcoa_df, by = "sample_id", all = TRUE) 
    
#### obtain centroids

oral_pneumonia_df %>% filter(cluster=="LowDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa1meancluster = mean(pcoa1, na.rm = TRUE))
oral_pneumonia_df %>% filter(cluster=="HighDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa1meancluster = mean(pcoa1, na.rm = TRUE))

oral_pneumonia_df <- oral_pneumonia_df %>% mutate(pcoa1meancluster = case_when(cluster=="LowDiversity" ~ -0.004846234,  cluster=="HighDiversity" ~  0.005884712))

oral_pneumonia_df %>% filter(cluster=="LowDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa2meancluster = mean(pcoa2, na.rm = TRUE))
oral_pneumonia_df %>% filter(cluster=="HighDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa2meancluster = mean(pcoa2, na.rm = TRUE))

oral_pneumonia_df <- oral_pneumonia_df %>% mutate(pcoa2meancluster = case_when(cluster=="LowDiversity" ~ -0.006206808,  cluster=="HighDiversity" ~  0.007536838))


oral_pcoa_clusters <- 
ggplot(oral_pneumonia_df, aes(x=pcoa1, xend = pcoa1meancluster,y = pcoa2, yend = pcoa2meancluster, color = cluster, fill = cluster)) + 
 stat_ellipse(level = 0.8,  show.legend = FALSE, size = 1.2) +
  geom_point(size=1.5, show.legend = FALSE) +
geom_point(data = oral_pneumonia_df, mapping = aes(x=pcoa1meancluster, y = pcoa2meancluster), size = 5, show.legend = FALSE) +
geom_segment(size = 0.1, show.legend = FALSE) +
labs(x= "PCoA axis1", y = "PCoA axis 2") + scale_color_manual(name = NULL, breaks = c("LowDiversity", "HighDiversity"), values = c("#D55E00", "#0072B2")) + theme_classic() + ggtitle("PCoA for OPS samples by DMM cluster") + annotate(geom="text", x=-0.03, y=-0.04, label="R2=0.058, p<0.0001",color="black")
ggsave("./results/oral_pcoa_clusters.jpeg", width = 6, height = 5)
    
    
 #######################################################
    ##### Permanova for oral
    #######################################################
      colData(oral_pneumonia_se)$cluster <- oral_pneumonia_df$cluster
    permanova_oralcluster <- vegan::adonis2(t(assay(oral_pneumonia_se,"relabundance")) ~ cluster,
                                           data = colData(oral_pneumonia_se),
                                           permutations = 9999)
   #      Df SumOfSqs      R2      F Pr(>F)    
# cluster    1    2.053 0.05815 7.5328  1e-04 ***


  #######################################################################################################
    # Beta Diversity TRACHEAL BY DMM CLUSTERS
    ######################################################################################################
    tracheal_pneumonia_rel_abund_assay <- assays(tracheal_pneumonia_se)$relabundance
    
    # Calculates distances between samples. Because taxa is in
    # columns, it is used to compare different samples. We transpose the
    # assay to get taxa to columns
    tracheal_pneumonia_manhattan_dist <- vegan::vegdist(t(tracheal_pneumonia_rel_abund_assay), method = "manhattan")
    tracheal_pneumonia_manhattan_pcoa <- ecodist::pco(tracheal_pneumonia_manhattan_dist)
    # Creates a data frame from principal coordinates
    tracheal_pneu_manhattan_pcoa_df <- data.frame(pcoa1 = tracheal_pneumonia_manhattan_pcoa$vectors[,1], 
                                     pcoa2 = tracheal_pneumonia_manhattan_pcoa$vectors[,2])
    
    ## merge with metadata to make this analyzable by ggpot as per Schloss youtube video approach
    rownames(tracheal_pneu_manhattan_pcoa_df) <-  tracheal_pneumonia_df$sample_id
    tracheal_pneu_manhattan_pcoa_df$sample_id <-  rownames(tracheal_pneu_manhattan_pcoa_df) 
    tracheal_pneumonia_df <- merge(tracheal_pneumonia_df, tracheal_pneu_manhattan_pcoa_df, by = "sample_id", all = TRUE) 
    
#### obtain centroids

tracheal_pneumonia_df %>% filter(cluster=="LowDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa1meancluster = mean(pcoa1, na.rm = TRUE))
tracheal_pneumonia_df %>% filter(cluster=="IntermediateDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa1meancluster = mean(pcoa1, na.rm = TRUE))
tracheal_pneumonia_df %>% filter(cluster=="HighDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa1meancluster = mean(pcoa1, na.rm = TRUE))

tracheal_pneumonia_df <- tracheal_pneumonia_df %>% mutate(pcoa1meancluster = case_when(cluster=="LowDiversity" ~ 0.003208632,  cluster=="IntermediateDiversity" ~   -0.01149292,  cluster=="HighDiversity" ~  0.0054998))

tracheal_pneumonia_df %>% filter(cluster=="LowDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa2meancluster = mean(pcoa2, na.rm = TRUE))
tracheal_pneumonia_df %>% filter(cluster=="IntermediateDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa2meancluster = mean(pcoa2, na.rm = TRUE))
tracheal_pneumonia_df %>% filter(cluster=="HighDiversity") %>% dplyr::group_by(cluster) %>% summarize(pcoa2meancluster = mean(pcoa2, na.rm = TRUE))

tracheal_pneumonia_df <- tracheal_pneumonia_df %>% mutate(pcoa2meancluster = case_when(cluster=="LowDiversity" ~ -0.008908625,  cluster=="IntermediateDiversity" ~  0.002455425,  cluster=="HighDiversity" ~  0.00927519))


tracheal_pcoa_clusters <- 
ggplot(tracheal_pneumonia_df, aes(x=pcoa1, xend = pcoa1meancluster,y = pcoa2, yend = pcoa2meancluster, color = cluster, fill = cluster)) + 
 stat_ellipse(level = 0.8,  show.legend = FALSE, size = 1.2) +
  geom_point(size=1.5, show.legend = FALSE) +
geom_point(data = tracheal_pneumonia_df, mapping = aes(x=pcoa1meancluster, y = pcoa2meancluster), size = 5, show.legend = FALSE) +
geom_segment(size = 0.1, show.legend = FALSE) +
labs(x= "PCoA axis1", y = "PCoA axis 2") + scale_color_manual(name = NULL, breaks = c("LowDiversity", "IntermediateDiversity", "HighDiversity"), values = c("#D55E00", "#999999", "#0072B2")) + theme_classic() + ggtitle("PCoA for ETA samples by DMM cluster") + annotate(geom="text", x=-0.02, y=-0.04, label="R2=0.118, p<0.0001",color="black")
ggsave("./results/tracheal_pcoa_clusters.jpeg", width = 6, height = 5)
    
    
 #######################################################
    ##### Permanova for tracheal
    #######################################################
      colData(tracheal_pneumonia_se)$cluster <- tracheal_pneumonia_df$cluster
    permanova_trachealcluster <- vegan::adonis2(t(assay(tracheal_pneumonia_se,"relabundance")) ~ cluster,
                                           data = colData(tracheal_pneumonia_se),
                                           permutations = 9999)

    # Df SumOfSqs      R2      F Pr(>F)    
# cluster    2    5.948 0.11836 9.8003  1e-04 ***


####### arrange graph with basic ecological features for pneumonia by clusters. 

PneumoniaClustersEcologicalFigure <- ggarrange(qpcr_oral_cluster, shannon_oral_cluster, oral_pcoa_clusters, qpcr_tracheal_cluster, shannon_tracheal_cluster, tracheal_pcoa_clusters, ncol = 3, nrow = 2, widths = c(1, 1, 2))
ggsave("./results/PneumoniaClustersEcologicalFigure.jpeg", width = 16, height = 10)
 

#################### Relative abundance per cluster ###################

  
trachealClusterDensityAbundance <- plotAbundanceDensity(tracheal_pneumonia_se, layout = "density", abund_values = "relabundance", n = 10, colour_by="cluster" ) + scale_x_log10() + theme(legend.position = "bottom") + scale_fill_manual(name = NULL, breaks = c("LowDiversity", "IntermediateDiversity", "HighDiversity"), values = c("#D55E00", "#999999", "#0072B2")) + 
 scale_color_manual(name = NULL, breaks = c("LowDiversity","IntermediateDiversity",  "HighDiversity"), values = c("#D55E00","#999999", "#0072B2")) 

oralClusterDensityAbundance  <- plotAbundanceDensity(oral_pneumonia_se, layout = "density", abund_values = "relabundance",n = 10, colour_by="cluster" ) + scale_x_log10()+ theme(legend.position = "bottom") + scale_fill_manual(name = NULL, breaks = c("LowDiversity", "HighDiversity"), values = c("#D55E00", "#0072B2")) + 
 scale_color_manual(name = NULL, breaks = c("LowDiversity", "HighDiversity"), values = c("#D55E00", "#0072B2")) 

ClusterDensityAbundance <- ggarrange(oralClusterDensityAbundance, trachealClusterDensityAbundance, ncol = 2, nrow = 1)
ggsave("./results/ClusterDensityAbundance.jpeg", width = 16, height = 8)


################################ VOLCANO PLOTS BY CLUSTERS

   ###### CLR FOR ORAL WITH LIMMA
clr = assays(oral_pneumonia_se)$clr
    group_ = colData(oral_pneumonia_se)$cluster
    mm <- model.matrix(~0 + group_)
    fit <- lmFit(clr, mm)
    head(fit)
    
      # Comparison between group_MAsP minus group_IntubControl
        contr <- makeContrasts(group_HighDiversity - group_LowDiversity, levels = colnames(coef(fit)))

      # Estimate contrast for each gene
      tmp <- contrasts.fit(fit, contr)
      # Empirical Bayes smoothing of standard errors
      tmp <- eBayes(tmp)
            top.table <- topTable(tmp, sort.by = "P", n = Inf)
      write.csv(top.table, './results/oral_clusters_diffabundance.csv')
     
      diff_taxa = top.table
      # row.names(diff_genes) = diff_genes$X
      diff_df <- diff_taxa[c( "logFC","adj.P.Val")]
      names(diff_df) <- c("logFC","adj.P.Val")
      
      diff_df["Genus"]<-row.names(diff_taxa)
      head(diff_df)
      diff_df["Significance"] <- "NotSig"
      diff_df[which(diff_df['adj.P.Val'] < 0.05),"Significance"] <- "adj.P.Val < 0.05"
      
      sig_diff_df = diff_df[which(diff_df['adj.P.Val'] < 0.05),]
      pos_fc = sig_diff_df[which(sig_diff_df$logFC>0),]
      top_peaks <- pos_fc[with(pos_fc, order(-pos_fc$logFC,pos_fc$adj.P.Val)),][1:20,]
      top_peaks <- top_peaks %>% drop_na()
      
      neg_fc = sig_diff_df[which(sig_diff_df$logFC<0),]
      top_peaks <- rbind(top_peaks, neg_fc[with(neg_fc, order(neg_fc$logFC,neg_fc$adj.P.Val)),][1:10,])
      top_peaks <- top_peaks %>% drop_na()
      

      oral_clusters_diffabundance <- ggplot(diff_df, aes(x = logFC, y = -log10(adj.P.Val))) +
        geom_point(aes(color = Significance)) +
        scale_color_manual(values = c("blue","black")) +
        theme_bw(base_size = 16) +
        geom_text_repel(
          data = top_peaks,
          aes(label = Genus),
          size = 5,
          box.padding = unit(0.35, "lines"),
          point.padding = unit(0.3, "lines")
        )
      ggsave("./results/oral_clusters_diffabundance.jpeg", width = 10, height = 6)
    

########### differential abundance for tracheal clusters
      

         ###### CLR FOR tracheal WITH LIMMA
clr = assays(tracheal_pneumonia_se)$clr
    group_ = colData(tracheal_pneumonia_se)$cluster
    mm <- model.matrix(~0 + group_)
    fit <- lmFit(clr, mm)
    head(fit)
    
    for(comp in 1:3){
      if(comp == 1){
        # Comparison between group_LowDiversity minus group_IntermediateDiversity
        contr <- makeContrasts(group_LowDiversity - group_IntermediateDiversity, levels = colnames(coef(fit)))
        file_name = "tracheal_diff_abundance_Low_vs_Interm"
      }else if(comp == 2){
        # Comparison between group_LowDiversity minus group_HighDiversity
        contr <- makeContrasts(group_LowDiversity - group_HighDiversity, levels = colnames(coef(fit)))
        file_name = "tracheal_diff_abundance_Low_vs_High"
      }else if(comp == 3){
        # Comparison between group_IntermediateDiversity minus group_HighDiversity
        contr <- makeContrasts(group_IntermediateDiversity - group_HighDiversity, levels = colnames(coef(fit)))
        file_name = "tracheal_diff_abundance_Interm_vs_High"
      }  
      
      # Estimate contrast for each gene
      tmp <- contrasts.fit(fit, contr)
      # Empirical Bayes smoothing of standard errors
      tmp <- eBayes(tmp)
            top.table <- topTable(tmp, sort.by = "P", n = Inf)
      write.csv(top.table,file = paste0('./results/',file_name,'.csv'))
      
      diff_taxa = top.table
      # row.names(diff_genes) = diff_genes$X
      diff_df <- diff_taxa[c( "logFC","adj.P.Val")]
      names(diff_df) <- c("logFC","adj.P.Val")
      
      diff_df["Genus"]<-row.names(diff_taxa)
      head(diff_df)
      diff_df["Significance"] <- "NotSig"
      diff_df[which(diff_df['adj.P.Val'] < 0.05),"Significance"] <- "adj.P.Val < 0.05"
      
      sig_diff_df = diff_df[which(diff_df['adj.P.Val'] < 0.05),]
      pos_fc = sig_diff_df[which(sig_diff_df$logFC>0),]
      top_peaks <- pos_fc[with(pos_fc, order(-pos_fc$logFC,pos_fc$adj.P.Val)),][1:20,]
      top_peaks <- top_peaks %>% drop_na()
      
      neg_fc = sig_diff_df[which(sig_diff_df$logFC<0),]
      top_peaks <- rbind(top_peaks, neg_fc[with(neg_fc, order(neg_fc$logFC,neg_fc$adj.P.Val)),][1:10,])
      top_peaks <- top_peaks %>% drop_na()
    
      p <- ggplot(diff_df, aes(x = logFC, y = -log10(adj.P.Val))) +
        geom_point(aes(color = Significance)) +
        scale_color_manual(values = c("blue","black")) +
        theme_bw(base_size = 16) +
        geom_text_repel(
          data = top_peaks,
          aes(label = Genus),
          size = 5,
          box.padding = unit(0.35, "lines"),
          point.padding = unit(0.3, "lines")
        )
      
      save_plot(paste0('./results/','_',file_name,'.jpg'), p,
                base_aspect_ratio = 1.5, base_height =7)
    }
    
    
    
    
    
##########################################3
    ################## ANALYSES IN THE PNEUMONIA DF FOR MORTALITY
    
#### oral mortality
oral_pneumonia_df <- oral_pneumonia_df %>% mutate(Mortality60day = case_when(Mortality90Day==1 ~ "NonSurvivors", 
                                                                            Mortality90Day==0 ~ "Survivors" ))



 # SHANNON BY MORTALITY - ORAL. 
shannon_oral_mortality <- ggboxplot(oral_pneumonia_df, x = "Mortality60day", y = "shannon", 
                   color = "Mortality60day",add = "jitter" , palette = c("#000000", "#FF99CC"), xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + 
     stat_compare_means() 
ggsave("./results/shannon_oral_mortality.jpeg",  width = 6, height = 5)

ggboxplot(oral_pneumonia_df, x = "Mortality60day", y = "shannon", 
                   color = "Mortality60day",add = "jitter" , palette = c("#000000", "#FF99CC"), xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + 
     stat_compare_means() + facet_wrap(~AspPnaGroup)
    
  # QPCR BY MORTALITY - ORAL. 
qpcr_oral_mortality <- ggboxplot(oral_pneumonia_df, x = "Mortality60day", y = "qPCR_16S_mean", 
                   color = "Mortality60day",add = "jitter" , palette = c("#000000", "#FF99CC"), xlab = "", ylab = "16S qPCR copies (log) - OPS", legend = "none" ) + scale_y_log10() +
     stat_compare_means() 
ggsave("./results/qpcr_oral_mortality.jpeg",  width = 6, height = 5)
    

  #######################################################################################################
    # Beta Diversity oral BY DMM MORTALITY
    ######################################################################################################

    
#### obtain centroids

oral_pneumonia_df %>% filter(Mortality60day=="NonSurvivors") %>% dplyr::group_by(Mortality60day) %>% summarize(pcoa1meanmortality = mean(pcoa1, na.rm = TRUE))
oral_pneumonia_df %>% filter(Mortality60day=="Survivors") %>% dplyr::group_by(Mortality60day) %>% summarize(pcoa1meanmortality = mean(pcoa1, na.rm = TRUE))

oral_pneumonia_df <- oral_pneumonia_df %>% mutate(pcoa1meanmortality = case_when(Mortality60day=="NonSurvivors" ~ -0.003273606,  Mortality60day=="Survivors" ~   0.0009548019))

oral_pneumonia_df %>% filter(Mortality60day=="NonSurvivors") %>% dplyr::group_by(Mortality60day) %>% summarize(pcoa2meanmortality = mean(pcoa2, na.rm = TRUE))
oral_pneumonia_df %>% filter(Mortality60day=="Survivors") %>% dplyr::group_by(Mortality60day) %>% summarize(pcoa2meanmortality = mean(pcoa2, na.rm = TRUE))

oral_pneumonia_df <- oral_pneumonia_df %>% mutate(pcoa2meanmortality = case_when(Mortality60day=="NonSurvivors" ~ -0.002709817,  Mortality60day=="Survivors" ~   0.0007903632))


oral_pcoa_mortality <- 
ggplot(oral_pneumonia_df, aes(x=pcoa1, xend = pcoa1meanmortality,y = pcoa2, yend = pcoa2meanmortality, color = Mortality60day, fill = Mortality60day)) + 
 stat_ellipse(level = 0.8,  show.legend = FALSE, size = 1.2) +
  geom_point(size=1.5, show.legend = FALSE) +
geom_point(data = oral_pneumonia_df, mapping = aes(x=pcoa1meanmortality, y = pcoa2meanmortality), size = 5, show.legend = FALSE) +
geom_segment(size = 0.1, show.legend = FALSE) +
labs(x= "PCoA axis1", y = "PCoA axis 2") + scale_color_manual(name = NULL, breaks = c("NonSurvivors", "Survivors"), values = c("#000000", "#FF99CC")) + theme_classic() + ggtitle("PCoA for OPS samples by 60 Day Mortality") + annotate(geom="text", x=-0.03, y=-0.04, label="R2=0.008, p=0.37",color="black")
ggsave("./results/oral_pcoa_mortality.jpeg", width = 6, height = 5)


      colData(oral_pneumonia_se)$Mortality60day <- oral_pneumonia_df$Mortality60day
    permanova_oralcluster <- vegan::adonis2(t(assay(oral_pneumonia_se,"relabundance")) ~ Mortality60day,
                                           data = colData(oral_pneumonia_se),
                                           permutations = 9999)

    
    

#### tracheal mortality
tracheal_pneumonia_df <- tracheal_pneumonia_df %>% mutate(Mortality60day = case_when(Mortality90Day==1 ~ "NonSurvivors", 
                                                                            Mortality90Day==0 ~ "Survivors" ))



 # SHANNON BY MORTALITY - tracheal. 
shannon_tracheal_mortality <- ggboxplot(tracheal_pneumonia_df, x = "Mortality60day", y = "shannon", 
                   color = "Mortality60day",add = "jitter" , palette = c("#000000", "#FF99CC"), xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + 
     stat_compare_means() 
ggsave("./results/shannon_tracheal_mortality.jpeg",  width = 6, height = 5)
    
  # QPCR BY MORTALITY - tracheal. 
qpcr_tracheal_mortality <- ggboxplot(tracheal_pneumonia_df, x = "Mortality60day", y = "qPCR_16S_mean", 
                   color = "Mortality60day",add = "jitter" , palette = c("#000000", "#FF99CC"), xlab = "", ylab = "16S qPCR copies (log) - ETA", legend = "none" ) + scale_y_log10() +
     stat_compare_means() 
ggsave("./results/qpcr_tracheal_mortality.jpeg",  width = 6, height = 5)
    

#### examine shannon difference within the MAsP subjects
shannon_tracheal_mortality_AspGroup <- ggboxplot(tracheal_pneumonia_df, x = "Mortality60day", y = "shannon", 
                   color = "Mortality60day",add = "jitter" , palette = c("#000000", "#FF99CC"), xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + 
     stat_compare_means() + facet_wrap(~ AspPnaGroup)
ggsave("./results/shannon_tracheal_mortality_AspGroup.jpeg",  width = 6, height = 5)


  #######################################################################################################
    # Beta Diversity tracheal BY DMM MORTALITY
    ######################################################################################################

    
#### obtain centroids

tracheal_pneumonia_df %>% filter(Mortality60day=="NonSurvivors") %>% dplyr::group_by(Mortality60day) %>% summarize(pcoa1meanmortality = mean(pcoa1, na.rm = TRUE))
tracheal_pneumonia_df %>% filter(Mortality60day=="Survivors") %>% dplyr::group_by(Mortality60day) %>% summarize(pcoa1meanmortality = mean(pcoa1, na.rm = TRUE))

tracheal_pneumonia_df <- tracheal_pneumonia_df %>% mutate(pcoa1meanmortality = case_when(Mortality60day=="NonSurvivors" ~ -0.001151045,  Mortality60day=="Survivors" ~   0.0003667045))

tracheal_pneumonia_df %>% filter(Mortality60day=="NonSurvivors") %>% dplyr::group_by(Mortality60day) %>% summarize(pcoa2meanmortality = mean(pcoa2, na.rm = TRUE))
tracheal_pneumonia_df %>% filter(Mortality60day=="Survivors") %>% dplyr::group_by(Mortality60day) %>% summarize(pcoa2meanmortality = mean(pcoa2, na.rm = TRUE))

tracheal_pneumonia_df <- tracheal_pneumonia_df %>% mutate(pcoa2meanmortality = case_when(Mortality60day=="NonSurvivors" ~ -0.007569371,  Mortality60day=="Survivors" ~     0.002411481))


tracheal_pcoa_mortality <- 
ggplot(tracheal_pneumonia_df, aes(x=pcoa1, xend = pcoa1meanmortality,y = pcoa2, yend = pcoa2meanmortality, color = Mortality60day, fill = Mortality60day)) + 
 stat_ellipse(level = 0.8,  show.legend = FALSE, size = 1.2) +
  geom_point(size=1.5, show.legend = FALSE) +
geom_point(data = tracheal_pneumonia_df, mapping = aes(x=pcoa1meanmortality, y = pcoa2meanmortality), size = 5, show.legend = FALSE) +
geom_segment(size = 0.1, show.legend = FALSE) +
labs(x= "PCoA axis1", y = "PCoA axis 2") + scale_color_manual(name = NULL, breaks = c("NonSurvivors", "Survivors"), values = c("#000000", "#FF99CC")) + theme_classic() + ggtitle("PCoA for ETA samples by 60 Day Mortality") + annotate(geom="text", x=-0.01, y=-0.04, label="R2=0.016, p=0.0.008",color="black")
ggsave("./results/tracheal_pcoa_mortality.jpeg", width = 6, height = 5)


      colData(tracheal_pneumonia_se)$Mortality60day <- tracheal_pneumonia_df$Mortality60day
    permanova_trachealcluster <- vegan::adonis2(t(assay(tracheal_pneumonia_se,"relabundance")) ~ Mortality60day,
                                           data = colData(tracheal_pneumonia_se),
                                           permutations = 9999)



    
    
    
    
    
```








```{r  clinical analyses by bacterial DMM clusters}



###### survival analysis by AspGroup
###### create a dataframe with alir cases only and baseline only samples. 
# create 60 day survival
AspMetadata$survival_time60_fromICUAdmission <- AspMetadata$survival_time90_fromICUAdmission
AspMetadata$survival_time60_fromICUAdmission[AspMetadata$survival_time60_fromICUAdmission>60] <-61
AspMetadata$survival_status60 <- AspMetadata$survival_status90
table(AspMetadata$survival_status60, as.factor(AspMetadata$survival_time60_fromICUAdmission>60))
# no events beyond day 60 recorded

AspMetadata_basl_alir <- filter(AspMetadata, study_id=="alir" & studydayfollowup=="baseline")
#### obtain only single SubjectIDDay entry per Subject
AspMetadata_basl_alir <- filter(AspMetadata_basl_alir, !duplicated(SubjectIDDay))

# AspMetadata_basl_alir$AspPnaGroup <- ordered(AspMetadata_basl_alir$AspPnaGroup, levels = c("MAsP", "NonMAsP", "IntubControl"))

##### incporate the anaerobic antibiotic data
# AspMetadata_basl_alir <- merge(AspMetadata_basl_alir, tblAntibioticHistory, by = "SubjectID", all.x = TRUE)


survival_fit_bygroup60 <- survfit(Surv(survival_time60_fromICUAdmission, survival_status60)~AspPnaGroup , data=AspMetadata_basl_alir)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green", "red"), title = "Kaplan-Meier Curve within 60 days", risk.table = TRUE, tables.theme = clean_theme()) 

# publishable plot
### #
# surv_AspPnagroup <-  ggsurvplot(survival_fit_bygroup60, pval=TRUE, title = "60 day survival by Aspiration Diagnosis", legend.labs=c("MAsP",  "NonMAsP", "IntubControl"), palette = c("#000000", "#E69F00", "#56B4E9"), risk.table = TRUE, tables.theme = clean_theme(), legend.title = "Group:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlab = "Days Post Intubation")  
# jpeg("surv_AspPnagroup.jpeg", width = 600, height = 500)
# print(surv_AspPnagroup, newpage = FALSE)
# dev.off()


############# liberation analysis
AspMetadata_basl_alir$timetoliberation[AspMetadata_basl_alir$timetoliberation>60] <-61 


liberation_fit_bygroup60 <- survfit(Surv(timetoliberation, intubation_status)~AspPnaGroup, data=AspMetadata_basl_alir)
ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Kaplan-Meier Curve within 90 days") 

liberation_AspPnaGroups <-  ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Liberation from mechanical ventilation by Aspiration Diagnosis", legend.labs=c("MAsP",  "NonMAsP", "IntubControl"), palette = c("#000000", "#E69F00", "#56B4E9"), risk.table = TRUE, legend.title = "Group:", font.legend = c(14, "bold", "black"), font.title = c(12, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("liberation_AspPnaGroups.jpeg", width = 600, height = 500)
print(liberation_AspPnaGroups, newpage = FALSE)
while (!is.null(dev.list()))  dev.off()


liberation_fit_bygroup60 <- survfit(Surv(timetoliberation, intubation_status)~RiskFactor_categ, data=AspMetadataSubjectID_MASP)
ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Kaplan-Meier Curve within 90 days") 


liberation_RiskFactorMASP <-  ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Liberation from mechanical ventilation Risk Factor Category in MAsP", legend.labs=c("AcuteRF",  "ChronicRF", "None"), palette = c("#CC0033", "#006633", "#333333"), risk.table = TRUE, legend.title = "Group:", font.legend = c(14, "bold", "black"), font.title = c(12, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("liberation_RiskFactorMASP.jpeg", width = 600, height = 500)
print(liberation_RiskFactorMASP, newpage = FALSE)
dev.off()

liberation_fit_bygroup60 <- survfit(Surv(timetoliberation, intubation_status)~RiskFactor_categ, data=AspMetadataSubjectID_NonMASP)
ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Kaplan-Meier Curve within 90 days") 


liberation_RiskFactorNonMASP <-  ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Liberation from mechanical ventilation Risk Factor Category in NonMASP", legend.labs=c("AcuteRF",  "ChronicRF", "None"), palette = c("#CC0033", "#006633", "#333333"), risk.table = TRUE, legend.title = "Group:", font.legend = c(14, "bold", "black"), font.title = c(12, "bold", "black"), xlab = "Days Post Intubation") 
jpeg("liberation_RiskFactorNonMASP.jpeg", width = 600, height = 500)
print(liberation_RiskFactorNonMASP, newpage = FALSE)
dev.off()




############## similar graphs for the clusters in the oral and tracheal dataframes separately. 
##### oral clusters survival
oral_pneumonia_df$cluster <- ordered(oral_pneumonia_df$cluster, levels = c("LowDiversity", "HighDiversity")) 
survival_fit_bygroup60 <- survfit(Surv(survival_time90_fromICUAdmission, survival_status90)~cluster , data=oral_pneumonia_df)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "green"), title = "Kaplan-Meier Curve within 60 days") 

# publishable plot
surv_OralClusters<-  ggsurvplot(survival_fit_bygroup60, pval=TRUE, title = "60 day survival by Cluster - OPS", legend.labs=c("LowDiversity",  "HighDiversity"), palette = c("#D55E00", "#0072B2"), risk.table = TRUE, tables.theme = clean_theme(), legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(16, "bold", "black"), xlim = c(0,60) , xlab = "Days Post Intubation") 
jpeg("surv_OralClusters.jpeg", width = 600, height = 500)
print(surv_OralClusters, newpage = FALSE)
dev.off()

oral_pneumonia_df$Home02

#### obtain HR for oral cluster
survival_coxph_bygroup90 <- coxph(Surv(survival_time90_fromICUAdmission, survival_status90)~ cluster , data=oral_pneumonia_df)
#  coef exp(coef) se(coef)      z Pr(>|z|)   
# cluster.L -0.8479    0.4283   0.3259 -2.602  0.00928 **
# cluster.L    0.4283      2.335    0.2261    0.8113

###### adjusted for confounders
survival_coxph_bygroup90 <- coxph(Surv(survival_time90_fromICUAdmission, survival_status90)~ cluster +    total_ABx_score_day1_Convex + HistCOPD + Age + Gender + Home02, data=oral_pneumonia_df)
summary(survival_coxph_bygroup90)
#       coef exp(coef) se(coef)      z Pr(>|z|)    
 #cluster.L           -0.7853    0.4560   0.3312 -2.371   0.0177 *
#     AspPnaGroupNonMAsP    0.3730    1.4521   0.4266  0.874   0.3819  
# day1AnaerobicAbxyes  0.2253    1.2527   0.4132  0.545   0.5856  

# cluster.L               0.456     2.1931    0.2382    0.8727
# AspPnaGroupNonMAsP       1.452     0.6886    0.6294    3.3505
# day1AnaerobicAbxyes     1.253     0.7983    0.5573    2.8157

#### oral cluster liberation

liberation_fit_bygroup60 <- survfit(Surv(timetoliberation, intubation_status)~cluster, data=oral_pneumonia_df)
ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Kaplan-Meier Curve within 90 days") 

liberation_oralCluster <-  ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Liberation from mechanical ventilation by Cluster - OPS", legend.labs=c("LowDiversity",  "HighDiversity"), palette = c("#D55E00", "#0072B2"), risk.table = TRUE, legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(12, "bold", "black"),xlim = c(0,60), xlab = "Days Post Intubation") 
jpeg("liberation_oralCluster.jpeg", width = 600, height = 500)
print(liberation_oralCluster, newpage = FALSE)
dev.off()

liberation_coxph_bygroup90 <- coxph(Surv(timetoliberation, intubation_status)~ cluster  + total_ABx_score_day1_Convex + HistCOPD + Age + Gender + Home02, data=oral_pneumonia_df)
summary(liberation_coxph_bygroup90)


########### tracheal clusters survival
survival_fit_bygroup60 <- survfit(Surv(survival_time90_fromICUAdmission, survival_status90)~cluster , data=tracheal_pneumonia_df)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "orange", "green"), title = "Kaplan-Meier Curve within 60 days") 

# publishable plot
surv_TrachealClusters<-  ggsurvplot(survival_fit_bygroup60, pval=TRUE, title = "60 day survival by Cluster - ETA", legend.labs=c("LowDiversity", "IntermediateDiversity", "HighDiversity"), palette = c("#D55E00", "#999999", "#0072B2"), risk.table = TRUE, tables.theme = clean_theme(), legend.title = "Cluster:", font.legend = c(10, "bold", "black"), font.title = c(16, "bold", "black"), xlim = c(0,60) , xlab = "Days Post Intubation") 
jpeg("surv_TrachealClusters.jpeg", width = 600, height = 500)
print(surv_TrachealClusters, newpage = FALSE)
dev.off()


# obtain antibiotic data
# tracheal_pneumonia_df <-  merge(tracheal_pneumonia_df, tblAntibioticHistory, by = "SubjectID", all.x = TRUE)

#### define high vs. nonhigh cluster

tracheal_pneumonia_df <- tracheal_pneumonia_df %>% mutate(highcluster = case_when(cluster=="LowDiversity" ~ "no",
                                                                                 cluster=="IntermediateDiversity" ~ "no", 
                                                                                cluster=="HighDiversity" ~ "yes"
                                                                                ))
#### obtain HR for tracheal cluster
survival_coxph_bygroup90 <- coxph(Surv(survival_time90_fromICUAdmission, survival_status90)~ highcluster , data=tracheal_pneumonia_df)
summary(survival_coxph_bygroup90)
#  coef exp(coef) se(coef)      z Pr(>|z|) 
# cluster.L -1.2895    0.2754   0.4366 -2.953  0.00314 **
# cluster.Q -0.6711    0.5111   0.3451 -1.945  0.05180 . 
#      exp(coef) exp(-coef) lower .95 upper .95
# cluster.L    0.2754      3.631    0.1170    0.6481
# cluster.Q    0.5111      1.956    0.2599    1.0053

###### adjusted for confounders
survival_coxph_bygroup90 <- coxph(Surv(survival_time90_fromICUAdmission, survival_status90)~ highcluster +total_ABx_score_day1_Convex + HistCOPD + Age + Gender + Home02  , data=tracheal_pneumonia_df)
summary(survival_coxph_bygroup90)

#                       coef exp(coef) se(coef)      z Pr(>|z|)    
# highclusteryes      -1.7843    0.1679   0.6058 -2.945  0.00323 **
# AspPnaGroupNonMAsP   -0.1281    0.8797   0.3489 -0.367  0.71347   
 #day1AnaerobicAbxyes  0.4285    1.5350   0.3678  1.165  0.24398  

#                   exp(coef) exp(-coef) lower .95 upper .95
# highclusteryes         0.1679     5.9553   0.05122    0.5505
# AspPnaGroupNonMAsP      0.8797     1.1367   0.44396    1.7433
# day1AnaerobicAbxyes    1.5350     0.6515   0.74650    3.1564
                  

liberation_fit_bygroup60 <- survfit(Surv(timetoliberation, intubation_status)~cluster, data=tracheal_pneumonia_df)
ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Kaplan-Meier Curve within 90 days") 

liberation_trachealCluster <-  ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Liberation from mechanical ventilation by Cluster - ETA", llegend.labs=c("LowDiversity", "IntermediateDiversity", "HighDiversity"), palette = c("#D55E00", "#999999", "#0072B2"), risk.table = TRUE, legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(12, "bold", "black"), xlim = c(0,60),xlab = "Days Post Intubation") 
jpeg("liberation_trachealCluster.jpeg",  width = 600, height = 500)
print(liberation_trachealCluster, newpage = FALSE)
dev.off()


liberation_coxph_bygroup90 <- coxph(Surv(timetoliberation, intubation_status)~ cluster  , data=tracheal_pneumonia_df)
summary(liberation_coxph_bygroup90)


######## examine a combined dysbiosis survival analysis
                  ##### the high diversity cluster does well in both sampletypes
####### then examine which subjects are high diversity in both sample types and then examine for their survival compared to all others
                  
highDiv_Oral_Subjects <- oral_pneumonia_df$SubjectID[which(oral_pneumonia_df$cluster=="HighDiversity")] # 56 subjects. 
lowDiv_Oral_Subjects <- oral_pneumonia_df$SubjectID[which(oral_pneumonia_df$cluster=="LowDiversity")] # 68 subjects. 

#### mutate a HighDiversityMembership variable in the tracheal dataframe
tracheal_pneumonia_df <- tracheal_pneumonia_df %>% mutate(oralCluster = case_when(
  SubjectID %in% highDiv_Oral_Subjects ~ "highOral", 
  SubjectID %in% lowDiv_Oral_Subjects ~ "lowOral"
  ))

table(tracheal_pneumonia_df$oralCluster)           

tracheal_pneumonia_df <- tracheal_pneumonia_df %>% mutate(CombinedClusters = case_when(
  oralCluster== "highOral" & cluster=="HighDiversity" ~ "CombinedHigh",
  oralCluster== "lowOral" & cluster=="LowDiversity" ~ "CombinedLow",
  oralCluster== "lowOral" & cluster=="IntermediateDiversity" ~ "CombinedLow",
  oralCluster== "lowOral" & cluster=="HighDiversity" ~ "OneComptHigh",
  oralCluster== "highOral" & cluster=="IntermediateDiversity" ~ "OneComptHigh",
  oralCluster== "highOral" & cluster=="LowDiversity" ~ "OneComptHigh"
  ))
table(tracheal_pneumonia_df$CombinedClusters)

######### look at survival by combined clusters. 
tracheal_pneumonia_df$CombinedClusters <- ordered(tracheal_pneumonia_df$CombinedClusters, levels = c("CombinedLow", "OneComptHigh", "CombinedHigh"))
########### tracheal clusters survival
survival_fit_bygroup60 <- survfit(Surv(survival_time90_fromICUAdmission, survival_status90)~CombinedClusters , data=tracheal_pneumonia_df)
summary(survival_fit_bygroup60)
ggsurvplot(survival_fit_bygroup60, pval=TRUE,  palette = c("black", "orange", "green"), title = "Kaplan-Meier Curve within 60 days") 


# publishable plot
surv_CombinedClusters<-  ggsurvplot(survival_fit_bygroup60, pval=TRUE, title = "60 day survival by Combined URT/LRT Clusters", legend.labs=c("CombinedLow", "OneComptHigh", "CombinedHigh"), palette = c("#D55E00", "#CC79A7", "#0072B2"), risk.table = TRUE, tables.theme = clean_theme(), legend.title = "Combined Cluster:", font.legend = c(9, "bold", "black"), font.title = c(16, "bold", "black"), xlim = c(0,60) , xlab = "Days Post Intubation") 
jpeg("surv_CombinedClusters.jpeg", width = 600, height = 500)
print(surv_CombinedClusters, newpage = FALSE)
dev.off()

liberation_fit_bygroup60 <- survfit(Surv(timetoliberation, intubation_status)~CombinedClusters, data=tracheal_pneumonia_df)

liberation_CombinedClusters <-  ggsurvplot(liberation_fit_bygroup60, pval=TRUE, fun = "event", censor.shape = "|", censor = TRUE, title = "Liberation from mechanical ventilation by by Combined URT/LRT Clusters", llegend.labs=c("CombinedLow", "OneComptHigh", "CombinedHigh"), palette = c("#D55E00", "#CC79A7", "#0072B2"), risk.table = TRUE, legend.title = "Cluster:", font.legend = c(14, "bold", "black"), font.title = c(12, "bold", "black"), xlim = c(0,60), xlab = "Days Post Intubation") 
jpeg("liberation_CombinedClusters.jpeg",  width = 1200, height = 500)
print(liberation_CombinedClusters, newpage = FALSE)
dev.off()


###### waffle plot for combined dysbiosis.
table(tracheal_pneumonia_df$oralCluster, tracheal_pneumonia_df$cluster)

  #                  LowDiversity IntermediateDiversity HighDiversity
#   highOral            5                    17            38
#   lowOral            45                    20             5

######## create a publishable waffle graph for the agreement between the calfee and nouraie phenotypes
library(waffle)
clusters_overlap <- c(45, 25, 22, 38)
waffle(clusters_overlap, colors = c("#D55E00", "#CC79A7", "pink", "#0072B2"), legend_pos = "none")
ggsave("./results/clustersorverlap.jpeg")




```

```{r descriptives by DMM bacterial clusters}

#### use the same continuous and categorical variables defined upfront. 

#### oral clusters

cat_table_oralClusters<- print(CreateCatTable(vars = vars_categorical, strata = "cluster" , data=oral_pneumonia_df, test = TRUE, testExact = fisher.test, includeNA = FALSE), pDigits = 2)
con_table_oralClusters <- print(CreateContTable(vars = vars_continuous, strata = "cluster" , data=oral_pneumonia_df), nonnormal = vars_continuous, digits  = 1, pDigits = 2)

write.csv(cat_table_oralClusters, "./results/cat_table_oralClusters.csv")
write.csv(con_table_oralClusters, "./results/con_table_oralClusters.csv")


cat_table_trachealClusters<- print(CreateCatTable(vars = vars_categorical, strata = "cluster" , data=tracheal_pneumonia_df, test = TRUE, testExact = fisher.test, includeNA = FALSE), pDigits = 2)
con_table_trachealClusters <- print(CreateContTable(vars = vars_continuous, strata = "cluster" , data=tracheal_pneumonia_df), nonnormal = vars_continuous, digits  = 1, pDigits = 2)

write.csv(cat_table_trachealClusters, "./results/cat_table_trachealClusters.csv")
write.csv(con_table_trachealClusters, "./results/con_table_trachealClusters.csv")

### supplemental tables included. 

```


```{r  bray curtis distance }
#### create different matrices for each of the sample types
###### take only the alir cases
indx = which(allSamples_se$study_id=="alir")
oraltrachalir_se <- allSamples_se[, indx]
oraltrachalir_se$sampletype
oraltrachalir_rel_abund_assay <- assays(oraltrachalir_se)$relabundance

bray_dist <- vegan::vegdist(t(oraltrachalir_rel_abund_assay), method = "bray")

bray_dist <- reshape2::melt(as.matrix(bray_dist), varnames = c("SampleID1", "SampleID2"))
class(bray_dist)

#### obtain sample type and AspGroupPna categories for each SampleID to allow for group comparisons. 

allSamplesMetadata$SampleID1 <- allSamplesMetadata$sample_id
allSamplesMetadata$SampleID2 <- allSamplesMetadata$sample_id
allSamplesMetadata$sampletype1 <- allSamplesMetadata$sampletype
allSamplesMetadata$sampletype2 <- allSamplesMetadata$sampletype
allSamplesMetadata$SubjectID1 <- allSamplesMetadata$SubjectID
allSamplesMetadata$SubjectID2 <- allSamplesMetadata$SubjectID
###subset small dataset for merging. 

oraltrachalirSampleID_type1 <- subset(allSamplesMetadata, select = c(SampleID1, sampletype1, SubjectID1, AspPnaGroup))
oraltrachalirSampleID_type2 <- subset(allSamplesMetadata, select = c(SampleID2, sampletype2, SubjectID2))

# merge with the distance file
bray_dist1 <- merge(bray_dist, oraltrachalirSampleID_type1, by = "SampleID1", all.x = TRUE)
bray_dist2 <- merge(bray_dist1, oraltrachalirSampleID_type2, by = "SampleID2", all.x = TRUE)

### filter for samples where the SampleID1 and SampleID2 are different and use exact matches for the same Subject
bray_dist3 <- filter(bray_dist2, sampletype1 !=sampletype2)
bray_dist4 <- filter(bray_dist3, SubjectID1== SubjectID2)
ggplot(bray_dist4, aes(x=AspPnaGroup, y = value)) + geom_boxplot() + geom_jitter(size = 0.01) + stat_compare_means()


# mycomparisons <- list(c("MAsP", "NonMAsP"), c("MAsP", "IntubControl"))
brayCurtisbyGroup <- ggboxplot(bray_dist4, x = "AspPnaGroup", y = "value", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "OPS to ETA Bray Curtis Dissimilarity", legend = "none" ) + 
   stat_compare_means() 
ggsave("./results/brayCurtisbyGroup.jpeg",  width = 6, height = 5)




#### repeat analysis for manhattan
manhattan_dist <- vegan::vegdist(t(oraltrachalir_rel_abund_assay), method = "manhattan")

manhattan_dist <- reshape2::melt(as.matrix(manhattan_dist), varnames = c("SampleID1", "SampleID2"))
class(manhattan_dist)

#### obtain sample type and AspGroupPna categories for each SampleID to allow for group comparisons. 

allSamplesMetadata$SampleID1 <- allSamplesMetadata$sample_id
allSamplesMetadata$SampleID2 <- allSamplesMetadata$sample_id
allSamplesMetadata$sampletype1 <- allSamplesMetadata$sampletype
allSamplesMetadata$sampletype2 <- allSamplesMetadata$sampletype
allSamplesMetadata$SubjectID1 <- allSamplesMetadata$SubjectID
allSamplesMetadata$SubjectID2 <- allSamplesMetadata$SubjectID
###subset small dataset for merging. 

oraltrachalirSampleID_type1 <- subset(allSamplesMetadata, select = c(SampleID1, sampletype1, SubjectID1, AspPnaGroup))
oraltrachalirSampleID_type2 <- subset(allSamplesMetadata, select = c(SampleID2, sampletype2, SubjectID2))

# merge with the distance file
manhattan_dist1 <- merge(manhattan_dist, oraltrachalirSampleID_type1, by = "SampleID1", all.x = TRUE)
manhattan_dist2 <- merge(manhattan_dist1, oraltrachalirSampleID_type2, by = "SampleID2", all.x = TRUE)

### filter for samples where the SampleID1 and SampleID2 are different and use exact matches for the same Subject
manhattan_dist3 <- filter(manhattan_dist2, sampletype1 !=sampletype2)
manhattan_dist4 <- filter(manhattan_dist3, SubjectID1== SubjectID2)
ggplot(manhattan_dist4, aes(x=AspPnaGroup, y = value)) + geom_boxplot() + geom_jitter(size = 0.01) + stat_compare_means()



```



```{r cross-correlation analysis}
# https://microbiome.github.io/OMA/multi-assay_analyses.html#cross-correlation-analysis
mae <- microbiomeDataSets::HintikkaXOData()

library(stringr)

mae[[1]] <- mae[[1]][!is.na(rowData(mae[[1]])$Phylum), ]
# Clean taxonomy data, so that names do not include additional characters
rowData(mae[[1]]) <- DataFrame(apply(rowData(mae[[1]]), 2, 
                                     str_remove, pattern = "._[0-9]__"))
# Microbiome data
mae[[1]]
mae[[2]]
mae[[3]]

# Agglomerate microbiome data at family level
mae[[1]] <- agglomerateByPrevalence(mae[[1]], rank = "Family")
# Does log10 transform for microbiome data
mae[[1]] <- transformSamples(mae[[1]], method = "log10", pseudocount = 1)

rownames(mae[[1]]) <- getTaxonomyLabels(mae[[1]])

# Cross correlates data sets
correlations <- testExperimentCrossCorrelation(mae, 
                                               experiment1 = 1,
                                               experiment2 = 2,
                                               abund_values1 = "log10", 
                                               abund_values2 = "nmr",
                                               method = "spearman", 
                                               p_adj_threshold = NULL,
                                               cor_threshold = NULL,
                                               # Remove when mia is fixed
                                               mode = "matrix",
                                               sort = TRUE,
                                               show_warnings = FALSE)


plot <- Heatmap(correlations$cor,
                # Print values to cells
                cell_fun = function(j, i, x, y, width, height, fill) {
                    # If the p-value is under threshold
                    if( !is.na(correlations$p_adj[i, j]) & correlations$p_adj[i, j] < 0.05 ){
                        # Print "X"
                        grid.text(sprintf("%s", "X"), x, y, gp = gpar(fontsize = 8, col = "black"))
                        }
                    },
                heatmap_legend_param = list(title = "", legend_height = unit(5, "cm"))
                )
plot


```



```{r cross-correlation }



    ##################################
    # cross-correlation
    ################################
    ################### .calculate_correlation_for_numeric_values ##################
    # This function calculates correlation between feature-pair. If significance test
    # is specified, then it uses cor.test(), if not then cor().
    
    # Input: Vector of names that belong to feature pair, and assays.
    # Output: Correlation value or list that includes correlation value and p-value.
    #' @importFrom stats cor.test cor 
    calculate_correlation_for_numeric_values <- function(feature_pair, test_significance, 
                                                         assay1, assay2, method, show_warnings,
                                                         ...){
      # Get features
      feature1 <- assay1[ , feature_pair[1]]
      feature2 <- assay2[ , feature_pair[2]]
      # Whether to test significance
      if( test_significance ){
        # Calculate correlation
        # If user does not want warnings, 
        # suppress warnings that might occur when calculating correlaitons (NAs...)
        # or p-values (ties, and exact p-values cannot be calculated...)
        if( show_warnings ){
          temp <- cor.test(feature1,
                           feature2, 
                           method = method,
                           use = "pairwise.complete.obs")
        } else {
          temp <- suppressWarnings( cor.test(feature1,
                                             feature2, 
                                             method = method,
                                             use = "pairwise.complete.obs") )
        }
        
        # Take only correlation and p-value
        temp <- c(temp$estimate, temp$p.value)
      } else{
        # Calculate only correlation value
        # If user does not want warnings, 
        # suppress warnings that might occur when there are NAs in the data
        if( show_warnings ){
          temp <- cor(feature1,
                      feature2, 
                      method = method,
                      use = "pairwise.complete.obs")
        } else {
          temp <- suppressWarnings( cor(feature1,
                                        feature2, 
                                        method = method,
                                        use = "pairwise.complete.obs") )
        }
      }
      return(temp)
    }
    
    FUN <- calculate_correlation_for_numeric_values
    
    
    ########################################3
    ##### @@@@@@@@ TRACHEAL
    #######################################
    
    clr_data = as.data.frame(assays(tracheal_se)$clr)
    clr_data$mean = rowMeans(clr_data)
    ordered.clr_data = clr_data[order(clr_data$mean,decreasing = TRUE),]
    
    # write.csv(ordered.clr_data,file = paste0("./results/",followup,"_",stype,'_ordered_taxa.csv'))
    
    top.taxa = ordered.clr_data[1:20,1:dim(clr_data)[2]-1]
    assay1 <- t(top.taxa)
    
    assay2 <- data.matrix(as.data.frame(colData(tracheal_se)[,c("il6","il8","il10", "tnfr1_final", "st2", "fractalkine",
                                                       "rage","ang2","procalcitonin","pentraxin3",
                                                      "spd", "cd14", "LBP", "bdg")]))
    
    # assay1 <- assay1[cluster_indices,]
    # assay2 <- assay2[cluster_indices,]
    
   
    feature_pairs <- expand.grid(colnames(assay1), colnames(assay2))
    # Calculate correlations
    test_significance = TRUE
    method = "pearson"
    correlations_and_p_values <- apply(feature_pairs, 1, 
                                       FUN = FUN, 
                                       test_significance = test_significance, 
                                       assay1 = assay1, 
                                       assay2 = assay2,
                                       method = method,
                                       show_warnings = FALSE)
    
    # otherwise transpose into the same orientation as feature-pairs if it's not a vector
    if( is.vector(correlations_and_p_values) ){
      correlations_and_p_values <- data.frame(correlations_and_p_values)
    } else{
      correlations_and_p_values  <- t(correlations_and_p_values)
    }
    # Give names
    if( ncol(correlations_and_p_values) == 1 ){
      colnames(correlations_and_p_values) <- c("cor")
    }else if( ncol(correlations_and_p_values) == 2 ){
      colnames(correlations_and_p_values) <- c("cor", "pval")
    }
    # Combine feature-pair names with correlation values and p-values
    correlations_and_p_values <- cbind(feature_pairs, correlations_and_p_values)
    # If there are p_values, adjust them
    if( !is.null(correlations_and_p_values$pval) ){
      correlations_and_p_values$p_adj <- 
        p.adjust(correlations_and_p_values$pval,
                 method = "fdr")
    }
    correlation_table = correlations_and_p_values
    
    
    # Converts data to matrix, correlations as values
    mat <- reshape2::acast(correlation_table, Var2 ~ Var1, value.var = "cor")
    
    # Hierarchical clustering, gets the order of taxa 
    taxa_indices <- hclust(as.dist(1 - cor(mat, use="pairwise.complete.obs")))$order
    order_taxa <- colnames(mat)[taxa_indices]
    biomarkers_indices <- hclust(as.dist(1 - cor(t(mat), use="pairwise.complete.obs")))$order
    order_biomarkers <- rownames(mat)[biomarkers_indices]
    # Converts taxa and lipids columns to factor so that they have the desired order
    correlation_table[["Var1"]] <- factor(correlation_table[["Var1"]], levels = order_taxa)
    correlation_table[["Var2"]] <- factor(correlation_table[["Var2"]], levels = order_biomarkers)
    
    # Determines the scaling of colors
    limits <- c(-1, 1)
    breaks <- seq(from = min(limits), to = max(limits), by = 0.2)
    colours <- c("darkblue", "blue", "white", "red", "darkred")
    
    # Which observation have p-value under 0.05? --> creates a subset
    cor_table_sub <- correlation_table[which(correlation_table[["pval"]] < 0.05), ]
    
    
    # Creates a ggplot object
    trachealTaxaBiomarkersHeatmap <- ggplot(correlation_table, aes(x = Var1, y = Var2, fill = cor)) +
      geom_tile() +
      
      scale_fill_gradientn(name = "Correlation", 
                           breaks = breaks, limits = limits, colours = colours) + 
      
      # Adds label to those observations that have p-value under 0.05
      geom_text(data = cor_table_sub, aes(x = Var1, y = Var2, label = "+")) +
      
      theme(text = element_text(size=16),
            axis.text.x = element_text(angle=45, hjust=1, size = 10),
            legend.key.size = unit(1, "cm")) +
      labs(x = "Taxa", y = "Biomarkers")+ 
      ggtitle("ETA taxa and plasma biomarker correlations") 
    ggsave("./results/trachealTaxaBiomarkersHeatmap.jpeg", width = 12, height = 9)
    
    
    ########### repeat analysis for oral
    
        ########################################3
    ##### @@@@@@@@ oral
    #######################################
    
    clr_data = as.data.frame(assays(oral_se)$clr)
    clr_data$mean = rowMeans(clr_data)
    ordered.clr_data = clr_data[order(clr_data$mean,decreasing = TRUE),]
    
    # write.csv(ordered.clr_data,file = paste0("./results/",followup,"_",stype,'_ordered_taxa.csv'))
    
    top.taxa = ordered.clr_data[1:20,1:dim(clr_data)[2]-1]
    assay1 <- t(top.taxa)
    
    assay2 <- data.matrix(as.data.frame(colData(oral_se)[,c("il6","il8","il10", "tnfr1_final", "st2", "fractalkine",
                                                       "rage","ang2","procalcitonin","pentraxin3",
                                                      "spd", "cd14", "LBP", "bdg")]))
    
    # assay1 <- assay1[cluster_indices,]
    # assay2 <- assay2[cluster_indices,]
    
    
    feature_pairs <- expand.grid(colnames(assay1), colnames(assay2))
    # Calculate correlations
    test_significance = TRUE
    method = "pearson"
    correlations_and_p_values <- apply(feature_pairs, 1, 
                                       FUN = FUN, 
                                       test_significance = test_significance, 
                                       assay1 = assay1, 
                                       assay2 = assay2,
                                       method = method,
                                       show_warnings = FALSE)
    
    # otherwise transpose into the same orientation as feature-pairs if it's not a vector
    if( is.vector(correlations_and_p_values) ){
      correlations_and_p_values <- data.frame(correlations_and_p_values)
    } else{
      correlations_and_p_values  <- t(correlations_and_p_values)
    }
    # Give names
    if( ncol(correlations_and_p_values) == 1 ){
      colnames(correlations_and_p_values) <- c("cor")
    }else if( ncol(correlations_and_p_values) == 2 ){
      colnames(correlations_and_p_values) <- c("cor", "pval")
    }
    # Combine feature-pair names with correlation values and p-values
    correlations_and_p_values <- cbind(feature_pairs, correlations_and_p_values)
    # If there are p_values, adjust them
    if( !is.null(correlations_and_p_values$pval) ){
      correlations_and_p_values$p_adj <- 
        p.adjust(correlations_and_p_values$pval,
                 method = "fdr")
    }
    correlation_table = correlations_and_p_values
    
    
    # Converts data to matrix, correlations as values
    mat <- reshape2::acast(correlation_table, Var2 ~ Var1, value.var = "cor")
    
    # Hierarchical clustering, gets the order of taxa 
    taxa_indices <- hclust(as.dist(1 - cor(mat, use="pairwise.complete.obs")))$order
    order_taxa <- colnames(mat)[taxa_indices]
    biomarkers_indices <- hclust(as.dist(1 - cor(t(mat), use="pairwise.complete.obs")))$order
    order_biomarkers <- rownames(mat)[biomarkers_indices]
    # Converts taxa and lipids columns to factor so that they have the desired order
    correlation_table[["Var1"]] <- factor(correlation_table[["Var1"]], levels = order_taxa)
    correlation_table[["Var2"]] <- factor(correlation_table[["Var2"]], levels = order_biomarkers)
    
    # Determines the scaling of colors
    limits <- c(-1, 1)
    breaks <- seq(from = min(limits), to = max(limits), by = 0.2)
    colours <- c("darkblue", "blue", "white", "red", "darkred")
    
    # Which observation have p-value under 0.05? --> creates a subset
    cor_table_sub <- correlation_table[which(correlation_table[["pval"]] < 0.05), ]
    
    
    # Creates a ggplot object
    oralTaxaBiomarkersHeatmap <- ggplot(correlation_table, aes(x = Var1, y = Var2, fill = cor)) +
      geom_tile() +
      
      scale_fill_gradientn(name = "Correlation", 
                           breaks = breaks, limits = limits, colours = colours) + 
      
      # Adds label to those observations that have p-value under 0.05
      geom_text(data = cor_table_sub, aes(x = Var1, y = Var2, label = "+")) +
      
      theme(text = element_text(size=16),
            axis.text.x = element_text(angle=45, hjust=1, size = 10),
            legend.key.size = unit(1, "cm")) +
      labs(x = "Taxa", y = "Biomarkers")+ 
      ggtitle("OPS taxa and plasma biomarker correlations") 
    ggsave("./results/oralTaxaBiomarkersHeatmap.jpeg", width = 12, height = 9)
    
    
    #### take a summary figure
    TaxaBiomarkersHeatMap <- ggarrange(oralTaxaBiomarkersHeatmap, trachealTaxaBiomarkersHeatmap, ncol = 1, nrow = 2)
    ggsave("./results/TaxaBiomarkersHeatMap.jpeg", width = 12, height = 18)
    
    
```


```{r biomarkers}

###### look at systemic and ETA biomarker differences by aspiration groups first
# exploratory analyses
### systemic
AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = procalcitonin)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = pentraxin3)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = rage)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = ang2)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = st2)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = il8)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = il10)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = il6)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = tnfr1_final)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = fractalkine)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = spd)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = LBP)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = cd14)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = nDNA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = mtDNA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = human_MPM)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = TotalMicrobial_mcfDNA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = Bacteria_mcfDNA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = Fungi_mcfDNA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()


######### systemic by tracheal clusters
tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = procalcitonin)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = pentraxin3)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = rage)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = ang2)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = st2)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = il8)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = il10)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = il6)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = tnfr1_final)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = fractalkine)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = spd)) + geom_boxplot() + geom_jitter() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% filter(LBP>150000) %>% ggplot(aes(x=cluster, y = LBP)) + geom_boxplot() + geom_jitter() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = cd14)) + geom_boxplot() + geom_jitter() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = nDNA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = mtDNA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = tnfr1_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = TotalMicrobial_mcfDNA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = Bacteria_mcfDNA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = AH50)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

oral_pneumonia_df %>% ggplot(aes(x=cluster, y = WBC)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()


oral_pneumonia_df %>% ggplot(aes(x=cluster, y = rage_eta_2016)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

oral_pneumonia_df$NeutElas_eta

names(tracheal_pneumonia_df)


# LRT biomarkers
AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = procalcitonin_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = pentraxin3_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = rage_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = ang2_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = st2_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = il8_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = il10_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = il6_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = tnfr1_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = fractalkine_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = totProt_ETA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = IgM_ETA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = MPO_eta)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = TSP1_eta)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = NeutElas_eta)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = HumanReads_nanopore)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

AspMetadata_basl_alir %>% ggplot(aes(x=AspPnaGroup, y = BacterialReads_nanopore)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

########
tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = procalcitonin_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = pentraxin3_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = rage_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = ang2_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = st2_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = il8_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = il10_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = il6_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = tnfr1_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = fractalkine_eta_pradj)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = totProt_ETA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = IgM_ETA)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = MPO_eta)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = TSP1_eta)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = NeutElas_eta)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = HumanReads_nanopore)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()

tracheal_pneumonia_df %>% ggplot(aes(x=cluster, y = BacterialReads_nanopore)) + geom_boxplot() + geom_point() + stat_compare_means() + scale_y_log10()


###### focus on 5 important biomarkers
# procalcitonin
# rage
# SPD
# cd14
# TNFR1

#### graphs for oral and tracheal and by aspiraiton groups. 
  mycomparisons <- list(c("MAsP", "NonMAsP"), c("MAsP", "IntubControl"))
  
procal_AspPnagroup <- ggboxplot(AspMetadata_basl_alir, x = "AspPnaGroup", y = "procalcitonin", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Plasma levels (pg/ml)", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8)) + ggtitle("Procalcitonin")

rage_AspPnagroup <- ggboxplot(AspMetadata_basl_alir, x = "AspPnaGroup", y = "rage", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8)) + ggtitle("RAGE")

tnfr1_AspPnagroup <- ggboxplot(AspMetadata_basl_alir, x = "AspPnaGroup", y = "tnfr1_final", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8)) + ggtitle("TNFR1")

spd_AspPnagroup <- ggboxplot(AspMetadata_basl_alir, x = "AspPnaGroup", y = "spd", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8)) + ggtitle("SPD")

cd14_AspPnagroup <- ggboxplot(AspMetadata_basl_alir, x = "AspPnaGroup", y = "cd14", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8))  + ggtitle("CD14")


biomarkersByAspGroup <- ggarrange(procal_AspPnagroup, rage_AspPnagroup, tnfr1_AspPnagroup, spd_AspPnagroup, cd14_AspPnagroup, ncol = 5, nrow = 1)
ggsave("./results/biomarkersByAspGroup.jpeg", width =  16, height = 4)



############ examine the same biomarkers by oral clusters. 

    mycomparisons <- list(c("LowDiversity", "HighDiversity"))
procal_oralCluster <- ggboxplot(oral_pneumonia_df, x = "cluster", y = "procalcitonin", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#0072B2"), xlab = "", ylab = "Plasma levels (pg/ml)", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8)) + ggtitle("Procalcitonin")

rage_oralCluster <- ggboxplot(oral_pneumonia_df, x = "cluster", y = "rage", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#0072B2"), xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8)) + ggtitle("sRAGE")

tnfr1_oralCluster <- ggboxplot(oral_pneumonia_df, x = "cluster", y = "tnfr1_final", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#0072B2"), xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8)) + ggtitle("sTNFR1")

spd_oralCluster <- ggboxplot(oral_pneumonia_df, x = "cluster", y = "spd", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#0072B2"), xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8)) +  ggtitle("SPD")

cd14_oralCluster <- ggboxplot(oral_pneumonia_df, x = "cluster", y = "cd14", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#0072B2"), xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 8)) + ggtitle("cd14")


biomarkersByOralCluster<- ggarrange(procal_oralCluster, rage_oralCluster, tnfr1_oralCluster, spd_oralCluster, cd14_oralCluster, ncol = 5, nrow = 1)
ggsave("./results/biomarkersByOralCluster.jpeg", width =  16, height = 4)


########################### Biomarkers by Tracheal Clusters

   mycomparisons <- list(c("LowDiversity", "IntermediateDiversity"), c("LowDiversity", "HighDiversity"))

 procal_trachealCluster <- ggboxplot(tracheal_pneumonia_df, x = "cluster", y = "procalcitonin", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00","#999999",  "#0072B2"), xlab = "", ylab = "Plasma levels (pg/ml)", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 6))

rage_trachealCluster <- ggboxplot(tracheal_pneumonia_df, x = "cluster", y = "rage", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00","#999999",  "#0072B2"), xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 6))

tnfr1_trachealCluster <- ggboxplot(tracheal_pneumonia_df, x = "cluster", y = "tnfr1_final", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00","#999999",  "#0072B2"), xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 6))

spd_trachealCluster <- ggboxplot(tracheal_pneumonia_df, x = "cluster", y = "spd", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00","#999999",  "#0072B2"), xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 6))

cd14_trachealCluster <- ggboxplot(tracheal_pneumonia_df, x = "cluster", y = "cd14", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00","#999999",  "#0072B2"), xlab = "", ylab = "", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10() + theme(axis.text.x = element_text(size = 6))

# of note, bdg has a signal.

biomarkersBytrachealCluster<- ggarrange(procal_trachealCluster, rage_trachealCluster, tnfr1_trachealCluster, spd_trachealCluster, cd14_trachealCluster, ncol = 5, nrow = 1)
ggsave("./results/biomarkersBytrachealCluster.jpeg", width =  14, height = 4)


biomarkers_all <- ggarrange(biomarkersByAspGroup, biomarkersByOralCluster, biomarkersBytrachealCluster, ncol = 1, nrow = 3)

ggsave("./results/biomarkers_all.jpeg", width =  16, height = 12)


biomarkers_byCluster <- ggarrange(biomarkersByOralCluster, biomarkersBytrachealCluster, ncol = 1, nrow = 2)

ggsave("./results/biomarkers_byCluster.jpeg", width =  16, height = 8)



```



```{r analyses by anaerobic content}

####### perform de novo analyses for the anaerobe counts

bugclass <- read.csv('./data/bugclassif_6.22.22.csv') #import classifications 
bugclass <- bugclass[!is.na(bugclass$O2Req),] # 160 classified bugs. 

######## take the taxatable from the tracheal samples 
# tracheal_rel_abund_assay

# separate column names by taxa
bugclass$Genera

bugclass_separatedtaxa <- bugclass %>% separate(Genera, c("Kingdom", "Phylum", "Class", "Order", 
                               "Family", "Genus"), sep = "\\.")
# use Genus as column names
bugclass_genus <- subset(bugclass_separatedtaxa, select = c(Genus, O2Req, OralorRespiratoryPath))


tracheal_rel_abund_assay_df <- as.data.frame(tracheal_rel_abund_assay)
tracheal_rel_abund_assay_df$Genus <- rownames(tracheal_rel_abund_assay_df)

tracheal_rel_abund_assay_df <- merge(tracheal_rel_abund_assay_df, bugclass_genus, by = "Genus", all.x = TRUE)

table(tracheal_rel_abund_assay_df$O2Req)
##### fix some missing classifications
 tracheal_rel_abund_assay_df$Genus[is.na(tracheal_rel_abund_assay_df$O2Req)]
 tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Cardiobacterium"] <- "Aerobe"
 tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Slackia"] <- "Anaerobe"
  tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Streptobacillus"] <- "Facultative anaerobe"
   tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Tumebacillus"] <- "Aerobe"
  tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Peptostreptococcaceae_ge"] <- "Anaerobe"
  
   tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Tropheryma"] <- "Aerobe"
   
    tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Paracoccus"] <- "Facultative anaerobe"
    
       tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Kocuria"] <- "Aerobe"
       
          tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Cryptobacterium"] <- "Anaerobe"
          
             tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Butyrivibrio"] <- "Anaerobe"
             
              tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Staphylococcaceae_uncl"] <- "Facultative anaerobe"
              
              tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Sphingomonadaceae_uncl"] <- "Facultative anaerobe"
              
               tracheal_rel_abund_assay_df$O2Req[tracheal_rel_abund_assay_df$Genus=="Burkholderia_Caballeronia_Paraburkholderia"] <- "Aerobe"
               
               names(tracheal_rel_abund_assay_df)
               

tracheal_bugsClassified <- subset(tracheal_rel_abund_assay_df, select = c(Genus, O2Req, OralorRespiratoryPath))
#########


table(tracheal_bugsClassified$O2Req)
tracheal_bugsClassified$O2Req[is.na(tracheal_bugsClassified$O2Req)] <- "Unclassifiable"
#### export this to create a table of bugs classified
write.csv(tracheal_bugsClassified, "./data/tracheal_bugsClassified.csv")

###### take the df from tracheal_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
tracheal_rel_abund_assay_df$O2Req[is.na(tracheal_rel_abund_assay_df$O2Req)] <- "Unclassifiable"

trachealbugsbyOxygen <- as.data.frame(t(tracheal_rel_abund_assay_df))
### remove last row of pathogens
trachealbugsbyOxygen <- trachealbugsbyOxygen[1:189, ]

#### take the names of the genera from the classified dataframe to create vectors that will then be used to calculate rowsums. 

anaerobeslist <- tracheal_bugsClassified$Genus[tracheal_bugsClassified$O2Req=="Anaerobe"]
aerobeslist <- tracheal_bugsClassified$Genus[tracheal_bugsClassified$O2Req=="Aerobe"]
facultanaerobeslist <- tracheal_bugsClassified$Genus[tracheal_bugsClassified$O2Req=="Facultative anaerobe"]
Microaerophilelist <- tracheal_bugsClassified$Genus[tracheal_bugsClassified$O2Req=="Microaerophile"]
Unclassifiableslist <- tracheal_bugsClassified$Genus[tracheal_bugsClassified$O2Req=="Unclassifiable"]
Variablelist <- tracheal_bugsClassified$Genus[tracheal_bugsClassified$O2Req=="Variable"]

#### in the trachealbugsbyoxygen, make colnames by the genus variable
colnames(trachealbugsbyOxygen) <- trachealbugsbyOxygen[1,]
#### remove the anaerobe and 02 req rows
trachealbugsbyOxygen <- trachealbugsbyOxygen[2:188, ]

#### calculate rowsums by the list selection criteria. 
# make all values numeric - include some code to maintain the rownames by sample_id with this conversion

trachealbugsbyOxygenSample_id  <-as.data.frame(rownames(trachealbugsbyOxygen))
trachealbugsbyOxygen <- as.data.frame(sapply(trachealbugsbyOxygen,as.numeric))

trachealbugsbyOxygen$x <- rownames(trachealbugsbyOxygen)
trachealbugsbyOxygenSample_id$x <- rownames(trachealbugsbyOxygenSample_id)
trachealbugsbyOxygen <- merge(trachealbugsbyOxygen, trachealbugsbyOxygenSample_id, by = "x")
rownames(trachealbugsbyOxygen) <- trachealbugsbyOxygen$`rownames(trachealbugsbyOxygen)`
trachealbugsbyOxygen$x <- NULL

# calculate row sums by the desired list
# https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums
trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(AnaerobeSum = rowSums(select(., .dots = all_of(anaerobeslist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(AerobeSum = rowSums(select(., .dots = all_of(aerobeslist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(FacultAnaerobeSum = rowSums(select(., .dots = all_of(facultanaerobeslist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(MicroaerophileSum = rowSums(select(., .dots = all_of(Microaerophilelist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(VariableSum = rowSums(select(., .dots = all_of(Variablelist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen <- trachealbugsbyOxygen %>% 
  mutate(UnclassifiableSum = rowSums(select(., .dots = all_of(Unclassifiableslist))))
trachealbugsbyOxygen$AnaerobeSum

trachealbugsbyOxygen$totalO2sum <- trachealbugsbyOxygen$AnaerobeSum + trachealbugsbyOxygen$AerobeSum + trachealbugsbyOxygen$FacultAnaerobeSum  + trachealbugsbyOxygen$MicroaerophileSum  + trachealbugsbyOxygen$VariableSum + trachealbugsbyOxygen$UnclassifiableSum 
###mutate total to 1 and then unclassifiable as difference from total
trachealbugsbyOxygen$totalO2sum <- 1
trachealbugsbyOxygen$UnclassifiableSum <- 1 - (trachealbugsbyOxygen$AnaerobeSum + trachealbugsbyOxygen$AerobeSum + trachealbugsbyOxygen$FacultAnaerobeSum  + trachealbugsbyOxygen$MicroaerophileSum  + trachealbugsbyOxygen$VariableSum )

tracheal_bugsClassifiedSum <- subset(trachealbugsbyOxygen, select = c(AnaerobeSum, AerobeSum, FacultAnaerobeSum,MicroaerophileSum, VariableSum , UnclassifiableSum))

### incorporate with the tracheal_df 
tracheal_df_sum <- merge(tracheal_df,tracheal_bugsClassifiedSum, by = 'row.names', all = TRUE)
#### plot of comparisons of abundance by clinical groups. 

# Pairwise comparisons.
mycomparisons <- list(c("MAsP", "NonMAsP"), c("MAsP", "IntubControl"), c("MAsP", "HealthyControls"))
anaerobestracheal <- ggboxplot(tracheal_df_sum, x = "AspPnaGroup", y = "AnaerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Obligative Anaerobe Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

aerobestracheal <- ggboxplot(tracheal_df_sum, x = "AspPnaGroup", y = "AerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Aerobe Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

facultativeanaerobestracheal <- ggboxplot(tracheal_df_sum, x = "AspPnaGroup", y = "FacultAnaerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Facultative Anaerobe Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

microaerophilesstracheal <- ggboxplot(tracheal_df_sum, x = "AspPnaGroup", y = "MicroaerophileSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Microaerophile Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) + ylim(0,1)

variablesstracheal <- ggboxplot(tracheal_df_sum, x = "AspPnaGroup", y = "VariableSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Variable 02 requirement Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) + ylim(0,1)

unclassifedtracheal <- ggboxplot(tracheal_df_sum, x = "AspPnaGroup", y = "UnclassifiableSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Unclassifiable 02 requirement Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) + ylim(0,1)

comparAbundO2reqTracheal <- ggarrange(anaerobestracheal, aerobestracheal, facultativeanaerobestracheal, microaerophilesstracheal, variablesstracheal, unclassifedtracheal, ncol = 3, nrow = 2)
ggsave("./results/comparAbundO2reqTracheal.jpeg", width = 17, height = 10)

##### save the 3 more common for tracheal which we will use in a 2x3 graph with OPS

comparAbundO2reqTracheal_3common <- ggarrange(anaerobestracheal, aerobestracheal, facultativeanaerobestracheal, ncol = 3, nrow = 1)
ggsave("./results/comparAbundO2reqTracheal_3common.jpeg", width = 17, height =5)
#############3


##### create revised taxa barplot for these categories. 

# O2 req with bar per ID
dfo <- subset(tracheal_df_sum, select = c(AspPnaGroup, AnaerobeSum, AerobeSum, FacultAnaerobeSum, MicroaerophileSum, VariableSum, UnclassifiableSum))
dfo <- dfo %>% arrange(AnaerobeSum) %>%
  arrange(factor(AspPnaGroup, levels = c('MAsP','NonMAsP','IntubControl', 'HealthyControls'))) #to order aesthetic by increasing anaerobic presence 
dfo$Order <- seq(1, length.out=nrow(dfo), by=1)
o2long <- dfo %>% gather(O2_Requirement, Abundance, -c(AspPnaGroup, Order))
o2long <- data.frame(o2long)
# o2long$AnaerobeOrder <- as.character(o2long$AnaerobeOrder)
o2long$AspPnaGroup <- factor(o2long$AspPnaGroup, levels = c('MAsP','NonMAsP','IntubControl', 'HealthyControls')) 
o2long$O2_Requirement <-  factor(o2long$O2_Requirement, levels = c("AnaerobeSum", "AerobeSum",
                                                                   "FacultAnaerobeSum", "MicroaerophileSum", "VariableSum", "UnclassifiableSum"))

######### Plot for anaerobes by clinical group
o2plot <- ggplot(o2long, aes(x=Order, y=Abundance, fill=O2_Requirement)) +
  geom_bar(stat="identity") + 
  geom_col(width=1) +
  scale_fill_viridis(discrete=TRUE, direction=1, labels = c("Anaerobes", "Aerobes", "Facultative Anaerobes", 
                                                             "Microaerophiles", "Variable", "Unclassifiable")) +
  facet_grid(~AspPnaGroup, scale="free_x", space = "free_x", shrink=TRUE, drop=TRUE) +
  ggtitle("Oxygen requirements of component bacteria by clinical group - ETA") +
  theme(axis.line=element_blank(), axis.text.x=element_blank(),
        axis.ticks=element_blank(),
        axis.title.x=element_blank(), axis.text=element_text(size=12),
        axis.title=element_text(size=18), plot.title=element_text(size=18),
        panel.background=element_blank(), panel.border=element_blank(),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(), plot.background=element_blank(),
        legend.title=element_blank(), legend.text=element_text(size=12),
        strip.text.x = element_text(size=12), legend.position = "bottom")
ggsave("./results/o2plot.jpeg", width = 10, height = 6)


############## incorporate the taxa classifications to the tracheal_pneumonia_df to reproduce such figures by clusters. 

### incorporate with the tracheal_df 
rownames(tracheal_pneumonia_df) <- tracheal_pneumonia_df$sample_id
tracheal_pneumonia_df_sum <- merge(tracheal_pneumonia_df,tracheal_bugsClassifiedSum, by = 'row.names', all.x = TRUE)
tracheal_pneumonia_df_sum$AnaerobeSum

### remove duplicated columns
# tracheal_pneumonia_df_sum <- tracheal_pneumonia_df_sum[!duplicated(as.list(tracheal_pneumonia_df))]

#### plot of comparisons of abundance by clusters groups. 

# Pairwise comparisons.
mycomparisons <- list(c("LowDiversity", "IntermediateDiversity"), c("LowDiversity", "HighDiversity"))
anaerobestrachealcluster <- ggboxplot(tracheal_pneumonia_df_sum, x = "cluster", y = "AnaerobeSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Obligative Anaerobe Abundance ETA", legend = "none" ) +    stat_compare_means(comparisons = mycomparisons) 

aerobestrachealcluster <- ggboxplot(tracheal_pneumonia_df_sum, x = "cluster", y = "AerobeSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Aerobe Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

facultativeanaerobestrachealcluster <- ggboxplot(tracheal_pneumonia_df_sum, x = "cluster", y = "FacultAnaerobeSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Facultative Anaerobe Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

microaerophilesstrachealcluster <- ggboxplot(tracheal_pneumonia_df_sum, x = "cluster", y = "MicroaerophileSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Microaerophile Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons)  + ylim(0,1)

variablesstrachealcluster <- ggboxplot(tracheal_pneumonia_df_sum, x = "cluster", y = "VariableSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Variable 02 requirement Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons)  + ylim(0,1)

unclassifedtrachealcluster <- ggboxplot(tracheal_pneumonia_df_sum, x = "cluster", y = "UnclassifiableSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Unclassifiable 02 requirement Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons)  + ylim(0,1)

comparAbundO2reqTrachealcluster <- ggarrange(anaerobestrachealcluster, aerobestrachealcluster, facultativeanaerobestrachealcluster, microaerophilesstrachealcluster, variablesstrachealcluster, unclassifedtrachealcluster, ncol = 3, nrow = 2)
ggsave("./results/comparAbundO2reqTrachealcluster.jpeg", width = 17, height = 10)

#############
### save the 3 more common

comparAbundO2reqTrachealcluster_3common <- ggarrange(anaerobestrachealcluster, aerobestrachealcluster, facultativeanaerobestrachealcluster,  ncol = 3, nrow = 1)
ggsave("./results/comparAbundO2reqTrachealcluster_3common.jpeg", width = 17, height = 5)

##### create revised taxa barplot for these categories. 


######## abundance plot by clusters
# O2 req with bar per ID
trachealclusters_dfo <- subset(tracheal_pneumonia_df_sum, select = c(cluster, AnaerobeSum, AerobeSum, FacultAnaerobeSum, MicroaerophileSum, VariableSum, UnclassifiableSum))
trachealclusters_dfo <- trachealclusters_dfo %>% arrange(AnaerobeSum) %>%
  arrange(factor(cluster, levels = c('LowDiversity','IntermediateDiversity','HighDiversity'))) #to order aesthetic by increasing anaerobic presence 
trachealclusters_dfo$Order <- seq(1, length.out=nrow(trachealclusters_dfo), by=1)
trachealcluster_o2long <- trachealclusters_dfo %>% gather(O2_Requirement, Abundance, -c(cluster, Order))
trachealcluster_o2long <- data.frame(trachealcluster_o2long)
# o2long$AnaerobeOrder <- as.character(o2long$AnaerobeOrder)
trachealcluster_o2long$cluster <- factor(trachealcluster_o2long$cluster, levels = c('LowDiversity','IntermediateDiversity','HighDiversity')) 
trachealcluster_o2long$O2_Requirement <-  factor(trachealcluster_o2long$O2_Requirement, levels = c("AnaerobeSum", "AerobeSum",
                                                                   "FacultAnaerobeSum", "MicroaerophileSum", "VariableSum", "UnclassifiableSum"))

######### Plot for anaerobes by clinical group
trachealcluster_o2plot <- ggplot(trachealcluster_o2long, aes(x=Order, y=Abundance, fill=O2_Requirement)) +
  geom_bar(stat="identity") + 
  geom_col(width=1) +
  scale_fill_viridis(discrete=TRUE, direction=1, labels = c("Anaerobes", "Aerobes", "Facultative Anaerobes", 
                                                             "Microaerophiles", "Variable", "Unclassifiable")) +
  facet_grid(~cluster, scale="free_x", space = "free_x", shrink=TRUE, drop=TRUE) +
  ggtitle("Oxygen requirements of component bacteria by DMM clusters - ETA") +
  theme(axis.line=element_blank(), axis.text.x=element_blank(),
        axis.ticks=element_blank(),
        axis.title.x=element_blank(), axis.text=element_text(size=12),
        axis.title=element_text(size=18), plot.title=element_text(size=18),
        panel.background=element_blank(), panel.border=element_blank(),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(), plot.background=element_blank(),
        legend.title=element_blank(), legend.text=element_text(size=12),
        strip.text.x = element_text(size=12), legend.position = "bottom")
ggsave("./results/trachealcluster_o2plot.jpeg", width = 10, height = 6)



################# similar graph for the OPS samples. 
################################################################################################
oral_rel_abund_assay_df <- as.data.frame(oral_rel_abund_assay)

oral_rel_abund_assay_df$Genus <- rownames(oral_rel_abund_assay_df)

oral_rel_abund_assay_df <- merge(oral_rel_abund_assay_df, bugclass_genus, by = "Genus", all.x = TRUE)

table(oral_rel_abund_assay_df$O2Req)
##### fix some missing classifications
 oral_rel_abund_assay_df$Genus[is.na(oral_rel_abund_assay_df$O2Req)]
 oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Cardiobacterium"] <- "Aerobe"
 oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Slackia"] <- "Anaerobe"
  oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Streptobacillus"] <- "Facultative anaerobe"
   oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Tumebacillus"] <- "Aerobe"
  oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Peptostreptococcaceae_ge"] <- "Anaerobe"
  
   oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Tropheryma"] <- "Aerobe"
   
    oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Paracoccus"] <- "Facultative anaerobe"
    
       oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Kocuria"] <- "Aerobe"
       
          oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Cryptobacterium"] <- "Anaerobe"
          
             oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Butyrivibrio"] <- "Anaerobe"
             
              oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Staphylococcaceae_uncl"] <- "Facultative anaerobe"
              
              oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Sphingomonadaceae_uncl"] <- "Facultative anaerobe"
              
               oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Burkholderia_Caballeronia_Paraburkholderia"] <- "Aerobe"
               oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Eggerthia"] <- "Anaerobe"
                              oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Peptoanaerobacter"] <- "Anaerobe"
  oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Prevotellaceae_Uncltrd"] <- "Anaerobe"
               
           oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Comamonas"] <- "Aerobe"
           oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Bilophila"] <- "Anaerobe"
           
           oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Mobiluncus"] <- "Anaerobe"
           oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Moryella"] <- "Anaerobe"
           oral_rel_abund_assay_df$O2Req[oral_rel_abund_assay_df$Genus=="Burkholderiales_uncl"] <- "Aerobe"
           
           
           names(oral_rel_abund_assay_df)
               

           
oral_bugsClassified <- subset(oral_rel_abund_assay_df, select = c(Genus, O2Req, OralorRespiratoryPath))
#########


table(oral_bugsClassified$O2Req)
oral_bugsClassified$O2Req[is.na(oral_bugsClassified$O2Req)] <- "Unclassifiable"
#### export this to create a table of bugs classified
write.csv(oral_bugsClassified, "./data/oral_bugsClassified.csv")

###### take the df from tracheal_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
oral_rel_abund_assay_df$O2Req[is.na(oral_rel_abund_assay_df$O2Req)] <- "Unclassifiable"

oralbugsbyOxygen <- as.data.frame(t(oral_rel_abund_assay_df))
### remove last row of pathogens
oralbugsbyOxygen <- oralbugsbyOxygen[1:158, ]

#### take the names of the genera from the classified dataframe to create vectors that will then be used to calculate rowsums. 

anaerobeslist <- oral_bugsClassified$Genus[oral_bugsClassified$O2Req=="Anaerobe"]
aerobeslist <- oral_bugsClassified$Genus[oral_bugsClassified$O2Req=="Aerobe"]
facultanaerobeslist <- oral_bugsClassified$Genus[oral_bugsClassified$O2Req=="Facultative anaerobe"]
Microaerophilelist <- oral_bugsClassified$Genus[oral_bugsClassified$O2Req=="Microaerophile"]
Unclassifiableslist <- oral_bugsClassified$Genus[oral_bugsClassified$O2Req=="Unclassifiable"]
Variablelist <- oral_bugsClassified$Genus[oral_bugsClassified$O2Req=="Variable"]

#### in the oralbugsbyoxygen, make colnames by the genus variable
colnames(oralbugsbyOxygen) <- oralbugsbyOxygen[1,]
#### remove the anaerobe and 02 req rows
oralbugsbyOxygen <- oralbugsbyOxygen[2:157, ]

#### calculate rowsums by the list selection criteria. 
# make all values numeric - include some code to maintain the rownames by sample_id with this conversion

oralbugsbyOxygenSample_id  <-as.data.frame(rownames(oralbugsbyOxygen))
oralbugsbyOxygen <- as.data.frame(sapply(oralbugsbyOxygen,as.numeric))

oralbugsbyOxygen$x <- rownames(oralbugsbyOxygen)
oralbugsbyOxygenSample_id$x <- rownames(oralbugsbyOxygenSample_id)
oralbugsbyOxygen <- merge(oralbugsbyOxygen, oralbugsbyOxygenSample_id, by = "x")
rownames(oralbugsbyOxygen) <- oralbugsbyOxygen$`rownames(oralbugsbyOxygen)`
oralbugsbyOxygen$x <- NULL

# calculate row sums by the desired list
# https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums
oralbugsbyOxygen <- oralbugsbyOxygen %>% 
  mutate(AnaerobeSum = rowSums(select(., .dots = all_of(anaerobeslist))))
oralbugsbyOxygen$AnaerobeSum

oralbugsbyOxygen <- oralbugsbyOxygen %>% 
  mutate(AerobeSum = rowSums(select(., .dots = all_of(aerobeslist))))
oralbugsbyOxygen$AnaerobeSum

oralbugsbyOxygen <- oralbugsbyOxygen %>% 
  mutate(FacultAnaerobeSum = rowSums(select(., .dots = all_of(facultanaerobeslist))))
oralbugsbyOxygen$AnaerobeSum

oralbugsbyOxygen <- oralbugsbyOxygen %>% 
  mutate(MicroaerophileSum = rowSums(select(., .dots = all_of(Microaerophilelist))))
oralbugsbyOxygen$AnaerobeSum

oralbugsbyOxygen <- oralbugsbyOxygen %>% 
  mutate(VariableSum = rowSums(select(., .dots = all_of(Variablelist))))
oralbugsbyOxygen$AnaerobeSum

oralbugsbyOxygen <- oralbugsbyOxygen %>% 
  mutate(UnclassifiableSum = rowSums(select(., .dots = all_of(Unclassifiableslist))))
oralbugsbyOxygen$AnaerobeSum

oralbugsbyOxygen$totalO2sum <- oralbugsbyOxygen$AnaerobeSum + oralbugsbyOxygen$AerobeSum + oralbugsbyOxygen$FacultAnaerobeSum  + oralbugsbyOxygen$MicroaerophileSum  + oralbugsbyOxygen$VariableSum + oralbugsbyOxygen$UnclassifiableSum 
###mutate total to 1 and then unclassifiable as difference from total
oralbugsbyOxygen$totalO2sum <- 1
oralbugsbyOxygen$UnclassifiableSum <- 1 - (oralbugsbyOxygen$AnaerobeSum + oralbugsbyOxygen$AerobeSum + oralbugsbyOxygen$FacultAnaerobeSum  + oralbugsbyOxygen$MicroaerophileSum  + oralbugsbyOxygen$VariableSum )

oral_bugsClassifiedSum <- subset(oralbugsbyOxygen, select = c(AnaerobeSum, AerobeSum, FacultAnaerobeSum,MicroaerophileSum, VariableSum , UnclassifiableSum))


### incorporate with the tracheal_df 
oral_df_sum <- merge(oral_df,oral_bugsClassifiedSum, by = 'row.names', all = TRUE)

#### plot of comparisons of abundance by clinical groups. 

# Pairwise comparisons.
mycomparisons <- list(c("MAsP", "NonMAsP"), c("MAsP", "IntubControl"), c("MAsP", "HealthyControls"))
anaerobesoral <- ggboxplot(oral_df_sum, x = "AspPnaGroup", y = "AnaerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Obligative Anaerobe Abundance OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

aerobesoral <- ggboxplot(oral_df_sum, x = "AspPnaGroup", y = "AerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Aerobe Abundance OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

facultativeanaerobesoral <- ggboxplot(oral_df_sum, x = "AspPnaGroup", y = "FacultAnaerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Facultative Anaerobe Abundance OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

microaerophilessoral <- ggboxplot(oral_df_sum, x = "AspPnaGroup", y = "MicroaerophileSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Microaerophile Abundance OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

variablessoral <- ggboxplot(oral_df_sum, x = "AspPnaGroup", y = "VariableSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Variable 02 requirement Abundance OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

unclassifedoral <- ggboxplot(oral_df_sum, x = "AspPnaGroup", y = "UnclassifiableSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Unclassifiable 02 requirement Abundance OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

comparAbundO2reqoral <- ggarrange(anaerobesoral, aerobesoral, facultativeanaerobesoral, microaerophilessoral, variablessoral, unclassifedoral, ncol = 3, nrow = 2)
ggsave("./results/comparAbundO2reqoral.jpeg", width = 15, height = 10)

#############3

### take the 3 most common
comparAbundO2reqoral_3common <- ggarrange(anaerobesoral, aerobesoral, facultativeanaerobesoral,  ncol = 3, nrow = 1)
ggsave("./results/comparAbundO2reqoral_3common.jpeg", width = 17, height = 5)


#### combine with the oral to make a new figure
comparAbundO2reqOPSTrach_3common <- ggarrange(comparAbundO2reqoral_3common, comparAbundO2reqTracheal_3common,  ncol = 1, nrow = 2)
ggsave("./results/comparAbundO2reqOPSTrach_3common.jpeg", width = 17, height = 12)

##### create revised taxa barplot for these categories. 

# O2 req with bar per ID
oral_dfo <- subset(oral_df_sum, select = c(AspPnaGroup, AnaerobeSum, AerobeSum, FacultAnaerobeSum, MicroaerophileSum, VariableSum, UnclassifiableSum))
oral_dfo <- oral_dfo %>% arrange(AnaerobeSum) %>%
  arrange(factor(AspPnaGroup, levels = c('MAsP','NonMAsP','IntubControl', 'HealthyControls'))) #to order aesthetic by increasing anaerobic presence 
oral_dfo$Order <- seq(1, length.out=nrow(oral_dfo), by=1)
oral_o2long <- oral_dfo %>% gather(O2_Requirement, Abundance, -c(AspPnaGroup, Order))
oral_o2long <- data.frame(oral_o2long)
# o2long$AnaerobeOrder <- as.character(o2long$AnaerobeOrder)
oral_o2long$AspPnaGroup <- factor(oral_o2long$AspPnaGroup, levels = c('MAsP','NonMAsP','IntubControl', 'HealthyControls')) 
oral_o2long$O2_Requirement <-  factor(oral_o2long$O2_Requirement, levels = c("AnaerobeSum", "AerobeSum",
                                                                   "FacultAnaerobeSum", "MicroaerophileSum", "VariableSum", "UnclassifiableSum"))

######### Plot for anaerobes by clinical group
oral_o2plot <- ggplot(oral_o2long, aes(x=Order, y=Abundance, fill=O2_Requirement)) +
  geom_bar(stat="identity") + 
  geom_col(width=1) +
  scale_fill_viridis(discrete=TRUE, direction=1, labels = c("Anaerobes", "Aerobes", "Facultative Anaerobes", 
                                                             "Microaerophiles", "Variable", "Unclassifiable")) +
  facet_grid(~AspPnaGroup, scale="free_x", space = "free_x", shrink=TRUE, drop=TRUE) +
  ggtitle("Oxygen requirements of component bacteria by clinical group - OPS") +
  theme(axis.line=element_blank(), axis.text.x=element_blank(),
        axis.ticks=element_blank(),
        axis.title.x=element_blank(), axis.text=element_text(size=12),
        axis.title=element_text(size=18), plot.title=element_text(size=18),
        panel.background=element_blank(), panel.border=element_blank(),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(), plot.background=element_blank(),
        legend.title=element_blank(), legend.text=element_text(size=12),
        strip.text.x = element_text(size=12), legend.position = "bottom")
ggsave("./results/oral_o2plot.jpeg", width = 10, height = 6)



############## incorporate the taxa classifications to the oral_pneumonia_df to reproduce such figures by clusters. 

### incorporate with the 
rownames(oral_pneumonia_df) <- oral_pneumonia_df$sample_id
oral_pneumonia_df_sum <- merge(oral_pneumonia_df,oral_bugsClassifiedSum, by = 'row.names', all.x = TRUE)
oral_pneumonia_df_sum$AnaerobeSum

### remove duplicated columns
# oral_pneumonia_df <- oral_pneumonia_df[!duplicated(as.list(oral_pneumonia_df))]

#### plot of comparisons of abundance by clusters groups. 

# Pairwise comparisons.

anaerobesoralcluster <- ggboxplot(oral_pneumonia_df_sum, x = "cluster", y = "AnaerobeSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00",  "#0072B2"), xlab = "", ylab = "Obligative Anaerobe Abundance OPS", legend = "none" ) + 
   stat_compare_means() 

aerobesoralcluster <- ggboxplot(oral_pneumonia_df_sum, x = "cluster", y = "AerobeSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00",  "#0072B2"), xlab = "", ylab = "Aerobe Abundance OPS", legend = "none" ) + 
   stat_compare_means() 

facultativeanaerobesoralcluster <- ggboxplot(oral_pneumonia_df_sum, x = "cluster", y = "FacultAnaerobeSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#0072B2"), xlab = "", ylab = "Facultative Anaerobe Abundance OPS", legend = "none" ) + 
   stat_compare_means() 

microaerophilessoralcluster <- ggboxplot(oral_pneumonia_df_sum, x = "cluster", y = "MicroaerophileSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00",  "#0072B2"), xlab = "", ylab = "Microaerophile Abundance OPS", legend = "none" ) + 
   stat_compare_means() 

variablessoralcluster <- ggboxplot(oral_pneumonia_df_sum, x = "cluster", y = "VariableSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#0072B2"), xlab = "", ylab = "Variable 02 requirement Abundance OPS", legend = "none" ) + 
   stat_compare_means() 

unclassifedoralcluster <- ggboxplot(oral_pneumonia_df_sum, x = "cluster", y = "UnclassifiableSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00",  "#0072B2"), xlab = "", ylab = "Unclassifiable 02 requirement Abundance OPS", legend = "none" ) + 
   stat_compare_means() 

names(oral_pneumonia_df_sum)

comparAbundO2reqoralcluster <- ggarrange(anaerobesoralcluster, aerobesoralcluster, facultativeanaerobesoralcluster, microaerophilessoralcluster, variablessoralcluster, unclassifedoralcluster, ncol = 3, nrow = 2)
ggsave("./results/comparAbundO2reqoralcluster.jpeg", width = 15, height = 10)

## 3common
comparAbundO2reqoralcluster_3common <- ggarrange(anaerobesoralcluster, aerobesoralcluster, facultativeanaerobesoralcluster, ncol = 3, nrow = 1)
ggsave("./results/comparAbundO2reqoralcluster_3common.jpeg", width = 15, height = 5)

######## oral  abundance plot by clusters
# O2 req with bar per ID
oralclusters_dfo <- subset(oral_pneumonia_df_sum, select = c(cluster, AnaerobeSum, AerobeSum, FacultAnaerobeSum, MicroaerophileSum, VariableSum, UnclassifiableSum))
oralclusters_dfo <- oralclusters_dfo %>% arrange(AnaerobeSum) %>%
  arrange(factor(cluster, levels = c('LowDiversity','HighDiversity'))) #to order aesthetic by increasing anaerobic presence 
oralclusters_dfo$Order <- seq(1, length.out=nrow(oralclusters_dfo), by=1)
oralcluster_o2long <- oralclusters_dfo %>% gather(O2_Requirement, Abundance, -c(cluster, Order))
oralcluster_o2long <- data.frame(oralcluster_o2long)
# o2long$AnaerobeOrder <- as.character(o2long$AnaerobeOrder)
oralcluster_o2long$cluster <- factor(oralcluster_o2long$cluster, levels = c('LowDiversity','HighDiversity')) 
oralcluster_o2long$O2_Requirement <-  factor(oralcluster_o2long$O2_Requirement, levels = c("AnaerobeSum", "AerobeSum",
                                                                   "FacultAnaerobeSum", "MicroaerophileSum", "VariableSum", "UnclassifiableSum"))

######### Plot for anaerobes by clinical group
oralcluster_o2plot <- ggplot(oralcluster_o2long, aes(x=Order, y=Abundance, fill=O2_Requirement)) +
  geom_bar(stat="identity") + 
  geom_col(width=1) +
  scale_fill_viridis(discrete=TRUE, direction=1, labels = c("Anaerobes", "Aerobes", "Facultative Anaerobes", 
                                                             "Microaerophiles", "Variable", "Unclassifiable")) +
  facet_grid(~cluster, scale="free_x", space = "free_x", shrink=TRUE, drop=TRUE) +
  ggtitle("Oxygen requirements of component bacteria by DMM clusters - OPS") +
  theme(axis.line=element_blank(), axis.text.x=element_blank(),
        axis.ticks=element_blank(),
        axis.title.x=element_blank(), axis.text=element_text(size=12),
        axis.title=element_text(size=18), plot.title=element_text(size=18),
        panel.background=element_blank(), panel.border=element_blank(),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(), plot.background=element_blank(),
        legend.title=element_blank(), legend.text=element_text(size=12),
        strip.text.x = element_text(size=12), legend.position = "bottom")
ggsave("./results/oralcluster_o2plot.jpeg", width = 10, height = 6)



### 3 common categories by cluster
comparAbundO2reqTrachOral_cluster3common <- ggarrange(comparAbundO2reqoralcluster_3common, comparAbundO2reqTrachealcluster_3common,ncol= 1, nrow = 2)
ggsave("./results/comparAbundO2reqTrachOral_cluster3common.jpeg", width = 15, height = 12)



######## perform biomarker analyses

trachealBiom_Anaerobs <- subset(tracheal_pneumonia_df_sum, select = c( AnaerobeSum, AerobeSum, FacultAnaerobeSum, MicroaerophileSum, VariableSum, UnclassifiableSum, il6, il8, il10, tnfr1_final, st2, fractalkine, rage, ang2, procalcitonin, pentraxin3, spd ))

# https://stackoverflow.com/questions/15215848/apply-function-to-every-value-in-an-r-dataframe


# compute correlation matrix
corr <- round(cor(trachealBiom_Anaerobs, use = "pairwise.complete.obs"), 2)
p.mat <- cor_pmat(trachealBiom_Anaerobs)

## adjust # make matrix dataframe and then convert back to matrix
p.mat_df <- as.data.frame(p.mat)
p.mat_df <-apply(p.mat_df,2, p.adjust, method = "BH")

# adjusted for multiple comparisons

trachealBiom_Anaerobs_correlogram <- ggcorrplot(corr,  colors = c("#6D9EC1", "white", "#E46726"), lab = FALSE, hc.order = FALSE, type = "lower", insig = "blank", p.mat = p.mat_df) + ggtitle("ETA taxa by O2 requirement and plasma biomarkers")
ggsave("./results/trachealBiom_Anaerobs_correlogram.jpeg", units = "in", width = 10, height = 8)


# oral anaerobes with biomarkers

oralBiom_Anaerobs <- subset(oral_pneumonia_df_sum, select = c( AnaerobeSum, AerobeSum, FacultAnaerobeSum, MicroaerophileSum, VariableSum, UnclassifiableSum, il6, il8, il10, tnfr1_final, st2, fractalkine, rage, ang2, procalcitonin, pentraxin3, spd ))

# https://stackoverflow.com/questions/15215848/apply-function-to-every-value-in-an-r-dataframe


# compute correlation matrix
corr <- round(cor(oralBiom_Anaerobs, use = "pairwise.complete.obs"), 2)
p.mat <- cor_pmat(oralBiom_Anaerobs)

## adjust # make matrix dataframe and then convert back to matrix
p.mat_df <- as.data.frame(p.mat)
p.mat_df <-apply(p.mat_df,2, p.adjust, method = "BH")

# adjusted for multiple comparisons

oralBiom_Anaerobs_correlogram <- ggcorrplot(corr,  colors = c("#6D9EC1", "white", "#E46726"), lab = FALSE, hc.order = FALSE, type = "lower", insig = "blank", p.mat = p.mat_df) + ggtitle("Oral taxa by O2 requirement and plasma biomarkers")
ggsave("./results/oralBiom_Anaerobs_correlogram.jpeg", units = "in", width = 10, height = 8)





##### obtain distribution of anaerobes by group

vars_cont_oxygenreqabund <-  c("AnaerobeSum", "AerobeSum", "FacultAnaerobeSum", "MicroaerophileSum", "VariableSum", "UnclassifiableSum")

con_table_AspPna_oxygenreqabund <- print(CreateContTable(vars = vars_cont_oxygenreqabund, strata = "AspPnaGroup" , data=tracheal_df_sum), nonnormal = vars_cont_oxygenreqabund, digits  = 3, pDigits = 2)

write.csv(con_table_AspPna_oxygenreqabund, "./results/con_table_AspPna_oxygenreqabund.csv")





####### survival analysis by oxygen requirement. 
oral_pneumonia_df_sum$AnaerobeSum

wilcox.test(oral_pneumonia_df_sum$AnaerobeSum ~ oral_pneumonia_df_sum$Mortality60day)

anaerobesoralmortality60Day <-  ggplot(oral_pneumonia_df_sum, aes(x=Mortality60day, y = AnaerobeSum, fill = Mortality60day)) + geom_boxplot(alpha = 0.5) + geom_beeswarm() + scale_fill_manual(values = c("#000000", "#FF99CC")) + xlab("") + ylab("Obligate Anaerobe Abundance - OPS") + theme_pubr() + theme(legend.position = "none") + stat_compare_means()
                                                                                                                                                    
aerobesoralmortality60Day <-  ggplot(oral_pneumonia_df_sum, aes(x=Mortality60day, y = AerobeSum, fill = Mortality60day)) + geom_boxplot(alpha = 0.5) + geom_beeswarm() + scale_fill_manual(values = c("#000000", "#FF99CC")) + xlab("") + ylab("Aerobe Abundance - OPS") + theme_pubr() + theme(legend.position = "none") + stat_compare_means()

facultanaerobesoralmortality60Day <-  ggplot(oral_pneumonia_df_sum, aes(x=Mortality60day, y = FacultAnaerobeSum, fill = Mortality60day)) + geom_boxplot(alpha = 0.5) + geom_beeswarm() + scale_fill_manual(values = c("#000000", "#FF99CC")) + xlab("") + ylab("Facultative Anaerobe Abundance - OPS") + theme_pubr() + theme(legend.position = "none") + stat_compare_means()

anaerobestrachealmortality60Day <-  ggplot(tracheal_pneumonia_df_sum, aes(x=Mortality60day, y = AnaerobeSum, fill = Mortality60day)) + geom_boxplot(alpha = 0.5) + geom_beeswarm() + scale_fill_manual(values = c("#000000", "#FF99CC")) + xlab("") + ylab("Obligate Anaerobe Abundance - ETA") + theme_pubr() + theme(legend.position = "none") + stat_compare_means()
                                                                                                                                                    
aerobestrachealmortality60Day <-  ggplot(tracheal_pneumonia_df_sum, aes(x=Mortality60day, y = AerobeSum, fill = Mortality60day)) + geom_boxplot(alpha = 0.5) + geom_beeswarm() + scale_fill_manual(values = c("#000000", "#FF99CC")) + xlab("") + ylab("Aerobe Abundance - ETA") + theme_pubr() + theme(legend.position = "none") + stat_compare_means()

facultanaerobestrachealmortality60Day <-  ggplot(tracheal_pneumonia_df_sum, aes(x=Mortality60day, y = FacultAnaerobeSum, fill = Mortality60day)) + geom_boxplot(alpha = 0.5) + geom_beeswarm() + scale_fill_manual(values = c("#000000", "#FF99CC")) + xlab("") + ylab("Facultative Anaerobe Abundance - ETA") + theme_pubr() + theme(legend.position = "none") + stat_compare_means()

oxygenreqabund_byMortality <- ggarrange(anaerobesoralmortality60Day,aerobesoralmortality60Day , facultanaerobesoralmortality60Day, anaerobestrachealmortality60Day, aerobestrachealmortality60Day,facultanaerobestrachealmortality60Day, ncol = 3, nrow = 2 )
ggsave("./results/oxygenreqabund_byMortality.jpeg", width = 12, height = 10)




table(tracheal_pneumonia_df_sum$AspPnaGroup, tracheal_pneumonia_df_sum$AnaerobeSum>0.5)

15/(46+15)



```


```{r analyses by pathogens vs. oral origin}


tracheal_rel_abund_assay_df$OralorRespiratoryPath
tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$OralorRespiratoryPath==""]<- NA
table(tracheal_rel_abund_assay_df$OralorRespiratoryPath)

 tracheal_rel_abund_assay_df$Genus[is.na(tracheal_rel_abund_assay_df$OralorRespiratoryPath)]

##### fix some missing classifications
 tracheal_rel_abund_assay_df$Genus[is.na(tracheal_rel_abund_assay_df$OralorRespiratoryPath)]
 tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Actinomyces"] <- "Oral"
 tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Anaerococcus"] <- "Oral"

   tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Atopobium"] <- "Oral"
  tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Bacteroides"] <- "Oral"
  
   tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Capnocytophaga"] <- "Oral"
   
       tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Corynebacterium"] <- "Oral"
       
          tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Burkholderia_Caballeronia_Paraburkholderia"] <- "Pathogenic"
          
             tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Peptostreptococcus"] <- "Oral"
             
 tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Moraxella"] <- "Pathogenic"       
  tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Yersinia"] <- "Pathogenic"  
   tracheal_rel_abund_assay_df$OralorRespiratoryPath[tracheal_rel_abund_assay_df$Genus=="Veillonellaceae_uncl"] <- "Oral"  
   
               names(tracheal_rel_abund_assay_df)

 tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$OralorRespiratoryPath=="Pathogenic"]


###### take the df from tracheal_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
tracheal_rel_abund_assay_df$OralorRespiratoryPath[is.na(tracheal_rel_abund_assay_df$OralorRespiratoryPath)] <- "Other"

trachealbugsbyOralPath <- as.data.frame(t(tracheal_rel_abund_assay_df))
### remove  row of oxygen requirement
trachealbugsbyOralPath <- trachealbugsbyOralPath[-189, ]

#### take the names of the genera from the classified dataframe to create vectors that will then be used to calculate rowsums. 

oralcommensallist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$OralorRespiratoryPath=="Oral"]
Pathogeniclist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$OralorRespiratoryPath=="Pathogenic"]
Otherlist <- tracheal_rel_abund_assay_df$Genus[tracheal_rel_abund_assay_df$OralorRespiratoryPath=="Other"]


#### in the trachealbugsbyoxygen, make colnames by the genus variable
colnames(trachealbugsbyOralPath) <- trachealbugsbyOralPath[1,]
#### remove the anaerobe and pathogen rows
trachealbugsbyOralPath <- trachealbugsbyOralPath[2:188, ]

#### calculate rowsums by the list selection criteria. 
# make all values numeric - include some code to maintain the rownames by sample_id with this conversion

trachealbugsbyOralPathSample_id  <-as.data.frame(rownames(trachealbugsbyOralPath))
trachealbugsbyOralPath <- as.data.frame(sapply(trachealbugsbyOralPath,as.numeric))

trachealbugsbyOralPath$x <- rownames(trachealbugsbyOralPath)
trachealbugsbyOralPathSample_id$x <- rownames(trachealbugsbyOralPathSample_id)
trachealbugsbyOralPath <- merge(trachealbugsbyOralPath, trachealbugsbyOralPathSample_id, by = "x")
rownames(trachealbugsbyOralPath) <- trachealbugsbyOralPath$`rownames(trachealbugsbyOralPath)`
trachealbugsbyOralPath$x <- NULL

# calculate row sums by the desired list
# https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums
trachealbugsbyOralPath <- trachealbugsbyOralPath %>% 
  mutate(OralCommensalSum = rowSums(select(., .dots = all_of(oralcommensallist))))
trachealbugsbyOralPath$OralCommensalSum

trachealbugsbyOralPath <- trachealbugsbyOralPath %>% 
  mutate(PathogenicSum = rowSums(select(., .dots = all_of(Pathogeniclist))))
trachealbugsbyOralPath$PathogenicSum

trachealbugsbyOralPath <- trachealbugsbyOralPath %>% 
  mutate(OtherSum = rowSums(select(., .dots = all_of(Otherlist))))
trachealbugsbyOralPath$OtherSum



trachealbugsbyOralPath$totalTaxaClassifSum <- trachealbugsbyOralPath$OralCommensalSum + trachealbugsbyOralPath$PathogenicSum + trachealbugsbyOralPath$OtherSum  

###mutate total to 1 and then unclassifiable as difference from total
trachealbugsbyOralPath$totalTaxaClassifSum <- 1
trachealbugsbyOralPath$OtherSum <- 1 - ( trachealbugsbyOralPath$OralCommensalSum + trachealbugsbyOralPath$PathogenicSum  )

trachealbugsbyOralPathSum <- subset(trachealbugsbyOralPath, select = c(OralCommensalSum, PathogenicSum, OtherSum))

### incorporate with the tracheal_df 
tracheal_df_sum$Row.names
rownames(tracheal_df_sum) <- tracheal_df_sum$Row.names
rownames(trachealbugsbyOralPathSum)

tracheal_df_sum <- merge(tracheal_df_sum,trachealbugsbyOralPathSum, by = 'row.names', all = TRUE)    
tracheal_df_sum <- tracheal_df_sum[, -1]
               
# remove duplicate columns
# tracheal_df_sum <- tracheal_df_sum[!duplicated(as.list(tracheal_df_sum))]

      
#### plot of comparisons of abundance by clinical groups. 

# Pairwise comparisons.
mycomparisons <- list(c("MAsP", "NonMAsP"), c("MAsP", "IntubControl"), c("MAsP", "HealthyControls"))
oralcommensalsETA <- ggboxplot(tracheal_df_sum, x = "AspPnaGroup", y = "OralCommensalSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Oral Commensal Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

pathogensETA <- ggboxplot(tracheal_df_sum, x = "AspPnaGroup", y = "PathogenicSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Pathogens Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

othertaxaETA <- ggboxplot(tracheal_df_sum, x = "AspPnaGroup", y = "OtherSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Other Bacteria Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

comparAbundOralCommensvsPathogens_ETAgroups <- ggarrange(oralcommensalsETA, pathogensETA, othertaxaETA, ncol = 3, nrow = 1)
ggsave("./results/comparAbundOralCommensvsPathogens_ETAgroups.jpeg", width = 17, height = 5)

#############3


##### create revised taxa barplot for these categories. 

# O2 req with bar per ID
dforalcommensals <- subset(tracheal_df_sum, select = c(AspPnaGroup, OralCommensalSum, PathogenicSum, OtherSum))
dforalcommensals <- dforalcommensals %>% arrange(PathogenicSum) %>%
  arrange(factor(AspPnaGroup, levels = c('MAsP','NonMAsP','IntubControl', 'HealthyControls'))) #to order aesthetic by increasing anaerobic presence 
dforalcommensals$Order <- seq(1, length.out=nrow(dforalcommensals), by=1)
oralcommensalslong <- dforalcommensals %>% gather(TaxaCategory, Abundance, -c(AspPnaGroup, Order))
oralcommensalslong <- data.frame(oralcommensalslong)
# o2long$AnaerobeOrder <- as.character(o2long$AnaerobeOrder)
oralcommensalslong$AspPnaGroup <- factor(oralcommensalslong$AspPnaGroup, levels = c('MAsP','NonMAsP','IntubControl', 'HealthyControls')) 
oralcommensalslong$TaxaCategory <-  factor(oralcommensalslong$TaxaCategory, levels = c("≈", "PathogenicSum",
                                                                   "OtherSum"))

######### Plot for anaerobes by clinical group
oralcommensalplot <- ggplot(oralcommensalslong, aes(x=Order, y=Abundance, fill=TaxaCategory)) +
  geom_bar(stat="identity") + 
  geom_col(width=1) +
  scale_fill_viridis(discrete=TRUE,   option = "C", direction=1, labels = c("OralCommensals", "Pathogenic", "OtherBacteria")) +
  facet_grid(~AspPnaGroup, scale="free_x", space = "free_x", shrink=TRUE, drop=TRUE) +
  ggtitle("Taxa categories by clinical group - ETA") +
  theme(axis.line=element_blank(), axis.text.x=element_blank(),
        axis.ticks=element_blank(),
        axis.title.x=element_blank(), axis.text=element_text(size=12),
        axis.title=element_text(size=18), plot.title=element_text(size=18),
        panel.background=element_blank(), panel.border=element_blank(),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(), plot.background=element_blank(),
        legend.title=element_blank(), legend.text=element_text(size=12),
        strip.text.x = element_text(size=12), legend.position = "bottom")
ggsave("./results/oralcommensalplot.jpeg", width = 10, height = 6)


############## incorporate the taxa classifications to the tracheal_pneumonia_df to reproduce such figures by clusters. 
tracheal_pneumonia_df_sum <- merge(tracheal_pneumonia_df,trachealbugsbyOralPathSum, by = 'row.names', all.x = TRUE)

#### plot of comparisons of abundance by clusters groups. 

# Pairwise comparisons.
mycomparisons <- list(c("LowDiversity", "IntermediateDiversity"), c("LowDiversity", "HighDiversity"))
oralcommensalstrachealcluster <- ggboxplot(tracheal_pneumonia_df_sum, x = "cluster", y = "OralCommensalSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Oral Commensal Abundance ETA", legend = "none" ) +    stat_compare_means(comparisons = mycomparisons) 

pathogenstrachealcluster <- ggboxplot(tracheal_pneumonia_df_sum, x = "cluster", y = "PathogenicSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Pathogens Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 

othertaxatrachealcluster <- ggboxplot(tracheal_pneumonia_df_sum, x = "cluster", y = "OtherSum", 
                   color = "cluster",add = "jitter" , palette = c("#D55E00", "#999999", "#0072B2"), xlab = "", ylab = "Other Bacteria Abundance ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 


comparAbundOralCommensvsPathogens_clustersETA<- ggarrange(oralcommensalstrachealcluster, pathogenstrachealcluster, othertaxatrachealcluster, ncol = 3, nrow = 1)
ggsave("./results/comparAbundOralCommensvsPathogens_clustersETA.jpeg", width = 17, height = 5)


# O2 req with bar per ID
dforalcommensals_cluster <- subset(tracheal_pneumonia_df_sum, select = c(cluster, OralCommensalSum, PathogenicSum, OtherSum))
dforalcommensals_cluster <- dforalcommensals_cluster %>% arrange(PathogenicSum) %>%
  arrange(factor(cluster, levels = c('LowDiversity','IntermediateDiversity','HighDiversity'))) #to order aesthetic by increasing pathogens
dforalcommensals_cluster$Order <- seq(1, length.out=nrow(dforalcommensals_cluster), by=1)
oralcommensalslong_cluster <- dforalcommensals_cluster %>% gather(TaxaCategory, Abundance, -c(cluster, Order))
oralcommensalslong_cluster <- data.frame(oralcommensalslong_cluster)
# o2long$AnaerobeOrder <- as.character(o2long$AnaerobeOrder)
oralcommensalslong_cluster$AspPnaGroup <- factor(oralcommensalslong_cluster$cluster, levels = c('LowDiversity','IntermediateDiversity','HighDiversity')) 
oralcommensalslong_cluster$TaxaCategory <-  factor(oralcommensalslong_cluster$TaxaCategory, levels = c("OralCommensalSum", "PathogenicSum",
                                                                   "OtherSum"))

######### Plot for anaerobes by clinical group
oralcommensalplot_cluster <- ggplot(oralcommensalslong_cluster, aes(x=Order, y=Abundance, fill=TaxaCategory)) +
  geom_bar(stat="identity") + 
  geom_col(width=1) +
  scale_fill_viridis(discrete=TRUE,   option = "C", direction=1, labels = c("OralCommensals", "Pathogenic", "OtherBacteria")) +
  facet_grid(~AspPnaGroup, scale="free_x", space = "free_x", shrink=TRUE, drop=TRUE) +
  ggtitle("Taxa categories by ETA clusters") +
  theme(axis.line=element_blank(), axis.text.x=element_blank(),
        axis.ticks=element_blank(),
        axis.title.x=element_blank(), axis.text=element_text(size=12),
        axis.title=element_text(size=18), plot.title=element_text(size=18),
        panel.background=element_blank(), panel.border=element_blank(),
        panel.grid.major=element_blank(),
        panel.grid.minor=element_blank(), plot.background=element_blank(),
        legend.title=element_blank(), legend.text=element_text(size=12),
        strip.text.x = element_text(size=12), legend.position = "bottom")
ggsave("./results/oralcommensalplot_cluster.jpeg", width = 10, height = 6)




trachealBiom_oralvspathogens <- subset(tracheal_pneumonia_df_sum, select = c( OralCommensalSum, PathogenicSum, OtherSum, il6, il8, il10, tnfr1_final, st2, fractalkine, rage, ang2, procalcitonin, pentraxin3, spd ))

# https://stackoverflow.com/questions/15215848/apply-function-to-every-value-in-an-r-dataframe


# compute correlation matrix
corr <- round(cor(trachealBiom_oralvspathogens, use = "pairwise.complete.obs"), 2)
p.mat <- cor_pmat(trachealBiom_oralvspathogens)

## adjust # make matrix dataframe and then convert back to matrix
p.mat_df <- as.data.frame(p.mat)
p.mat_df <-apply(p.mat_df,2, p.adjust, method = "BH")

# adjusted for multiple comparisons

trachealBiom_oralvspathogens_correlogram <- ggcorrplot(corr,  colors = c("#6D9EC1", "white", "#E46726"), lab = FALSE, hc.order = FALSE, type = "lower", insig = "blank", p.mat = p.mat_df) + ggtitle("ETA taxa by clinical classifications and plasma biomarkers")
ggsave("./results/trachealBiom_oralvspathogens_correlogram.jpeg", units = "in", width = 10, height = 8)



##### median of anaerobes of oral origin
median(tracheal_df_sum$AnaerobeSum/tracheal_df_sum$OralCommensalSum, na.rm = TRUE) # 0.4748085
quantile(tracheal_df_sum$AnaerobeSum/tracheal_df_sum$OralCommensalSum, na.rm = TRUE, probs  = c(0.33, 0.66)) # 22.9-66.5
hist(tracheal_df_sum$AnaerobeSum/tracheal_df_sum$OralCommensalSum)




########## ETA categories taxa by mortality
oralcommensalstrachealmortality60Day <-  ggplot(tracheal_pneumonia_df_sum, aes(x=Mortality60day, y = OralCommensalSum, fill = Mortality60day)) + geom_boxplot(alpha = 0.5) + geom_beeswarm() + scale_fill_manual(values = c("#000000", "#FF99CC")) + xlab("") + ylab("Oral Commensal Abundance - ETA") + theme_pubr() + theme(legend.position = "none") + stat_compare_means()
                                                                                                                                                    
pathogensmortality60Day <-  ggplot(tracheal_pneumonia_df_sum, aes(x=Mortality60day, y = PathogenicSum, fill = Mortality60day)) + geom_boxplot(alpha = 0.5) + geom_beeswarm() + scale_fill_manual(values = c("#000000", "#FF99CC")) + xlab("") + ylab("Pathogens Abundance - ETA") + theme_pubr() + theme(legend.position = "none") + stat_compare_means()

othertrachealmortality60Day <-  ggplot(tracheal_pneumonia_df_sum, aes(x=Mortality60day, y = OtherSum, fill = Mortality60day)) + geom_boxplot(alpha = 0.5) + geom_beeswarm() + scale_fill_manual(values = c("#000000", "#FF99CC")) + xlab("") + ylab("Other Bacteria Abundance - ETA") + theme_pubr() + theme(legend.position = "none") + stat_compare_means()

taxacategoriesETAabund_byMortality <- ggarrange(oralcommensalstrachealmortality60Day,pathogensmortality60Day , othertrachealmortality60Day, ncol = 3, nrow = 1 )
ggsave("./results/taxacategoriesETAabund_byMortality.jpeg", width = 12, height = 5)


### combine figure with the anaerobes by survivors

taxaAabund_byMortality <- ggarrange(oxygenreqabund_byMortality, taxacategoriesETAabund_byMortality, ncol=1, nrow = 2, heights = c(1,  0.5))
ggsave("./results/taxaAabund_byMortality.jpeg", width = 12, height = 15)



```





```{r  longitudinal}

#### oral longitudinal
####################
# Read meta data
# data <-  read.table(file = "./data/aspirationPneumonia.csv", sep = ",")
# meta.data <- read.csv(file = "./data/aspirationPneumonia.meta.data.csv")


######################## CREATE SEPARATE DATASETS FOR ORAL AND TRACHEAL INCLUDING THE NORMAL CONTROLS. 
# change the oral wash to oral swab
table(AspMetadata$sampletype, AspMetadata$studydayfollowup, AspMetadata$study_id)

# ensure that metadata and taxa table match for the oral dataset
oralMetadataFollowup <- AspMetadata[which(AspMetadata$study_id=="alir" & AspMetadata$sampletype=="oral_swab" ),] # 238 observations.

# set rownames by sample_id
rownames(oralMetadataFollowup) <- oralMetadataFollowup$sample_id
alirmatchingids_v <- as.character(oralMetadataFollowup$sample_id)
oralTaxatableFollowup <- subset(taxatableAsp, rownames(taxatableAsp) %in% alirmatchingids_v)
#### arrange by rownames. 
oralMetadataFollowup <-oralMetadataFollowup[sort(rownames(oralMetadataFollowup)), ]
oralTaxatableFollowup <-oralTaxatableFollowup[sort(rownames(oralTaxatableFollowup)), ]
# ensure all rows equal
all.equal(rownames(oralMetadataFollowup), rownames(oralTaxatableFollowup))


  ##########################################################
  # quality control: checking counts per sample
  #########################################################
reads.per.sample <- oralMetadataFollowup$total
hist(reads.per.sample, breaks = 100)
samples_with_few_reads = oralMetadataFollowup$sample_id[reads.per.sample<300] # 23
### remove samples with <300 reads from analyses for 16s
oralMetadataFollowup <- subset(oralMetadataFollowup, !rownames(oralMetadataFollowup) %in% samples_with_few_reads) # 215 samples
oralTaxatableFollowup <- subset(oralTaxatableFollowup, !rownames(oralTaxatableFollowup) %in% samples_with_few_reads) # 215 samples
# remain matched
all.equal(rownames(oralMetadataFollowup), rownames(oralTaxatableFollowup))

  # remove total
  oralTaxatableFollowup = oralTaxatableFollowup[,-1]
  
  ####################################################################################
  # Creating a SingleCellExperiment object for the downstream analysis
  ################################################################################
  # separate column names by taxa
  oral_df_followup <- data.frame(x = names(oralTaxatableFollowup))
  taxa <- oral_df_followup %>% separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                               "Family", "Genus"), sep = "\\.")
  # use Genus as column names
  names(oralTaxatableFollowup) <- taxa$Genus
   # create a single cell experiment object
  oral_followup_se <- SingleCellExperiment(assays = list(counts = t(oralTaxatableFollowup)), ### transposed the taxatable (data) #The assays slot contains the experimental data as count matrices. Multiple matrices can be stored the result of assays is actually a list of matrices.
                             colData = oralMetadataFollowup, ### contains the data on samples
                             rowData = taxa) # rowData contains data on the features of the analyzed samples. Of particular interest for the microbiome field this is used to store taxonomic information. of note, the taxa dataframe here includes all 6 levels of the taxonomy
  
  
  ### Transformations
  oral_followup_se <- transformSamples(oral_followup_se, method = "relabundance")
  oral_followup_se <- transformSamples(x = oral_followup_se, abund_values = "counts", method = "clr", 
                         pseudocount = 1, name = "clr")
 #     head(assay(se, "clr"))
  
 

  # get taxa that exceed the given prevalence and detection thresholds
  # detection: relative abundance threshold
  # prevalence: relative population frequencies; at 1% compositional abundance threshold
  oral_followup_se <- subsetByPrevalentTaxa(oral_followup_se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                              prevalence = 5/100)
 
    ##########################################################################################
    # Alpha Diversity - oral
    #########################################################################################
    oral_followup_se <- mia::estimateDiversity(oral_followup_se, 
                                 abund_values = "relabundance", # can either be counts or relabundance
                                 index = "shannon", 
                                 name = "shannon")
    oral_followup_df <- as.data.frame(colData(oral_followup_se)[colData(oral_followup_se)$AspPnaGroup %in% 
                                      c("MAsP", "NonMAsP", "IntubControl"), ])
    oral_followup_df$AspPnaGroup <- factor(oral_followup_df$AspPnaGroup)
    oral_followup_df$AspPnaGroup <- ordered(oral_followup_df$AspPnaGroup, levels = c("MAsP", "NonMAsP", "IntubControl"))
    # this is a complicated way of finding the comparison groups. 
  #   comp <- split(t(combn(levels(df$AspPnaGroup), 2)), 
     #               seq(nrow(t(combn(levels(df$AspPnaGroup), 2)))))
    
    
# Pairwise comparisons.

oral_followup_df$studydayfollowup <- ordered(oral_followup_df$studydayfollowup, levels = c("baseline", "middle", "late"))
mycomparisons <- list(c("baseline", "middle"), c("middle", "late"), c("baseline", "late"))
shannon_followup_oral <- oral_followup_df %>% filter(!is.na(studydayfollowup)) %>% ggboxplot( x = "studydayfollowup", y = "shannon", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Shannon Index - OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons)
ggsave("./results/shannon_followup_oral.jpeg",  width = 10, height = 5)


qpcr_followup_oral <- oral_followup_df %>% filter(!is.na(studydayfollowup)) %>% ggboxplot( x = "studydayfollowup", y = "qPCR_16S_mean", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "16S rRNA gene copies (qpCR) - OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons)+ scale_y_log10()
ggsave("./results/qpcr_followup_oral.jpeg",  width = 10, height = 5)


##### model the impact of antibiotics and time from intubation. 
# shannon model over time
model=lme(shannon~timesamplefromintubation + total_ABx_score_day1_Convex, random=~1|SubjectID, data = oral_followup_df, method = "REML", na.action = na.omit)
anova(model) # significant for time. 
#   numDF denDF   F-value p-value
# (Intercept)                     1   137 1122.5293  <.0001
# timesamplefromintubation        1    73    9.3472  0.0031
fit <- aov(shannon~timesamplefromintubation + total_ABx_score_day1_Convex, data = oral_followup_df)
summary(fit) # significant for time. 

# qpcr model over time
model=lme(qPCR_16S_mean~timesamplefromintubation + total_ABx_score_day1_Convex, random=~1|SubjectID, data = oral_followup_df, method = "REML", na.action = na.omit)
anova(model) # significant for antibiotics 
#                              numDF denDF   F-value p-value
# (Intercept)                     1    94 17.236296  0.0001
# timesamplefromintubation        1    35  2.496297  0.1231
# total_ABx_score_day1_Convex     1    94  4.195511  0.0433
fit <- aov(qPCR_16S_mean~timesamplefromintubation + total_ABx_score_day1_Convex, data = oral_followup_df)
summary(fit) # significant for antibiotics 



    #######################################################################################################
    # Beta Diversity oral
    ######################################################################################################
    oral_followup_rel_abund_assay <- assays(oral_followup_se)$relabundance
    
    # Calculates distances between samples. Because taxa is in
    # columns, it is used to compare different samples. We transpose the
    # assay to get taxa to columns
    oral_followup_manhattan_dist <- vegan::vegdist(t(oral_followup_rel_abund_assay), method = "manhattan")
    # oral_manhattan_mds <- vegan::metaMDS(oral_manhattan_dist)
    oral__followup_manhattan_pcoa <- ecodist::pco(oral_followup_manhattan_dist)
    # Creates a data frame from principal coordinates
    oral_followup_manhattan_pcoa_df <- data.frame(pcoa1 = oral__followup_manhattan_pcoa$vectors[,1], 
                                     pcoa2 = oral__followup_manhattan_pcoa$vectors[,2])
    
    ## merge with metadata to make this analyzable by ggpot as per Schloss youtube video approach
    rownames(oral_followup_manhattan_pcoa_df) <-  rownames(oralMetadataFollowup)
    oral_followup_manhattan_pcoa_df$sample_id <-  rownames(oral_followup_manhattan_pcoa_df) 
    oralMetadataBas_nmds_followup <- merge(oralMetadataFollowup, oral_followup_manhattan_pcoa_df, by = "sample_id", all = TRUE) 
    
#### obtain centroids
centroid_oral_followup <- oralMetadataBas_nmds_followup %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa1mean = mean(pcoa1), pcoa2mean = mean(pcoa2))
    
oralMetadataBas_nmds_followup %>% filter(studydayfollowup=="baseline") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
oralMetadataBas_nmds_followup %>% filter(studydayfollowup=="middle") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
oralMetadataBas_nmds_followup %>% filter(studydayfollowup=="late") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))

oralMetadataBas_nmds_followup <- oralMetadataBas_nmds_followup %>% mutate(pcoa1mean = case_when(studydayfollowup=="baseline" ~ 0.001835966, 
                             studydayfollowup=="middle" ~  -0.002894432,
                             studydayfollowup=="late" ~ -0.004171709
                                                ))



oralMetadataBas_nmds_followup %>% filter(studydayfollowup=="baseline") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
oralMetadataBas_nmds_followup %>% filter(studydayfollowup=="baseline") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
oralMetadataBas_nmds_followup %>% filter(studydayfollowup=="late") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))


oralMetadataBas_nmds_followup <- oralMetadataBas_nmds_followup %>% mutate(pcoa2mean = case_when(studydayfollowup=="baseline" ~ 0.0006935192, 
                             studydayfollowup=="middle" ~  -0.003389736,
                             studydayfollowup=="late" ~ 0.003313538
                                                             ))

oral_followup_pcoa <- 
ggplot(oralMetadataBas_nmds_followup, aes(x=pcoa1, xend = pcoa1mean,y = pcoa2, yend = pcoa2mean, color = studydayfollowup, fill = studydayfollowup)) + 
 stat_ellipse(level = 0.8,  show.legend = FALSE, size = 1.2) +
  geom_point(size=1.5, show.legend = FALSE) +
geom_point(data = oralMetadataBas_nmds_followup, mapping = aes(x=pcoa1mean, y = pcoa2mean), size = 5, show.legend = FALSE) +
geom_segment(size = 0.1, show.legend = FALSE) +
labs(x= "PCoA axis1", y = "PCoA axis 2") + scale_color_manual(name = NULL, breaks = c("baseline", "middle", "late"), values = c("black", "blue", "red")) + theme_classic() +ggtitle("PCoA for OPS samples") + annotate(geom="text", x=0, y=-0.03, label="R2=0.033, p<0.0001",color="red")
#+ theme(legend.key.height = unit(0.5, "cm"), legend.position = c(0.8,0.2), legend.background = element_rect(fill = "NA", color = "gray"))
ggsave("./results/oral_followup_pcoa.jpeg", width = 6, height = 5)
    
    
 #######################################################
    ##### Permanova for oral
    #######################################################
    oral_followup_se$studydayfollowup[is.na(oral_followup_se$studydayfollowup)] <- "baseline"
oral_followup_se$total_ABx_score_day1_Convex[is.na(oral_followup_se$total_ABx_score_day1_Convex)] <- 0
    permanova_oralfollowup <- vegan::adonis2(t(assay(oral_followup_se,"relabundance")) ~ studydayfollowup + total_ABx_score_day1_Convex,
                                           data = colData(oral_followup_se),
                                           permutations = 9999)

# total_ABx_score_day1_Convex   1    0.408 0.00611 1.3428 0.1785 
    
    ############### examine anaerobes overtime     
    
    
################# similar graph for the OPS samples. 
################################################################################################
oral_followup_rel_abund_assay <- as.data.frame(oral_followup_rel_abund_assay)

oral_followup_rel_abund_assay$Genus <- rownames(oral_followup_rel_abund_assay)

oral_followup_rel_abund_assay <- merge(oral_followup_rel_abund_assay, bugclass_genus, by = "Genus", all.x = TRUE)

table(oral_followup_rel_abund_assay$O2Req)
##### fix some missing classifications
 oral_followup_rel_abund_assay$Genus[is.na(oral_followup_rel_abund_assay$O2Req)]
 oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Cardiobacterium"] <- "Aerobe"
 oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Slackia"] <- "Anaerobe"
  oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Streptobacillus"] <- "Facultative anaerobe"
   oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Tumebacillus"] <- "Aerobe"
  oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Peptostreptococcaceae_ge"] <- "Anaerobe"
  
   oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Tropheryma"] <- "Aerobe"
   
    oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Paracoccus"] <- "Facultative anaerobe"
    
       oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Kocuria"] <- "Aerobe"
       
          oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Cryptobacterium"] <- "Anaerobe"
          
             oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Butyrivibrio"] <- "Anaerobe"
             
              oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Staphylococcaceae_uncl"] <- "Facultative anaerobe"
              
              oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Sphingomonadaceae_uncl"] <- "Facultative anaerobe"
              
               oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Burkholderia_Caballeronia_Paraburkholderia"] <- "Aerobe"
               oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Eggerthia"] <- "Anaerobe"
                              oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Peptoanaerobacter"] <- "Anaerobe"
  oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Prevotellaceae_Uncltrd"] <- "Anaerobe"
               
           oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Comamonas"] <- "Aerobe"
           oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Bilophila"] <- "Anaerobe"
           
           oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Mobiluncus"] <- "Anaerobe"
           oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Moryella"] <- "Anaerobe"
           oral_followup_rel_abund_assay$O2Req[oral_followup_rel_abund_assay$Genus=="Burkholderiales_uncl"] <- "Aerobe"
           
           
           names(oral_followup_rel_abund_assay)
               

           
###### take the df from tracheal_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
oral_followup_rel_abund_assay$O2Req[is.na(oral_followup_rel_abund_assay$O2Req)] <- "Unclassifiable"

oralbugsbyOxygen_followup <- as.data.frame(t(oral_followup_rel_abund_assay))
### remove last row of pathogens
oralbugsbyOxygen_followup <- oralbugsbyOxygen_followup[1:217, ]

#### take the names of the genera from the classified dataframe to create vectors that will then be used to calculate rowsums. 

anaerobeslist <- oral_followup_rel_abund_assay$Genus[oral_followup_rel_abund_assay$O2Req=="Anaerobe"]
aerobeslist <- oral_followup_rel_abund_assay$Genus[oral_followup_rel_abund_assay$O2Req=="Aerobe"]
facultanaerobeslist <- oral_followup_rel_abund_assay$Genus[oral_followup_rel_abund_assay$O2Req=="Facultative anaerobe"]
Microaerophilelist <- oral_followup_rel_abund_assay$Genus[oral_followup_rel_abund_assay$O2Req=="Microaerophile"]
Unclassifiableslist <- oral_followup_rel_abund_assay$Genus[oral_followup_rel_abund_assay$O2Req=="Unclassifiable"]
Variablelist <- oral_followup_rel_abund_assay$Genus[oral_followup_rel_abund_assay$O2Req=="Variable"]

#### in the oralbugsbyoxygen, make colnames by the genus variable
colnames(oralbugsbyOxygen_followup) <- oralbugsbyOxygen_followup[1,]
#### remove the anaerobe and 02 req rows
oralbugsbyOxygen_followup <- oralbugsbyOxygen_followup[2:216, ]

#### calculate rowsums by the list selection criteria. 
# make all values numeric - include some code to maintain the rownames by sample_id with this conversion

oralbugsbyOxygen_followupSample_id  <-as.data.frame(rownames(oralbugsbyOxygen_followup))
oralbugsbyOxygen_followup <- as.data.frame(sapply(oralbugsbyOxygen_followup,as.numeric))

oralbugsbyOxygen_followup$x <- rownames(oralbugsbyOxygen_followup)
oralbugsbyOxygen_followupSample_id$x <- rownames(oralbugsbyOxygen_followupSample_id)
oralbugsbyOxygen_followup <- merge(oralbugsbyOxygen_followup, oralbugsbyOxygen_followupSample_id, by = "x")
rownames(oralbugsbyOxygen_followup) <- oralbugsbyOxygen_followup$`rownames(oralbugsbyOxygen_followup)`
oralbugsbyOxygen_followup$x <- NULL

# calculate row sums by the desired list
# https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums
oralbugsbyOxygen_followup <- oralbugsbyOxygen_followup %>% 
  mutate(AnaerobeSum = rowSums(select(., .dots = all_of(anaerobeslist))))
oralbugsbyOxygen_followup$AnaerobeSum

oralbugsbyOxygen_followup <- oralbugsbyOxygen_followup %>% 
  mutate(AerobeSum = rowSums(select(., .dots = all_of(aerobeslist))))
oralbugsbyOxygen_followup$AnaerobeSum

oralbugsbyOxygen_followup <- oralbugsbyOxygen_followup %>% 
  mutate(FacultAnaerobeSum = rowSums(select(., .dots = all_of(facultanaerobeslist))))
oralbugsbyOxygen_followup$AnaerobeSum

oralbugsbyOxygen_followup <- oralbugsbyOxygen_followup %>% 
  mutate(MicroaerophileSum = rowSums(select(., .dots = all_of(Microaerophilelist))))
oralbugsbyOxygen_followup$AnaerobeSum

oralbugsbyOxygen_followup <- oralbugsbyOxygen_followup %>% 
  mutate(VariableSum = rowSums(select(., .dots = all_of(Variablelist))))
oralbugsbyOxygen_followup$AnaerobeSum

oralbugsbyOxygen_followup <- oralbugsbyOxygen_followup %>% 
  mutate(UnclassifiableSum = rowSums(select(., .dots = all_of(Unclassifiableslist))))
oralbugsbyOxygen_followup$AnaerobeSum

oralbugsbyOxygen_followup$totalO2sum <- oralbugsbyOxygen_followup$AnaerobeSum + oralbugsbyOxygen_followup$AerobeSum + oralbugsbyOxygen_followup$FacultAnaerobeSum  + oralbugsbyOxygen_followup$MicroaerophileSum  + oralbugsbyOxygen_followup$VariableSum + oralbugsbyOxygen_followup$UnclassifiableSum 
###mutate total to 1 and then unclassifiable as difference from total
oralbugsbyOxygen_followup$totalO2sum <- 1
oralbugsbyOxygen_followup$UnclassifiableSum <- 1 - (oralbugsbyOxygen_followup$AnaerobeSum + oralbugsbyOxygen_followup$AerobeSum + oralbugsbyOxygen_followup$FacultAnaerobeSum  + oralbugsbyOxygen_followup$MicroaerophileSum  + oralbugsbyOxygen_followup$VariableSum )

oralbugsbyOxygen_followupSum <- subset(oralbugsbyOxygen_followup, select = c(AnaerobeSum, AerobeSum, FacultAnaerobeSum,MicroaerophileSum, VariableSum , UnclassifiableSum))


### incorporate with the oralfollowupdf

oral_df_followup_sum <- merge(oral_followup_df,oralbugsbyOxygen_followupSum, by = 'row.names', all = TRUE)
    
    
# Pairwise comparisons. stratified by anaerobe antibiotics.    ###### very important finding
oral_df_followup_sum$AbxAnaerobicCoverage[oral_df_followup_sum$day1AnaerobicAbx=="no"] <- "NoAnaerobicCoverage"
   oral_df_followup_sum$AbxAnaerobicCoverage[oral_df_followup_sum$day1AnaerobicAbx=="yes"] <- "AnaerobicCoverage"                                       
mycomparisons <- list(c("baseline", "middle"), c("middle", "late"), c("baseline", "late"))
anaerobes_oral_followup <-  oral_df_followup_sum %>% filter(!is.na(studydayfollowup)) %>% ggboxplot(x = "studydayfollowup", y = "AnaerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Obligative Anaerobe Abundance - OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) + facet_wrap(~AbxAnaerobicCoverage)


aerobes_oral_followup <-  oral_df_followup_sum %>% filter(!is.na(studydayfollowup)) %>% ggboxplot(x = "studydayfollowup", y = "AerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Aerobe Abundance - OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) + facet_wrap(~AbxAnaerobicCoverage)

facultativeanaerobes_oral_followup <-  oral_df_followup_sum %>% filter(!is.na(studydayfollowup)) %>% ggboxplot(x = "studydayfollowup", y = "FacultAnaerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Facultative Anaerobe Abundance - OPS", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) + facet_wrap(~AbxAnaerobicCoverage)


comparAbundO2reqOralFollowupByGroup <- ggarrange(anaerobes_oral_followup, aerobes_oral_followup, facultativeanaerobes_oral_followup, ncol = 3, nrow = 1)
ggsave("./results/comparAbundO2reqOralFollowupByGroup.jpeg", width = 15, height = 5)

##### look at anaerobes overtime by antibiotics. 

model=lme(AnaerobeSum~timesamplefromintubation + total_ABx_score_day1_Convex, random=~1|SubjectID, data = oral_df_followup_sum, method = "REML", na.action = na.omit)
anova(model) # significant for time # p=0.003 not for total antibiotics. 
model=lme(AnaerobeSum~timesamplefromintubation + day1AnaerobicAbx, random=~1|SubjectID, data = oral_df_followup_sum, method = "REML", na.action = na.omit)
anova(model) # significant for time # p=0.003 not for total antibiotics. 


################ TRACHEAL FOLLOWUP
    
    # ensure that metadata and taxa table match for the oral dataset
trachealMetadataFollowup <- AspMetadata[which(AspMetadata$study_id=="alir" & AspMetadata$sampletype=="tracheal" ),] # 238 observations.

# set rownames by sample_id
rownames(trachealMetadataFollowup) <- trachealMetadataFollowup$sample_id
alirmatchingids_v <- as.character(trachealMetadataFollowup$sample_id)
trachealTaxatableFollowup <- subset(taxatableAsp, rownames(taxatableAsp) %in% alirmatchingids_v)
#### arrange by rownames. 
trachealMetadataFollowup <-trachealMetadataFollowup[sort(rownames(trachealMetadataFollowup)), ]
trachealTaxatableFollowup <-trachealTaxatableFollowup[sort(rownames(trachealTaxatableFollowup)), ]
# ensure all rows equal
all.equal(rownames(trachealMetadataFollowup), rownames(trachealTaxatableFollowup))


  ##########################################################
  # quality control: checking counts per sample
  #########################################################
reads.per.sample <- trachealMetadataFollowup$total
hist(reads.per.sample, breaks = 100)
samples_with_few_reads = trachealMetadataFollowup$sample_id[reads.per.sample<300]
### remove samples with <300 reads from analyses for 16s
trachealMetadataFollowup <- subset(trachealMetadataFollowup, !rownames(trachealMetadataFollowup) %in% samples_with_few_reads) # 215 samples
trachealTaxatableFollowup <- subset(trachealTaxatableFollowup, !rownames(trachealTaxatableFollowup) %in% samples_with_few_reads) # 215 samples
# remain matched
all.equal(rownames(trachealMetadataFollowup), rownames(trachealTaxatableFollowup))

  # remove total
  trachealTaxatableFollowup = trachealTaxatableFollowup[,-1]
  
  ####################################################################################
  # Creating a SingleCellExperiment object for the downstream analysis
  ################################################################################
  # separate column names by taxa
  tracheal_df_followup <- data.frame(x = names(trachealTaxatableFollowup))
  taxa <- tracheal_df_followup %>% separate(x, c("Kingdom", "Phylum", "Class", "Order", 
                               "Family", "Genus"), sep = "\\.")
  # use Genus as column names
  names(trachealTaxatableFollowup) <- taxa$Genus
   # create a single cell experiment object
  tracheal_followup_se <- SingleCellExperiment(assays = list(counts = t(trachealTaxatableFollowup)), ### transposed the taxatable (data) #The assays slot contains the experimental data as count matrices. Multiple matrices can be stored the result of assays is actually a list of matrices.
                             colData = trachealMetadataFollowup, ### contains the data on samples
                             rowData = taxa) # rowData contains data on the features of the analyzed samples. Of particular interest for the microbiome field this is used to store taxonomic information. of note, the taxa dataframe here includes all 6 levels of the taxonomy
  
  
  ### Transformations
  tracheal_followup_se <- transformSamples(tracheal_followup_se, method = "relabundance")
  tracheal_followup_se <- transformSamples(x = tracheal_followup_se, abund_values = "counts", method = "clr", 
                         pseudocount = 1, name = "clr")
 #     head(assay(se, "clr"))
  
 

  # get taxa that exceed the given prevalence and detection thresholds
  # detection: relative abundance threshold
  # prevalence: relative population frequencies; at 1% compositional abundance threshold
  tracheal_followup_se <- subsetByPrevalentTaxa(tracheal_followup_se,rank = "Genus", detection = 1/10000, abund_values = "relabundance",
                              prevalence = 5/100)
 
    ##########################################################################################
    # Alpha Diversity - tracheal
    #########################################################################################
    tracheal_followup_se <- mia::estimateDiversity(tracheal_followup_se, 
                                 abund_values = "relabundance", # can either be counts or relabundance
                                 index = "shannon", 
                                 name = "shannon")
    tracheal_followup_df <- as.data.frame(colData(tracheal_followup_se)[colData(tracheal_followup_se)$AspPnaGroup %in% 
                                      c("MAsP", "NonMAsP", "IntubControl"), ])
    tracheal_followup_df$AspPnaGroup <- factor(tracheal_followup_df$AspPnaGroup)
    tracheal_followup_df$AspPnaGroup <- ordered(tracheal_followup_df$AspPnaGroup, levels = c("MAsP", "NonMAsP", "IntubControl"))
    # this is a complicated way of finding the comparison groups. 
  #   comp <- split(t(combn(levels(df$AspPnaGroup), 2)), 
     #               seq(nrow(t(combn(levels(df$AspPnaGroup), 2)))))
    
    
    #### obtain colorblind friendly palette
# The palette with grey:
cbPalette <- c("#999999", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")

# The palette with black:
cbbPalette <- c("#000000", "#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7")
    
# Pairwise comparisons.

tracheal_followup_df$studydayfollowup <- ordered(tracheal_followup_df$studydayfollowup, levels = c("baseline", "middle", "late"))
mycomparisons <- list(c("baseline", "middle"), c("middle", "late"), c("baseline", "late"))
shannon_followup_tracheal <- tracheal_followup_df %>% filter(!is.na(studydayfollowup)) %>% ggboxplot( x = "studydayfollowup", y = "shannon", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Shannon Index - ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) 
ggsave("./results/shannon_followup_tracheal.jpeg",  width = 10, height = 5)


qpcr_followup_tracheal <- tracheal_followup_df %>% filter(!is.na(studydayfollowup)) %>% ggboxplot( x = "studydayfollowup", y = "qPCR_16S_mean", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "16S rRNA gene copies (qpCR) - ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) + scale_y_log10()
ggsave("./results/qpcr_followup_tracheal.jpeg",  width = 10, height = 5)


##### model the impact of antibiotics and time from intubation. 
# shannon model over time
model=lme(shannon~timesamplefromintubation + total_ABx_score_day1_Convex, random=~1|SubjectID, data = tracheal_followup_df, method = "REML", na.action = na.omit)
anova(model) # significant for time. 0.00151 **

# qpcr model over time
model=lme(qPCR_16S_mean~timesamplefromintubation + total_ABx_score_day1_Convex, random=~1|SubjectID, data = tracheal_followup_df, method = "REML", na.action = na.omit)
anova(model) # ns








    #######################################################################################################
    # Beta Diversity tracheal
    ######################################################################################################
    tracheal_followup_rel_abund_assay <- assays(tracheal_followup_se)$relabundance
    
    # Calculates distances between samples. Because taxa is in
    # columns, it is used to compare different samples. We transpose the
    # assay to get taxa to columns
    tracheal_followup_manhattan_dist <- vegan::vegdist(t(tracheal_followup_rel_abund_assay), method = "manhattan")
    # tracheal_manhattan_mds <- vegan::metaMDS(tracheal_manhattan_dist)
    tracheal__followup_manhattan_pcoa <- ecodist::pco(tracheal_followup_manhattan_dist)
    # Creates a data frame from principal coordinates
    tracheal_followup_manhattan_pcoa_df <- data.frame(pcoa1 = tracheal__followup_manhattan_pcoa$vectors[,1], 
                                     pcoa2 = tracheal__followup_manhattan_pcoa$vectors[,2])
    
    ## merge with metadata to make this analyzable by ggpot as per Schloss youtube video approach
    rownames(tracheal_followup_manhattan_pcoa_df) <-  rownames(trachealMetadataFollowup)
    tracheal_followup_manhattan_pcoa_df$sample_id <-  rownames(tracheal_followup_manhattan_pcoa_df) 
    trachealMetadataBas_nmds_followup <- merge(trachealMetadataFollowup, tracheal_followup_manhattan_pcoa_df, by = "sample_id", all = TRUE) 
    
#### obtain centroids
centroid_tracheal_followup <- trachealMetadataBas_nmds_followup %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa1mean = mean(pcoa1), pcoa2mean = mean(pcoa2))
    
trachealMetadataBas_nmds_followup %>% filter(studydayfollowup=="baseline") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
trachealMetadataBas_nmds_followup %>% filter(studydayfollowup=="middle") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))
trachealMetadataBas_nmds_followup %>% filter(studydayfollowup=="late") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa1mean = mean(pcoa1, na.rm = TRUE))

trachealMetadataBas_nmds_followup <- trachealMetadataBas_nmds_followup %>% mutate(pcoa1mean = case_when(studydayfollowup=="baseline" ~ 0.001463062, 
                             studydayfollowup=="middle" ~  -0.003963391,
                             studydayfollowup=="late" ~ -0.006372237
                                                ))



trachealMetadataBas_nmds_followup %>% filter(studydayfollowup=="baseline") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
trachealMetadataBas_nmds_followup %>% filter(studydayfollowup=="middle") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))
trachealMetadataBas_nmds_followup %>% filter(studydayfollowup=="late") %>% dplyr::group_by(studydayfollowup) %>% summarize(pcoa2mean = mean(pcoa2, na.rm = TRUE))


trachealMetadataBas_nmds_followup <- trachealMetadataBas_nmds_followup %>% mutate(pcoa2mean = case_when(studydayfollowup=="baseline" ~ -0.0005487967, 
                             studydayfollowup=="middle" ~  0.001033985,
                             studydayfollowup=="late" ~ 0.005748268
                                                             ))

tracheal_followup_pcoa <- 
ggplot(trachealMetadataBas_nmds_followup, aes(x=pcoa1, xend = pcoa1mean,y = pcoa2, yend = pcoa2mean, color = studydayfollowup, fill = studydayfollowup)) + 
 stat_ellipse(level = 0.8,  show.legend = FALSE, size = 1.2) +
  geom_point(size=1.5, show.legend = FALSE) +
geom_point(data = trachealMetadataBas_nmds_followup, mapping = aes(x=pcoa1mean, y = pcoa2mean), size = 5, show.legend = FALSE) +
geom_segment(size = 0.1, show.legend = FALSE) +
labs(x= "PCoA axis1", y = "PCoA axis 2") + scale_color_manual(name = NULL, breaks = c("baseline", "middle", "late"), values = c("black", "blue", "red")) + theme_classic() +ggtitle("PCoA for ETA samples") + annotate(geom="text", x=0, y=-0.03, label="R2=0.026, p<0.0001",color="red")
#+ theme(legend.key.height = unit(0.5, "cm"), legend.position = c(0.8,0.2), legend.background = element_rect(fill = "NA", color = "gray"))
ggsave("./results/tracheal_followup_pcoa.jpeg", width = 6, height = 5)
    
    
 #######################################################
    ##### Permanova for tracheal
    #######################################################
    tracheal_followup_se$studydayfollowup[is.na(tracheal_followup_se$studydayfollowup)] <- "baseline"
tracheal_followup_se$total_ABx_score_day1_Convex[is.na(tracheal_followup_se$total_ABx_score_day1_Convex)] <- 0
    permanova_trachealfollowup <- vegan::adonis2(t(assay(tracheal_followup_se,"relabundance")) ~ studydayfollowup + total_ABx_score_day1_Convex,
                                           data = colData(tracheal_followup_se),
                                           permutations = 9999)
    # ns for antibiotics
    
    
############# analyses by 02 requirement in followup tracheal samples
    
    

tracheal_followup_rel_abund_assay <- as.data.frame(tracheal_followup_rel_abund_assay)
tracheal_followup_rel_abund_assay$Genus <- rownames(tracheal_followup_rel_abund_assay)

tracheal_followup_rel_abund_assay_df <- merge(tracheal_followup_rel_abund_assay, bugclass_genus, by = "Genus", all.x = TRUE)

table(tracheal_followup_rel_abund_assay_df$O2Req)
##### fix some missing classifications
 tracheal_followup_rel_abund_assay_df$Genus[is.na(tracheal_followup_rel_abund_assay_df$O2Req)]
 tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Cardiobacterium"] <- "Aerobe"
 tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Slackia"] <- "Anaerobe"
  tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Streptobacillus"] <- "Facultative anaerobe"
   tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Tumebacillus"] <- "Aerobe"
  tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Peptostreptococcaceae_ge"] <- "Anaerobe"
  
   tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Tropheryma"] <- "Aerobe"
   
    tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Paracoccus"] <- "Facultative anaerobe"
    
       tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Kocuria"] <- "Aerobe"
       
          tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Cryptobacterium"] <- "Anaerobe"
          
             tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Butyrivibrio"] <- "Anaerobe"
             
              tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Staphylococcaceae_uncl"] <- "Facultative anaerobe"
              
              tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Sphingomonadaceae_uncl"] <- "Facultative anaerobe"
              
               tracheal_followup_rel_abund_assay_df$O2Req[tracheal_followup_rel_abund_assay_df$Genus=="Burkholderia_Caballeronia_Paraburkholderia"] <- "Aerobe"
               
               names(tracheal_followup_rel_abund_assay_df)
               


###### take the df from tracheal_rel abundance and transpose to have rows with sampleIDs and allow for rowsum calculations
tracheal_followup_rel_abund_assay_df$O2Req[is.na(tracheal_followup_rel_abund_assay_df$O2Req)] <- "Unclassifiable"

trachealbugsbyOxygen_followup <- as.data.frame(t(tracheal_followup_rel_abund_assay_df))
### remove last row of pathogens
trachealbugsbyOxygen_followup <- trachealbugsbyOxygen_followup[1:218, ]

#### take the names of the genera from the classified dataframe to create vectors that will then be used to calculate rowsums. 

anaerobeslist <- tracheal_followup_rel_abund_assay_df$Genus[tracheal_followup_rel_abund_assay_df$O2Req=="Anaerobe"]
aerobeslist <- tracheal_followup_rel_abund_assay_df$Genus[tracheal_followup_rel_abund_assay_df$O2Req=="Aerobe"]
facultanaerobeslist <- tracheal_followup_rel_abund_assay_df$Genus[tracheal_followup_rel_abund_assay_df$O2Req=="Facultative anaerobe"]


#### in the trachealbugsbyoxygen, make colnames by the genus variable
colnames(trachealbugsbyOxygen_followup) <- trachealbugsbyOxygen_followup[1,]
#### remove the anaerobe and 02 req rows
trachealbugsbyOxygen_followup <- trachealbugsbyOxygen_followup[2:217, ]

#### calculate rowsums by the list selection criteria. 
# make all values numeric - include some code to maintain the rownames by sample_id with this conversion

trachealbugsbyOxygen_followupSample_id  <-as.data.frame(rownames(trachealbugsbyOxygen_followup))
trachealbugsbyOxygen_followup <- as.data.frame(sapply(trachealbugsbyOxygen_followup,as.numeric))

trachealbugsbyOxygen_followup$x <- rownames(trachealbugsbyOxygen_followup)
trachealbugsbyOxygen_followupSample_id$x <- rownames(trachealbugsbyOxygen_followupSample_id)
trachealbugsbyOxygen_followup <- merge(trachealbugsbyOxygen_followup, trachealbugsbyOxygen_followupSample_id, by = "x")
rownames(trachealbugsbyOxygen_followup) <- trachealbugsbyOxygen_followup$`rownames(trachealbugsbyOxygen_followup)`
trachealbugsbyOxygen_followup$x <- NULL

# calculate row sums by the desired list
# https://stackoverflow.com/questions/41895432/mutating-column-in-dplyr-using-rowsums
trachealbugsbyOxygen_followup <- trachealbugsbyOxygen_followup %>% 
  mutate(AnaerobeSum = rowSums(select(., .dots = all_of(anaerobeslist))))
trachealbugsbyOxygen_followup$AnaerobeSum

trachealbugsbyOxygen_followup <- trachealbugsbyOxygen_followup %>% 
  mutate(AerobeSum = rowSums(select(., .dots = all_of(aerobeslist))))
trachealbugsbyOxygen_followup$AnaerobeSum

trachealbugsbyOxygen_followup <- trachealbugsbyOxygen_followup %>% 
  mutate(FacultAnaerobeSum = rowSums(select(., .dots = all_of(facultanaerobeslist))))
trachealbugsbyOxygen_followup$AnaerobeSum

###mutate total to 1 and then unclassifiable as difference from total

trachealbugsbyOxygen_followupSum <- subset(trachealbugsbyOxygen_followup, select = c(AnaerobeSum, AerobeSum, FacultAnaerobeSum))

### incorporate with the tracheal_df 
tracheal_followup_df_sum <- merge(tracheal_followup_df,trachealbugsbyOxygen_followupSum, by = 'row.names', all = TRUE)
    

# Pairwise comparisons.

tracheal_followup_df_sum$AbxAnaerobicCoverage[tracheal_followup_df_sum$day1AnaerobicAbx=="no"] <- "NoAnaerobicCoverage"
   tracheal_followup_df_sum$AbxAnaerobicCoverage[tracheal_followup_df_sum$day1AnaerobicAbx=="yes"] <- "AnaerobicCoverage"       

mycomparisons <- list(c("baseline", "middle"), c("middle", "late"), c("baseline", "late"))
anaerobes_tracheal_followup <-  tracheal_followup_df_sum %>% filter(!is.na(studydayfollowup)) %>% ggboxplot(x = "studydayfollowup", y = "AnaerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Obligative Anaerobe Abundance - ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons) + facet_wrap(~AbxAnaerobicCoverage)


aerobes_tracheal_followup <-  tracheal_followup_df_sum %>% filter(!is.na(studydayfollowup)) %>% ggboxplot(x = "studydayfollowup", y = "AerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Aerobe Abundance - ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons)  + facet_wrap(~AbxAnaerobicCoverage)

facultativeanaerobes_tracheal_followup <-  tracheal_followup_df_sum %>% filter(!is.na(studydayfollowup)) %>% ggboxplot(x = "studydayfollowup", y = "FacultAnaerobeSum", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Facultative Anaerobe Abundance - ETA", legend = "none" ) + 
   stat_compare_means(comparisons = mycomparisons)  + facet_wrap(~AbxAnaerobicCoverage)



comparAbundO2reqtrachealFollowupByGroup <- ggarrange(anaerobes_tracheal_followup, aerobes_tracheal_followup, facultativeanaerobes_tracheal_followup, ncol = 3, nrow = 1)
ggsave("./results/comparAbundO2reqtrachealFollowupByGroup.jpeg", width = 15, height = 5)


###### combine figure on oxygen requirements 

 comparAbundO2reqOPSETA_followup <- ggarrange(comparAbundO2reqOralFollowupByGroup, comparAbundO2reqtrachealFollowupByGroup,  ncol = 1, nrow = 2)
ggsave("./results/comparAbundO2reqOPSETA_followup.jpeg", width = 15, height = 10)
  



```

```{r nanopore analyses}

#### import the clean nanopore file by Haopu
nanoporekindgdoms <- read.csv("./data/Task1D.Nanopore_reads_alignment_kingdom_breakdown.csv")

##### take relevant parameters from tracheal df
tracheal_df_nanopore <- subset(tracheal_df, select = c(sample_id, SubjectIDDay, SubjectID, AspPnaGroup, Mortality90Day))

#merge by SubjectIDDay
tracheal_df_nanopore <- merge(tracheal_df_nanopore,nanoporekindgdoms, by = "SubjectIDDay", all.x = TRUE )

table(is.na(tracheal_df_nanopore$Bacteria)) # 64 nonmissing. 
tracheal_df_nanopore <- filter(tracheal_df_nanopore, !is.na(Bacteria))
names(tracheal_df_nanopore)

tracheal_df_nanopore$Bacteria[tracheal_df_nanopore$Bacteria==0] <- 1
tracheal_df_nanopore$Eukaryota_Human[tracheal_df_nanopore$Eukaryota_Human==0] <- 1
tracheal_df_nanopore$Eukaryota_Non_human[tracheal_df_nanopore$Eukaryota_Non_human==0] <- 1
tracheal_df_nanopore$Virus[tracheal_df_nanopore$Virus==0] <- 1

table(tracheal_df_nanopore$AspPnaGroup)

ggplot(tracheal_df_nanopore, aes(x=AspPnaGroup, y = Bacteria)) + geom_boxplot() + geom_beeswarm() + stat_compare_means() + scale_y_log10()
ggplot(tracheal_df_nanopore, aes(x=AspPnaGroup, y = Eukaryota_Non_human)) + geom_boxplot() + geom_beeswarm() + stat_compare_means()  + scale_y_log10()
ggplot(tracheal_df_nanopore, aes(x=AspPnaGroup, y = Virus)) + geom_boxplot() + geom_beeswarm() + stat_compare_means()  + scale_y_log10()
ggplot(tracheal_df_nanopore, aes(x=AspPnaGroup, y = Eukaryota_Human)) + geom_boxplot() + geom_beeswarm() + stat_compare_means()  + scale_y_log10()

    mycomparisons <- list(c("MAsP", "NonMAsP"), c("MAsP", "IntubControl"))
    
nanoporebacterialbygroup <-   ggboxplot(tracheal_df_nanopore, x = "AspPnaGroup", y = "Bacteria", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Bacterial reads - ETA", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10()

nanoporefungalbygroup <-   ggboxplot(tracheal_df_nanopore, x = "AspPnaGroup", y = "Eukaryota_Non_human", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Fungal reads - ETA", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10()

nanoporevirallbygroup <-   ggboxplot(tracheal_df_nanopore, x = "AspPnaGroup", y = "Virus", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Viral reads - ETA", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10()

nanoporehumanbygroup <-   ggboxplot(tracheal_df_nanopore, x = "AspPnaGroup", y = "Eukaryota_Human", 
                   color = "AspPnaGroup",add = "jitter" , palette = cbbPalette, xlab = "", ylab = "Human DNA reads - ETA", legend = "none" ) + 
     stat_compare_means(comparisons = mycomparisons) + scale_y_log10()
nanoporeReadslbygroup <- ggarrange(nanoporebacterialbygroup,nanoporefungalbygroup, nanoporevirallbygroup , nanoporehumanbygroup, ncol = 2, nrow = 2)
ggsave("./results/nanoporeReadslbygroup.jpeg",  width = 10, height = 8)


###### in next phase can work on diversity analyses of nanopore

###### fungal analyses
fungalnanopore <-  read.csv("./data/Task1A.all_available_Nanopore_results_LONG_all_kingdom_QC_passed_taxa_noise_filtered.csv")
fungalnanopore <-filter(fungalnanopore, kingdom_text =="Eukaryota")

fungalnanopore <- merge(fungalnanopore, tracheal_df_nanopore, by = "SubjectIDDay", all.y = TRUE)

fungalnanopore$NCBI_species_name[fungalnanopore$NCBI_species_name=="[Candida] glabrata"] <- "Candida glabrata"

fungalReadsNanoporeByGroup <-   ggplot(fungalnanopore, aes(x=AspPnaGroup, y = count, fill = NCBI_species_name)) + geom_bar(stat = "identity") + coord_flip() +  scale_fill_viridis(discrete=TRUE, direction=-1) + scale_y_log10() + theme_pubr() + theme(legend.position = "right") + xlab("") + ylab("log10 Nanopore reads")
ggsave("./results/fungalReadsNanoporeByGroup.jpeg", width = 10, height = 6)


fungalnanoporecandida <- filter(fungalnanopore,  NCBI_species_name == "Candida albicans" | NCBI_species_name == "Candida dubliniensis"| NCBI_species_name == "Candida glabrata" | NCBI_species_name == "Candida tropicalis" | NCBI_species_name == "Candida orthopsilosis" )


ggplot(fungalnanoporecandida, aes(x=AspPnaGroup, y = count, fill = NCBI_species_name)) + geom_bar(stat = "identity") + coord_flip() +  scale_fill_viridis(discrete=TRUE, direction=-1) + scale_y_log10() + theme_pubr() + theme(legend.position = "right") + xlab("") + ylab("log10 Nanopore reads")

sum(fungalnanopore$count, na.rm = TRUE) # 75843
sum(fungalnanoporecandida$count, na.rm = TRUE)  # 71921

71921/75843 # 94.8% candida reads



#### bacterial reads
bacterialnanopore <-  read.csv("./data/Task1A.all_available_Nanopore_results_LONG_all_kingdom_QC_passed_taxa_noise_filtered.csv")
bacterialnanopore <-filter(bacterialnanopore, kingdom_text =="Bacteria")

 bacterialnanopore <- merge(bacterialnanopore, tracheal_df_nanopore, by = "SubjectIDDay", all.y = TRUE)

 bacterialnanopore <- filter(bacterialnanopore, count>1000)
bacteriaNanoporebyAspGroups  <- ggplot(bacterialnanopore, aes(x=AspPnaGroup, y = count, fill = NCBI_species_name)) + geom_bar(stat = "identity") + coord_flip() +  scale_fill_viridis(discrete=TRUE, direction=-1) + scale_y_log10() + theme_pubr() + theme(legend.position = "right") + xlab("") + ylab("log10 Nanopore reads")
 ggsave("./results/bacteriaNanoporebyAspGroups.jpeg", width = 10, height = 6)

 
 ##### look at Streptococcus
 streptococcusNanopore <- bacterialnanopore %>% filter(str_detect(NCBI_species_name, "Streptococcus"))
  
 
sum(bacterialnanopore$count, na.rm = TRUE) # 3940235
 sum(streptococcusNanopore$count, na.rm = TRUE) # 1213800
 
 1213800/3940235 # 30.8% of all bacterial reads. 
 
 
streptoccoccusbyAspPnaGroups <-  ggplot(streptococcusNanopore, aes(x=NCBI_species_name, y = count, fill = AspPnaGroup)) + geom_bar(stat = "identity") + coord_flip() +  scale_fill_viridis(discrete=TRUE, direction=1) + scale_y_log10() + theme_pubr() + theme(legend.position = "right") + xlab("") + ylab("log10 Nanopore reads")
  ggsave("./results/streptoccoccusbyAspPnaGroups.jpeg", width = 10, height = 6)

 
   pneumococcusNanopore <- bacterialnanopore %>% filter(str_detect(NCBI_species_name, "Streptococcus pneumoniae"))
  
  sum(pneumococcusNanopore$count, na.rm = TRUE) # 296077
  
  296077/1213800  # 24.3 or 100-24.3 = 75.7% non-pneumococcus
  
  
  

```

```{r exploratory}

AspMetadata_basl_alir %>% ggplot(aes(x=day1AnaerobicAbx, y = st2))  + geom_boxplot() + geom_beeswarm()+ stat_compare_means() + scale_y_log10()


```




```{r identify overlap with ajrccm paper}

alir16sajrccm <- read.csv("alir_mbio_meta_11.19_basl_alironly_subjectid.csv")

alir16sajrccm <- filter(alir16sajrccm, sampletype=="tracheal")

overlap <- merge(alir16sajrccm, AspMetadataSubjectID, by = "SubjectID", all.x = TRUE)

table(overlap$AspPnaGroup)
table(is.na(overlap$AspPnaGroup))

75/158
```

